// @opentiny/vue-tag-input/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/tag-input/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var tagInputProps = _extends({}, $props, {
  modelValue: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  size: {
    type: String,
    default: "medium",
    validator: function validator(value) {
      return ["medium", "small"].includes(value);
    }
  },
  tagType: {
    type: String,
    default: "",
    validator: function validator2(value) {
      return ["info", "success", "warning", "danger", "primary"].includes(value);
    }
  },
  tagEffect: {
    type: String,
    default: "light",
    validator: function validator3(value) {
      return ["dark", "light", "plain"].includes(value);
    }
  },
  clearable: {
    type: Boolean,
    default: false
  },
  disabled: {
    type: Boolean,
    default: false
  },
  placeholder: {
    type: String,
    default: ""
  },
  max: {
    type: Number,
    default: Infinity
  },
  readonly: {
    type: Boolean,
    default: false
  },
  separator: {
    type: String,
    default: void 0
  },
  minCollapsedNum: {
    type: Number,
    default: Infinity
  },
  draggable: {
    type: Boolean,
    default: false
  }
});
var TagInput = defineComponent({
  name: $prefix + "TagInput",
  props: tagInputProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
TagInput.install = function(Vue) {
  Vue.component(TagInput.name, TagInput);
};
TagInput.version = version;
export {
  TagInput as default
};
