// @opentiny/vue-tag-input/lib/pc.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/tag-input/vue';
import Tag from '@opentiny/vue-tag';
import Tooltip from '@opentiny/vue-tooltip';
import { iconClose } from '@opentiny/vue-icon';
import '@opentiny/vue-theme/tag-input/index.css';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, renderSlot, Fragment, renderList, createBlock, withCtx, createTextVNode, toDisplayString, createVNode, createCommentVNode, withDirectives, withKeys, vModelText } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "TagInput",
  components: {
    TinyTag: Tag,
    TinyTooltip: Tooltip,
    TinyIconClose: iconClose()
  },
  props: [].concat(props, ["size", "modelValue", "tagType", "tagEffect", "placeholder", "readonly", "disabled", "minCollapsedNum", "max", "clearable", "separator", "draggable"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-tag-input-wrapper"
};
var _hoisted_2 = {
  class: "tiny-tag-input-collapsed-tags"
};
var _hoisted_3 = ["placeholder", "disabled"];
var _hoisted_4 = {
  class: "tiny-tag-input-suffix-wrapper"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_tag = resolveComponent("tiny-tag");
  var _component_tiny_tooltip = resolveComponent("tiny-tooltip");
  var _component_tiny_icon_close = resolveComponent("tiny-icon-close");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-tag-input", _ctx.disabled ? "tiny-tag-input-disabled" : ""]),
      onMouseenter: _cache[6] || (_cache[6] = function() {
        return _ctx.handleMouseOver && _ctx.handleMouseOver.apply(_ctx, arguments);
      }),
      onMouseleave: _cache[7] || (_cache[7] = function() {
        return _ctx.handleMouseLeave && _ctx.handleMouseLeave.apply(_ctx, arguments);
      })
    },
    [createElementVNode("div", _hoisted_1, [renderSlot(_ctx.$slots, "prefix"), (openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.showTagList, function(tag, index) {
        return openBlock(), createBlock(_component_tiny_tag, {
          key: index,
          size: _ctx.size,
          type: _ctx.tagType,
          effect: _ctx.tagEffect,
          closable: _ctx.state.closeable,
          draggable: _ctx.draggable,
          onClose: function onClose($event) {
            return _ctx.removeTag(index);
          },
          onDragstart: function onDragstart($event) {
            return _ctx.handleDragStart(index, $event);
          },
          onDragover: function onDragover($event) {
            return _ctx.handleDragOver(index, $event);
          },
          onDragenter: function onDragenter($event) {
            return _ctx.handleDragEnter(index, $event);
          },
          onDragend: function onDragend($event) {
            return _ctx.handleDrop(index, $event);
          }
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(tag),
              1
              /* TEXT */
            )];
          }),
          _: 2
          /* DYNAMIC */
        }, 1032, ["size", "type", "effect", "closable", "draggable", "onClose", "onDragstart", "onDragover", "onDragenter", "onDragend"]);
      }),
      128
      /* KEYED_FRAGMENT */
    )), _ctx.state.collapsedTagList.length ? (openBlock(), createBlock(_component_tiny_tooltip, {
      key: 0,
      placement: "top"
    }, {
      content: withCtx(function() {
        return [createElementVNode("div", _hoisted_2, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.collapsedTagList, function(tag, index) {
            return openBlock(), createBlock(_component_tiny_tag, {
              key: index,
              size: _ctx.size,
              type: _ctx.tagType,
              effect: _ctx.tagEffect,
              closable: _ctx.state.closeable,
              onClose: function onClose($event) {
                return _ctx.removeTag(_ctx.state.showTagList.length + index);
              }
            }, {
              default: withCtx(function() {
                return [createTextVNode(
                  toDisplayString(tag),
                  1
                  /* TEXT */
                )];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["size", "type", "effect", "closable", "onClose"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])];
      }),
      default: withCtx(function() {
        return [createVNode(_component_tiny_tag, {
          size: _ctx.size,
          type: _ctx.tagType,
          effect: _ctx.tagEffect
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              "+" + toDisplayString(_ctx.state.collapsedTagList.length),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["size", "type", "effect"])];
      }),
      _: 1
      /* STABLE */
    })) : createCommentVNode("v-if", true), !_ctx.readonly ? withDirectives((openBlock(), createElementBlock("input", {
      key: 1,
      class: "tiny-tag-input-inner",
      "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
        return _ctx.state.currentValue = $event;
      }),
      placeholder: _ctx.placeholder,
      disabled: _ctx.disabled,
      onKeydown: [_cache[1] || (_cache[1] = withKeys(function() {
        return _ctx.addTag && _ctx.addTag.apply(_ctx, arguments);
      }, ["enter"])), _cache[2] || (_cache[2] = withKeys(function() {
        return _ctx.handleBackspace && _ctx.handleBackspace.apply(_ctx, arguments);
      }, ["backspace"]))],
      onFocus: _cache[3] || (_cache[3] = function() {
        return _ctx.handleInputFocus && _ctx.handleInputFocus.apply(_ctx, arguments);
      }),
      onBlur: _cache[4] || (_cache[4] = function() {
        return _ctx.handleInputBlur && _ctx.handleInputBlur.apply(_ctx, arguments);
      })
    }, null, 40, _hoisted_3)), [[vModelText, _ctx.state.currentValue]]) : createCommentVNode("v-if", true)]), createElementVNode("div", _hoisted_4, [renderSlot(_ctx.$slots, "suffix"), _ctx.state.showClearIcon ? (openBlock(), createElementBlock("div", {
      key: 0,
      onClick: _cache[5] || (_cache[5] = function() {
        return _ctx.handleClear && _ctx.handleClear.apply(_ctx, arguments);
      })
    }, [createVNode(_component_tiny_icon_close, {
      class: "tiny-tag-input-clear-icon"
    })])) : createCommentVNode("v-if", true)])],
    34
    /* CLASS, NEED_HYDRATION */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
