// @opentiny/vue-hooks@3.30.0/dist/src/use-floating.js
import { autoUpdate as v, offset as C, flip as T, shift as B, limitShift as E, arrow as N, hide as O, computePosition as $ } from "@floating-ui/dom";
const h = {
  reference: null,
  popper: null,
  show: !1,
  autoUpdate: !0,
  strategy: "absolute",
  placement: "bottom",
  offset: 6,
  arrowVisible: !0,
  rootBoundary: "viewport",
  boundary: "clippingAncestors",
  boundaryPadding: 5,
  syncHide: !0,
  autoHide: !1,
  gpuAcceleration: !1,
  animate: !0,
  animateName: "fade-in-linear",
  appendToBody: !1,
  customClass: "",
  flipAble: !0,
  shiftAble: !0
}, j = { top: "bottom", right: "left", bottom: "top", left: "right" }, d = (e) => e === "auto" ? 0 : Number(e.slice(0, -1).replace(",", ".")) * 1e3, w = (e) => {
  const p = window.getComputedStyle(e);
  let r = d(p.transitionDelay) + d(p.transitionDuration);
  return r || (r = d(p.animationDelay) + d(p.animationDuration), r) ? r : 0;
}, _ = (e, p, r) => {
  p.split(" ").forEach((n) => n && e.classList.toggle(n, r));
}, b = (e) => {
  const p = [C(e.offset)];
  e.flipAble && p.push(
    T({
      rootBoundary: e.rootBoundary,
      boundary: e.boundary,
      padding: e.boundaryPadding
    })
  ), e.shiftAble && p.push(B({ limiter: E() })), e.arrowVisible && p.push(
    N({
      element: e.popper.querySelector(".tiny-popper__arrow"),
      padding: 8
    })
  ), p.push(O()), $(e.reference, e.popper, {
    placement: e.placement,
    strategy: e.strategy,
    middleware: p
  }).then(({ x: r, y: n, placement: c, strategy: o, middlewareData: i }) => {
    var u;
    if (e.autoHide && e._last.show && (/* @__PURE__ */ new Date()).getTime() > e._last.timestamp + 300) {
      e.show = !1;
      return;
    }
    const s = {};
    if (Object.assign(s, {
      position: o
    }), e.gpuAcceleration ? Object.assign(s, {
      transform: `translate(${r}px,${n}px)`,
      left: "0",
      top: "0"
    }) : Object.assign(s, {
      left: `${r}px`,
      top: `${n}px`
    }), e.syncHide && i.hide && Object.assign(s, {
      visibility: i.hide.referenceHidden ? "hidden" : "visible"
    }), Object.assign(e.popper.style, s), e._last.customClass && e._last.customClass !== e.customClass && _(e.popper, e._last.customClass, !1), e.customClass && ((u = e._last) == null ? void 0 : u.customClass) !== e.customClass && (_(e.popper, e.customClass, !0), e._last.customClass = e.customClass), e.arrowVisible) {
      const { x: t, y: l } = i.arrow, a = e._last.arrowEl, m = j[c.split("-")[0]], g = {
        left: t !== null ? `${t}px` : "",
        top: l !== null ? `${l}px` : "",
        right: "",
        bottom: "",
        [m]: "-4px",
        display: "block"
      };
      Object.assign(a.style, g);
    } else
      e._last.arrowInserted && (e._last.arrowEl.style.display = "none");
    f(e, "update", { x: r, y: n, placement: c, strategy: o, middlewareData: i });
  });
}, P = (e) => v(e.reference, e.popper, () => {
  b(e);
}), S = (e) => {
  if (!(e._last.show && e._last.popper === e.popper) && (e._last.popper && e._last.popper !== e.popper && (e._last.appendToBody ? e._last.popper.remove() : e._last.popper.style.display = "none", e._last.arrowInserted = !1, e._last.arrowEl = null), e.popper)) {
    if (e.appendToBody ? document.body.append(e.popper) : e.popper.style.display = "block", !e._last.arrowInserted) {
      const p = document.createElement("div");
      p.className = "tiny-popper__arrow", e.popper.append(p), e._last.arrowInserted = !0, e._last.arrowEl = p;
    }
    if (e.animate) {
      const p = `${e.animateName}-enter-from`, r = `${e.animateName}-enter-active`;
      e.popper.classList.add(p, r), setTimeout(() => {
        e.popper.classList.remove(p);
      }, 0);
      const n = w(e.popper);
      setTimeout(() => {
        e.popper.classList.remove(r);
      }, n);
    }
    f(e, "show");
  }
}, y = (e) => {
  if (e._last.show && e.popper)
    if (e.animate && e.animateName) {
      const p = `${e.animateName}-leave-to`, r = `${e.animateName}-leave-active`;
      e.popper.classList.add(p, r);
      const n = w(e.popper);
      setTimeout(() => {
        e.popper.classList.remove(p, r), e.appendToBody ? e.popper.remove() : e.popper.style.display = "none", f(e, "hide");
      }, n);
    } else
      e.appendToBody ? e.popper.remove() : e.popper.style.display = "none", f(e, "hide");
}, f = (e, p, r) => {
  e._events[p].forEach((n) => n(r));
}, A = (e, p, r = 0, n = 0) => ({
  getBoundingClientRect() {
    return {
      width: 0,
      height: 0,
      x: e,
      y: p,
      top: p,
      left: e,
      right: e + r,
      bottom: p + n
    };
  }
}), x = ({ reactive: e, watch: p, markRaw: r, onBeforeUnmount: n }) => (c = {}) => {
  const o = e(c);
  let i = null;
  Object.keys(h).forEach((l) => {
    Object.prototype.hasOwnProperty.call(o, l) || (o[l] = h[l]);
  }), o._last = r({}), o._events = r({ show: [], hide: [], update: [] });
  const s = () => {
    o.popper && o.reference && o.show ? (S(o), o.autoUpdate ? (i && i(), i = P(o)) : b(o)) : (i && i(), y(o)), o._last.popper = o.popper, o._last.reference = o.reference, o._last.show = o.show && o.popper && o.reference, o._last.appendToBody = o.appendToBody, o._last.timestamp = (/* @__PURE__ */ new Date()).getTime();
  };
  p(o, s, { immediate: !0 });
  const u = (l, a) => o._events[l].push(a), t = (l, a) => o._events[l] = o._events[l].filter((m) => m !== a);
  return n(() => {
    i && i(), y(o);
  }), { state: o, on: u, off: t, virtualEl: A, forceUpdate: s };
};
export {
  x as useFloating
};
