// @opentiny/vue-switch@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var $constants = {
  PC_PREFIXCLS: "tiny-switch",
  MOBILE_PREFIXCLS: "tiny-mobile-switch",
  Mode: "pc",
  prefixcls: function prefixcls(mode) {
    return mode === this.Mode ? this.PC_PREFIXCLS : this.MOBILE_PREFIXCLS;
  }
};
var switchProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  disabled: {
    type: Boolean,
    default: false
  },
  showText: {
    type: Boolean || void 0,
    default: false
  },
  types: {
    type: String
  },
  falseColor: String,
  falseValue: {
    type: [String, Number, Boolean],
    default: false
  },
  mini: {
    type: Boolean,
    default: false
  },
  modelValue: {
    type: [String, Number, Boolean],
    default: false
  },
  size: [Number, String],
  tabindex: {
    type: String,
    default: "1"
  },
  trueColor: String,
  trueValue: {
    type: [String, Number, Boolean],
    default: true
  },
  beforeChange: Function,
  displayOnly: {
    type: Boolean,
    default: false
  },
  loading: {
    type: Boolean,
    default: false
  },
  width: [Number, String]
});
var Switch = defineComponent({
  name: $prefix + "Switch",
  props: switchProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
Switch.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Switch.install = function(Vue) {
  Vue.component(Switch.name, Switch);
};
Switch.version = version;
export {
  Switch as default
};
