// @opentiny/vue-config-provider/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
import { defineComponent, isVue2, hooks, provideDesignConfig, props, $prefix } from "@opentiny/vue-common";
import "@opentiny/vue-theme/config-provider/index.css";
import TinyThemeTool from "@opentiny/vue-theme/theme-tool";
import { isObject } from "@opentiny/utils";
var hasAnyKey = function hasAnyKey2(obj, keys) {
  if (obj == null) {
    return false;
  }
  if (keys.length === 0) {
    return false;
  }
  for (var _iterator = _createForOfIteratorHelperLoose(keys), _step; !(_step = _iterator()).done; ) {
    var key = _step.value;
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      return true;
    }
  }
  return false;
};
var _sfc_main = defineComponent({
  name: $prefix + "ConfigProvider",
  props: _extends({
    design: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    breakPoints: {
      type: Object,
      default: function _default2() {
        return {
          breakPoints: {
            "xs": 480,
            "sm": 640,
            "md": 768,
            "lg": 1024,
            "xl": 1280,
            "2xl": 1536
          }
        };
      }
    },
    direction: {
      type: String,
      default: "ltr"
    },
    tag: {
      type: Object,
      default: function _default3() {
        return {
          enable: true,
          name: "div"
        };
      }
    },
    theme: {
      type: Object,
      default: function _default4() {
        return null;
      }
    }
  }, props.map(function(item) {
    var _ref;
    return _ref = {}, _ref[item] = {
      type: String
    }, _ref;
  }).reduce(function(pre, cur) {
    return _extends({}, pre, cur);
  })),
  setup: function setup(props2, _ref2) {
    var slots = _ref2.slots;
    var _hooks$toRefs = hooks.toRefs(props2), direction = _hooks$toRefs.direction, design = _hooks$toRefs.design, theme = _hooks$toRefs.theme;
    provideDesignConfig(design);
    hooks.watch(function() {
      return theme.value;
    }, function() {
      if (isObject(theme.value) && hasAnyKey(theme.value, ["data"])) {
        var themeTool = new TinyThemeTool();
        themeTool.changeTheme(theme.value);
      }
      if (isObject(theme.value) && !hasAnyKey(theme.value, ["data"])) {
        console.warn("configProvider组件的theme属性对象请配置data属性。e.g { data: {'tv-base-color-brand': '#000'}}");
      }
      if (theme.value && !isObject(theme.value)) {
        console.warn("configProvider组件的theme属性请配置对象格式数据");
      }
    }, {
      immediate: true
    });
    var isRTL = hooks.computed(function() {
      return direction.value === "rtl";
    });
    var cssVar = hooks.computed(function() {
      return {
        "--text-direction": direction.value
      };
    });
    var classNames = hooks.reactive({
      "tiny-config-provider": true,
      "tiny-config-provider__rtl": isRTL
    });
    hooks.provide(configProviderContextKey, props2);
    return {
      slots,
      classNames,
      cssVar,
      props: props2,
      isVue2
    };
  },
  render: function render() {
    var _this$$slots, _this$slots, _this$props$tag$name;
    var attr = {
      class: this.classNames,
      style: this.cssVar
    };
    var slots = isVue2 ? (_this$$slots = this.$slots) == null ? void 0 : _this$$slots.default : (_this$slots = this.slots) == null ? void 0 : _this$slots.default == null ? void 0 : _this$slots.default();
    if (!this.props.tag.enable) {
      var _slots$length;
      if (isVue2 && ((_slots$length = slots == null ? void 0 : slots.length) != null ? _slots$length : 1) > 1) {
        return hooks.h("div", {}, slots);
      }
      return slots;
    }
    var tagName = (_this$props$tag$name = this.props.tag.name) != null ? _this$props$tag$name : "div";
    return hooks.h(tagName, attr, slots);
  }
});
var version = "3.undefined";
var configProviderContextKey = Symbol("CONFIG_PROVIDER_CONTEXT_KEY");
_sfc_main.install = function(Vue) {
  Vue.component(_sfc_main.name, _sfc_main);
};
_sfc_main.version = version;
export {
  configProviderContextKey,
  _sfc_main as default
};
