// @opentiny/vue-pull-refresh/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t2 = arguments[e];
      for (var r in t2) ({}).hasOwnProperty.call(t2, r) && (n[r] = t2[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import { t } from "@opentiny/vue-locale";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var $constants = {
  DEFAULT_HEAD_HEIGHT: 50,
  STATUS: {
    NORMAL: "normal",
    LOADING: "loading",
    LOOSING: "loosing",
    PULLING: "pulling",
    SUCCESS: "success"
  }
};
var PullRefresh = defineComponent({
  name: $prefix + "PullRefresh",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: Boolean,
    loosingText: {
      type: String,
      default: t("ui.pullRefresh.loosing")
    },
    successText: {
      type: String,
      default: $constants.STATUS.SUCCESS
    },
    failedText: String,
    successDuration: {
      type: [Number, String],
      default: 500
    },
    animationDuration: {
      type: [Number, String],
      default: 300
    },
    disabled: {
      type: Boolean,
      default: false
    },
    pullUp: {
      type: Object,
      default: {}
    },
    pullDown: {
      type: Object,
      default: {}
    },
    hasMore: {
      type: Boolean,
      default: true
    },
    // mobile-first的props
    headHeight: {
      type: [Number, String],
      default: $constants.DEFAULT_HEAD_HEIGHT
    },
    pullingText: {
      type: String,
      default: t("ui.pullRefresh.pulling")
    },
    loadingText: {
      type: String,
      default: $constants.STATUS.LOADING
    },
    pullDistance: [Number, String],
    loadingOptions: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    selfSimulate: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
PullRefresh.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
PullRefresh.install = function(Vue) {
  Vue.component(PullRefresh.name, PullRefresh);
};
PullRefresh.version = version;
export {
  PullRefresh as default
};
