// @opentiny/vue-button-group@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/button-group/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var buttonGroupProps = _extends({}, $props, {
  size: {
    type: String,
    default: "",
    validator: function validator(val) {
      return ["medium", "small", "mini", ""].includes(val);
    }
  },
  data: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  plain: Boolean,
  modelValue: [String, Number],
  disabled: Boolean,
  valueField: {
    type: String,
    default: "value"
  },
  textField: {
    type: String,
    default: "text"
  },
  showMore: Number,
  showEdit: {
    type: Boolean,
    default: false
  },
  border: {
    type: Boolean,
    default: true
  },
  displayMode: {
    type: String,
    default: "default",
    validator: function validator2(val) {
      return ["default", "merged"].includes(val);
    }
  }
});
var ButtonGroup = defineComponent({
  name: $prefix + "ButtonGroup",
  provide: function provide() {
    return {
      buttonGroup: this
    };
  },
  props: buttonGroupProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
ButtonGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ButtonGroup.install = function(Vue) {
  Vue.component(ButtonGroup.name, ButtonGroup);
};
ButtonGroup.version = version;
export {
  ButtonGroup as default
};
