// @opentiny/vue-divider/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/divider/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, normalizeStyle, normalizeClass, createCommentVNode, renderSlot } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "status-default": "border-color-border",
  "status-info": "border-color-brand",
  "status-warning": "border-color-warning",
  "status-success": "border-color-success",
  "status-error": "border-color-error"
};
var _sfc_main = defineComponent({
  props: [].concat(props, ["type", "vertical", "lineColor", "textPosition", "status", "margin", "offset", "fontSize", "height"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "aui-divider",
      style: normalizeStyle(_ctx.state.rootStyle),
      class: normalizeClass(_ctx.m("overflow-hidden relative", _ctx.vertical ? "inline-block w-px border-l" : "h-auto flex items-center justify-center w-full", _ctx.gcls("status-" + _ctx.status)))
    },
    [!_ctx.vertical ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: normalizeClass(["h-px w-full border-t", [_ctx.gcls("status-" + _ctx.status)]]),
        style: normalizeStyle(_ctx.state.lineStyle)
      },
      null,
      6
      /* CLASS, STYLE */
    )) : createCommentVNode("v-if", true), !_ctx.vertical ? (openBlock(), createElementBlock(
      "span",
      {
        key: 1,
        ref: "text",
        class: "absolute py-0 px-3.5 bg-white",
        style: normalizeStyle(_ctx.state.textStyle)
      },
      [renderSlot(_ctx.$slots, "default")],
      4
      /* STYLE */
    )) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
