// @opentiny/vue-renderless@3.30.0/statistic/vue.js
import {
  __spreadProps,
  __spreadValues
} from "../chunk-G2ADBYYC.js";
import { getIntegerAndDecimal, animateValue } from "./index";
const api = ["state", "getIntegerAndDecimal", "animateValue"];
const renderless = (props, hooks) => {
  const { reactive, computed, watch } = hooks;
  const state = reactive({
    value: computed(() => api2.getIntegerAndDecimal(props)),
    animatingValue: props.value,
    displayValue: computed(() => {
      if (props.useAnimation) {
        const tempProps = __spreadProps(__spreadValues({}, props), { value: state.animatingValue });
        return getIntegerAndDecimal({ props: tempProps })();
      }
      return state.value;
    })
  });
  const api2 = {
    state,
    getIntegerAndDecimal: getIntegerAndDecimal({ props }),
    animateValue: animateValue({ props, state })
  };
  watch(
    () => props.value,
    (newVal) => {
      if (props.useAnimation) {
        state.animatingValue = props.startValue;
        api2.animateValue();
      }
    },
    { immediate: true }
  );
  return api2;
};
export {
  api,
  renderless
};
