// @opentiny/vue-tree-select/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var TreeSelect = defineComponent({
  name: $prefix + "TreeSelect",
  componentName: "TreeSelect",
  props: _extends({}, $props, {
    clearable: Boolean,
    filterable: Boolean,
    filterMethod: Function,
    modelValue: {},
    multiple: Boolean,
    textField: {
      type: String,
      default: "label"
    },
    treeOp: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    valueField: {
      type: String,
      default: "value"
    },
    // 基础 props
    size: String,
    disabled: Boolean,
    placeholder: String,
    readonly: Boolean,
    // 弹框相关
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    placement: {
      type: String,
      default: "bottom-start"
    },
    dropStyle: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    // 标签相关
    collapseTags: Boolean,
    copyable: Boolean,
    textSplit: {
      type: String,
      default: ","
    },
    hoverExpand: Boolean,
    clickExpand: Boolean,
    showTips: {
      type: Boolean,
      default: true
    },
    // 搜索和过滤
    searchable: Boolean,
    // 多选相关
    multipleLimit: {
      type: Number,
      default: 0
    },
    // 下拉图标
    dropdownIcon: {
      type: [Object, String],
      default: ""
    },
    // 标签类型
    tagType: {
      type: String,
      default: ""
    },
    // 输入框类型
    inputBoxType: {
      type: String,
      default: "normal",
      validator: function validator(value) {
        return ["normal", "underline"].includes(value);
      }
    },
    // 多行默认最大显示行数
    maxVisibleRows: {
      type: Number,
      default: 1
    },
    // 原生属性
    name: String,
    autocomplete: {
      type: String,
      default: "off"
    },
    // 懒加载相关
    lazy: {
      type: Boolean,
      default: false
    },
    load: Function,
    afterLoad: Function
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
TreeSelect.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
TreeSelect.install = function(Vue) {
  Vue.component(TreeSelect.name, TreeSelect);
};
TreeSelect.version = version;
export {
  TreeSelect as default
};
