// @opentiny/vue-virtual-scroll-box@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var VirtualScrollBox = defineComponent({
  name: $prefix + "VirtualScrollBox",
  props: _extends({}, $props, {
    width: [Number, String],
    // 区域的宽度
    height: [Number, String],
    // 区域的高度
    rowBuffer: {
      type: [Number, String],
      default: 200
    },
    // 行缓冲区高度
    columnBuffer: {
      type: [Number, String],
      default: 200
    },
    // 列缓冲区宽度
    scrollbarSize: [Number, String],
    // 滚动条尺寸
    columns: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    // 所有的列
    columnSizes: {
      type: Array,
      default: function _default2() {
        return [];
      }
    },
    // 每个列对应的宽度
    rows: {
      type: Array,
      default: function _default3() {
        return [];
      }
    },
    // 所有的行
    rowSizes: {
      type: Array,
      default: function _default4() {
        return [];
      }
    },
    // 每个行对应的高度
    fixedRows: {
      type: Array,
      default: function _default5() {
        return [];
      }
    },
    // 行固定配置
    fixedColumns: {
      type: Array,
      default: function _default6() {
        return [];
      }
    },
    // 列固定配置
    spanConfig: {
      type: Array,
      default: function _default7() {
        return [];
      }
    }
    // 单元格合并配置
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
VirtualScrollBox.install = function(Vue) {
  Vue.component(VirtualScrollBox.name, VirtualScrollBox);
};
VirtualScrollBox.version = version;
export {
  VirtualScrollBox as default
};
