// @opentiny/vue-color-picker/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/color-picker/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {};
var ColorPicker = defineComponent({
  name: $prefix + "ColorPicker",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: String,
    visible: Boolean,
    alpha: Boolean,
    predefine: Array,
    history: Array,
    size: {
      type: String,
      default: "",
      validator: function validator(val) {
        return ["large", "medium", "small", "mini", ""].includes(val);
      }
    },
    format: {
      type: Array,
      default: function _default2() {
        return [];
      },
      validator: function validator2(val) {
        if (!val.length) {
          return true;
        }
        if (val[val.length - 1] === "a") {
          console.warn("If you want enable alpha, You should set `alpha` prop to true");
        }
        return ["hsv", "hsl", "rgb", "hex"].includes(val[0]);
      }
    },
    enableHistory: {
      type: Boolean,
      default: false
    },
    enablePredefineColor: {
      type: Boolean,
      default: false
    },
    colorMode: {
      type: String,
      default: "monochrome",
      validator: function validator3(val) {
        return ["monochrome", "linear-gradient"].includes(val);
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
ColorPicker.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ColorPicker.install = function(Vue) {
  Vue.component(ColorPicker.name, ColorPicker);
};
ColorPicker.version = version;
export {
  ColorPicker as default
};
