// @opentiny/vue-tree-select@3.30.0/lib/mobile-first.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/tree-select/vue';
import Tree from '@opentiny/vue-tree';
import BaseSelect from '@opentiny/vue-base-select';
import { resolveComponent, openBlock, createBlock, withCtx, createVNode, mergeProps } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "TreeSelect",
  components: {
    TinyTree: Tree,
    TinyBaseSelect: BaseSelect
  },
  props: _extends({}, props, {
    clearable: Boolean,
    filterable: Boolean,
    filterMethod: Function,
    modelValue: {},
    multiple: Boolean,
    textField: {
      type: String,
      default: "label"
    },
    treeOp: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    valueField: {
      type: String,
      default: "value"
    },
    // 基础 props
    size: String,
    disabled: Boolean,
    placeholder: String,
    readonly: Boolean,
    // 弹框相关
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    placement: {
      type: String,
      default: "bottom-start"
    },
    dropStyle: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    // 标签相关
    collapseTags: Boolean,
    copyable: Boolean,
    textSplit: {
      type: String,
      default: ","
    },
    hoverExpand: Boolean,
    clickExpand: Boolean,
    showTips: {
      type: Boolean,
      default: true
    },
    // 搜索和过滤
    searchable: Boolean,
    // 多选相关
    multipleLimit: {
      type: Number,
      default: 0
    },
    // 下拉图标
    dropdownIcon: {
      type: [Object, String],
      default: ""
    },
    // 标签类型
    tagType: {
      type: String,
      default: ""
    },
    // 输入框类型
    inputBoxType: {
      type: String,
      default: "normal",
      validator: function validator(value) {
        return ["normal", "underline"].includes(value);
      }
    },
    // 多行默认最大显示行数
    maxVisibleRows: {
      type: Number,
      default: 1
    },
    // 原生属性
    name: String,
    autocomplete: {
      type: String,
      default: "off"
    },
    // 懒加载相关
    lazy: {
      type: Boolean,
      default: false
    },
    load: Function,
    afterLoad: Function
  }),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_tree = resolveComponent("tiny-tree");
  var _component_tiny_base_select = resolveComponent("tiny-base-select");
  return openBlock(), createBlock(_component_tiny_base_select, {
    ref: "baseSelectRef",
    "data-tag": "tiny-tree-select",
    modelValue: _ctx.state.modelValue,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.modelValue = $event;
    }),
    clearable: _ctx.clearable,
    filterable: _ctx.filterable,
    "filter-method": _ctx.filter,
    multiple: _ctx.multiple,
    "text-field": _ctx.textField,
    "value-field": _ctx.valueField,
    size: _ctx.size,
    disabled: _ctx.disabled,
    placeholder: _ctx.placeholder,
    readonly: _ctx.readonly,
    "popper-class": _ctx.popperClass,
    "popper-append-to-body": _ctx.popperAppendToBody,
    placement: _ctx.placement,
    "drop-style": _ctx.dropStyle,
    "hover-expand": _ctx.hoverExpand,
    "click-expand": _ctx.clickExpand,
    "collapse-tags": _ctx.collapseTags,
    copyable: _ctx.copyable,
    "text-split": _ctx.textSplit,
    "show-tips": _ctx.showTips,
    searchable: _ctx.searchable,
    "multiple-limit": _ctx.multipleLimit,
    "dropdown-icon": _ctx.dropdownIcon,
    "tag-type": _ctx.tagType,
    "input-box-type": _ctx.inputBoxType,
    "max-visible-rows": _ctx.maxVisibleRows,
    name: _ctx.name,
    autocomplete: _ctx.autocomplete
  }, {
    panel: withCtx(function() {
      return [createVNode(_component_tiny_tree, mergeProps({
        ref: "treeRef",
        "check-strictly": _ctx.treeOp.checkStrictly,
        "current-node-key": !_ctx.multiple ? _ctx.state.currentKey : "",
        data: _ctx.state.treeData,
        "default-checked-keys": _ctx.multiple ? _ctx.state.defaultCheckedKeys : _ctx.treeOp.defaultCheckedKeys || [],
        "default-expand-all": _ctx.treeOp.defaultExpandAll !== void 0 ? _ctx.treeOp.defaultExpandAll : (_ctx.lazy !== void 0 ? _ctx.lazy : _ctx.treeOp.lazy) ? false : true,
        "expand-on-click-node": false,
        "filter-node-method": _ctx.filterMethod,
        "icon-trigger-click-node": false,
        "node-key": _ctx.valueField,
        props: {
          label: _ctx.textField
        },
        "show-checkbox": _ctx.multiple,
        lazy: _ctx.lazy !== void 0 ? _ctx.lazy : _ctx.treeOp.lazy,
        load: _ctx.load || _ctx.treeOp.load,
        "after-load": _ctx.afterLoad || _ctx.treeOp.afterLoad,
        onCheck: _ctx.check,
        onNodeClick: _ctx.nodeClick
      }, _ctx.treeOp), null, 16, ["check-strictly", "current-node-key", "data", "default-checked-keys", "default-expand-all", "filter-node-method", "node-key", "props", "show-checkbox", "lazy", "load", "after-load", "onCheck", "onNodeClick"])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["modelValue", "clearable", "filterable", "filter-method", "multiple", "text-field", "value-field", "size", "disabled", "placeholder", "readonly", "popper-class", "popper-append-to-body", "placement", "drop-style", "hover-expand", "click-expand", "collapse-tags", "copyable", "text-split", "show-tips", "searchable", "multiple-limit", "dropdown-icon", "tag-type", "input-box-type", "max-visible-rows", "name", "autocomplete"]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
