// @opentiny/vue-space@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/space/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var spaceProps = _extends({}, $props, {
  /** 设置空间尺寸，可为字符串、数字或 [水平, 垂直] 数组 */
  size: {
    type: [String, Number, Array],
    default: "small"
  },
  /** 主轴方向，row 或 column */
  direction: {
    type: String,
    default: "row"
  },
  /** 对齐方式，start/center/end */
  align: {
    type: String,
    default: "start"
  },
  /** 主轴上的排列方式 */
  justify: {
    type: String,
    default: "start"
  },
  /** 是否自动换行 */
  wrap: {
    type: Boolean,
    default: false
  },
  /** 是否显示边框 */
  border: {
    type: Boolean,
    default: true
  },
  /** 自定义类名（仅 mobile-first 模板有效） */
  customClass: {
    type: String,
    default: ""
  },
  order: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  /** 自定义样式 */
  customStyle: {
    type: Object,
    default: function _default2() {
      return {};
    }
  },
  /** 提供内部使用的常量（保留扩展能力） */
  _constants: {
    type: Object,
    default: function _default3() {
      return {};
    }
  }
});
var Space = defineComponent({
  name: $prefix + "Space",
  props: spaceProps,
  slots: Object,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
Space.install = function(Vue) {
  Vue.component(Space.name, Space);
};
Space.version = version;
export {
  Space as default
};
