// @opentiny/vue-space@3.30.0/lib/pc.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, props, h, setup as _setup } from "@opentiny/vue-common";
import { api, renderless } from "@opentiny/vue-renderless/space/vue";
var _sfc_main = defineComponent({
  name: "TinySpace",
  props: [].concat(props, ["size", "direction", "align", "justify", "wrap", "order", "customClass", "customStyle"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  },
  render: function render() {
    var _this$orderedChildren, _this$$slots$default, _this$$slots, _this$state;
    var hFunc = this.$createElement || h;
    var children = (_this$orderedChildren = this.orderedChildren) != null && _this$orderedChildren.length ? this.orderedChildren : ((_this$$slots$default = (_this$$slots = this.$slots).default) == null ? void 0 : _this$$slots$default.call(_this$$slots)) || [];
    return hFunc("div", {
      class: this.customClass,
      style: [(_this$state = this.state) == null ? void 0 : _this$state.gapStyle, _extends({
        display: "flex",
        flexDirection: this.direction || "row",
        alignItems: this.align || "flex-start",
        justifyContent: this.justify || "flex-start",
        flexWrap: this.wrap ? "wrap" : "nowrap"
      }, this.customStyle)]
    }, children);
  }
});
export {
  _sfc_main as default
};
