// @opentiny/vue-exception@3.30.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/exception/vue';
import { defineComponent, setup, $props } from '@opentiny/vue-common';
import Button from '@opentiny/vue-button';
import '@opentiny/vue-theme/exception/index.css';
import { iconNoData, iconNoNews, iconNoPerm, iconNoResult, iconWeaknet, iconPageNoperm, iconPageNothing, iconPageServererror, iconPageWeaknet } from '@opentiny/vue-icon';
import { openBlock, createElementBlock, normalizeClass, createElementVNode, renderSlot, createBlock, resolveDynamicComponent, createCommentVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var $constants = {
  INTERNALCONFIG: {
    NODATA: "ui.exception.nodata",
    NOPERM: "ui.exception.noperm",
    WEAKNET: "ui.exception.weaknet",
    NORESULT: "ui.exception.noresult",
    NONEWS: "ui.exception.nonews",
    PAGENOPERM: "ui.exception.pagenoperm",
    PAGEWEAKNET: "ui.exception.pageweaknet",
    PAGENOTHING: "ui.exception.pagenothing",
    PAGESERVERERROR: "ui.exception.pageservererror"
  },
  ICONCONFIG: {
    NODATA: "icon-no-data",
    NOPERM: "icon-no-perm",
    NONEWS: "icon-no-news",
    WEAKNET: "icon-weaknet",
    NORESULT: "icon-no-result",
    PAGENOPERM: "icon-page-noperm",
    PAGENOTHING: "icon-page-nothing",
    PAGESERVERERROR: "icon-page-servererror",
    PAGEWEAKNET: "icon-page-weaknet"
  }
};
var _sfc_main = defineComponent({
  components: {
    TinyButton: Button,
    IconNoData: iconNoData(),
    IconNoNews: iconNoNews(),
    IconNoPerm: iconNoPerm(),
    IconNoResult: iconNoResult(),
    IconWeaknet: iconWeaknet(),
    IconPageNoperm: iconPageNoperm(),
    IconPageNothing: iconPageNothing(),
    IconPageServererror: iconPageServererror(),
    IconPageWeaknet: iconPageWeaknet()
  },
  emits: ["click"],
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    type: {
      type: String,
      default: "nodata"
    },
    subMessage: String,
    exceptionClass: String,
    pageEmpty: {
      type: Boolean,
      default: false
    },
    componentPage: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-exception-body"
};
var _hoisted_2 = {
  class: "tiny-exception__description"
};
var _hoisted_3 = {
  key: 0,
  class: "tiny-exception__submessage"
};
var _hoisted_4 = {
  class: "tiny-exception__footer"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-exception", _ctx.exceptionClass])
    },
    [createElementVNode("div", _hoisted_1, [createElementVNode(
      "div",
      {
        class: normalizeClass(["tiny-exception__image bg-cover", {
          "tiny-exception__componentpage-image": _ctx.componentPage
        }])
      },
      [renderSlot(_ctx.$slots, "icon", {}, function() {
        return [_ctx.state.urlType ? (openBlock(), createBlock(resolveDynamicComponent(_ctx._constants.ICONCONFIG[_ctx.type.toUpperCase()]), {
          key: 0
        })) : createCommentVNode("v-if", true)];
      })],
      2
      /* CLASS */
    ), renderSlot(_ctx.$slots, "content", {}, function() {
      return [createElementVNode("div", _hoisted_2, [createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-exception__message", {
            "tiny-exception__componentpage": _ctx.componentPage
          }])
        },
        toDisplayString(_ctx.t(_ctx._constants.INTERNALCONFIG[_ctx.type.toUpperCase()])),
        3
        /* TEXT, CLASS */
      ), _ctx.subMessage ? (openBlock(), createElementBlock(
        "div",
        _hoisted_3,
        toDisplayString(_ctx.subMessage),
        1
        /* TEXT */
      )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_4, [renderSlot(_ctx.$slots, "default")])])];
    })])],
    2
    /* CLASS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
