// @opentiny/vue@3.30.0/index.js
import ActionMenu from "@opentiny/vue-action-menu";
import { default as default2, default as default3 } from "@opentiny/vue-action-menu";
import ActionSheet from "@opentiny/vue-action-sheet";
import { default as default4, default as default5 } from "@opentiny/vue-action-sheet";
import Alert from "@opentiny/vue-alert";
import { default as default6, default as default7 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default8, default as default9 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default10, default as default11 } from "@opentiny/vue-anchor";
import Area from "@opentiny/vue-area";
import { default as default12, default as default13 } from "@opentiny/vue-area";
import AsyncFlowchart from "@opentiny/vue-async-flowchart";
import { default as default14, default as default15 } from "@opentiny/vue-async-flowchart";
import Autocomplete from "@opentiny/vue-autocomplete";
import { default as default16, default as default17 } from "@opentiny/vue-autocomplete";
import Badge from "@opentiny/vue-badge";
import { default as default18, default as default19 } from "@opentiny/vue-badge";
import BaseSelect from "@opentiny/vue-base-select";
import { default as default20, default as default21 } from "@opentiny/vue-base-select";
import Breadcrumb from "@opentiny/vue-breadcrumb";
import { default as default22, default as default23 } from "@opentiny/vue-breadcrumb";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { default as default24, default as default25 } from "@opentiny/vue-breadcrumb-item";
import BulletinBoard from "@opentiny/vue-bulletin-board";
import { default as default26, default as default27 } from "@opentiny/vue-bulletin-board";
import Button from "@opentiny/vue-button";
import { default as default28, default as default29 } from "@opentiny/vue-button";
import ButtonGroup from "@opentiny/vue-button-group";
import { default as default30, default as default31 } from "@opentiny/vue-button-group";
import Calendar from "@opentiny/vue-calendar";
import { default as default32, default as default33 } from "@opentiny/vue-calendar";
import CalendarBar from "@opentiny/vue-calendar-bar";
import { default as default34, default as default35 } from "@opentiny/vue-calendar-bar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default36, default as default37 } from "@opentiny/vue-calendar-view";
import Card from "@opentiny/vue-card";
import { default as default38, default as default39 } from "@opentiny/vue-card";
import CardGroup from "@opentiny/vue-card-group";
import { default as default40, default as default41 } from "@opentiny/vue-card-group";
import CardTemplate from "@opentiny/vue-card-template";
import { default as default42, default as default43 } from "@opentiny/vue-card-template";
import Carousel from "@opentiny/vue-carousel";
import { default as default44, default as default45 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default46, default as default47 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default48, default as default49 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default50, default as default51 } from "@opentiny/vue-cascader-menu";
import CascaderMobile from "@opentiny/vue-cascader-mobile";
import { default as default52, default as default53 } from "@opentiny/vue-cascader-mobile";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default54, default as default55 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default56, default as default57 } from "@opentiny/vue-cascader-panel";
import CascaderSelect from "@opentiny/vue-cascader-select";
import { default as default58, default as default59 } from "@opentiny/vue-cascader-select";
import CascaderView from "@opentiny/vue-cascader-view";
import { default as default60, default as default61 } from "@opentiny/vue-cascader-view";
import Cell from "@opentiny/vue-cell";
import { default as default62, default as default63 } from "@opentiny/vue-cell";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default64, default as default65 } from "@opentiny/vue-checkbox";
import CheckboxButton from "@opentiny/vue-checkbox-button";
import { default as default66, default as default67 } from "@opentiny/vue-checkbox-button";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default68, default as default69 } from "@opentiny/vue-checkbox-group";
import Col from "@opentiny/vue-col";
import { default as default70, default as default71 } from "@opentiny/vue-col";
import Collapse from "@opentiny/vue-collapse";
import { default as default72, default as default73 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default74, default as default75 } from "@opentiny/vue-collapse-item";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { default as default76, default as default77 } from "@opentiny/vue-collapse-transition";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default78, default as default79 } from "@opentiny/vue-color-picker";
import ColorSelectPanel from "@opentiny/vue-color-select-panel";
import { default as default80, default as default81 } from "@opentiny/vue-color-select-panel";
import ColumnListGroup from "@opentiny/vue-column-list-group";
import { default as default82, default as default83 } from "@opentiny/vue-column-list-group";
import ColumnListItem from "@opentiny/vue-column-list-item";
import { default as default84, default as default85 } from "@opentiny/vue-column-list-item";
import Company from "@opentiny/vue-company";
import { default as default86, default as default87 } from "@opentiny/vue-company";
import ConfigProvider from "@opentiny/vue-config-provider";
import { default as default88, default as default89 } from "@opentiny/vue-config-provider";
import Container from "@opentiny/vue-container";
import { default as default90, default as default91 } from "@opentiny/vue-container";
import Country from "@opentiny/vue-country";
import { default as default92, default as default93 } from "@opentiny/vue-country";
import Crop from "@opentiny/vue-crop";
import { default as default94, default as default95 } from "@opentiny/vue-crop";
import Currency from "@opentiny/vue-currency";
import { default as default96, default as default97 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default98, default as default99 } from "@opentiny/vue-date-panel";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default100, default as default101 } from "@opentiny/vue-date-picker";
import DatePickerMobileFirst from "@opentiny/vue-date-picker-mobile-first";
import { default as default102, default as default103 } from "@opentiny/vue-date-picker-mobile-first";
import DateRange from "@opentiny/vue-date-range";
import { default as default104, default as default105 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default106, default as default107 } from "@opentiny/vue-date-table";
import Dept from "@opentiny/vue-dept";
import { default as default108, default as default109 } from "@opentiny/vue-dept";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default110, default as default111 } from "@opentiny/vue-dialog-box";
import DialogSelect from "@opentiny/vue-dialog-select";
import { default as default112, default as default113 } from "@opentiny/vue-dialog-select";
import Divider from "@opentiny/vue-divider";
import { default as default114, default as default115 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default116, default as default117 } from "@opentiny/vue-drawer";
import DropRoles from "@opentiny/vue-drop-roles";
import { default as default118, default as default119 } from "@opentiny/vue-drop-roles";
import DropTimes from "@opentiny/vue-drop-times";
import { default as default120, default as default121 } from "@opentiny/vue-drop-times";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default122, default as default123 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default124, default as default125 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default126, default as default127 } from "@opentiny/vue-dropdown-menu";
import DynamicScroller from "@opentiny/vue-dynamic-scroller";
import { default as default128, default as default129 } from "@opentiny/vue-dynamic-scroller";
import DynamicScrollerItem from "@opentiny/vue-dynamic-scroller-item";
import { default as default130, default as default131 } from "@opentiny/vue-dynamic-scroller-item";
import Espace from "@opentiny/vue-espace";
import { default as default132, default as default133 } from "@opentiny/vue-espace";
import Exception from "@opentiny/vue-exception";
import { default as default134, default as default135 } from "@opentiny/vue-exception";
import FallMenu from "@opentiny/vue-fall-menu";
import { default as default136, default as default137 } from "@opentiny/vue-fall-menu";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default138, default as default139 } from "@opentiny/vue-file-upload";
import Filter from "@opentiny/vue-filter";
import { default as default140, default as default141 } from "@opentiny/vue-filter";
import FilterBar from "@opentiny/vue-filter-bar";
import { default as default142, default as default143 } from "@opentiny/vue-filter-bar";
import FilterBox from "@opentiny/vue-filter-box";
import { default as default144, default as default145 } from "@opentiny/vue-filter-box";
import FilterPanel from "@opentiny/vue-filter-panel";
import { default as default146, default as default147 } from "@opentiny/vue-filter-panel";
import FloatButton from "@opentiny/vue-float-button";
import { default as default148, default as default149 } from "@opentiny/vue-float-button";
import Floatbar from "@opentiny/vue-floatbar";
import { default as default150, default as default151 } from "@opentiny/vue-floatbar";
import FloatingButton from "@opentiny/vue-floating-button";
import { default as default152, default as default153 } from "@opentiny/vue-floating-button";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default154, default as default155 } from "@opentiny/vue-flowchart";
import FluentEditor from "@opentiny/vue-fluent-editor";
import { default as default156, default as default157 } from "@opentiny/vue-fluent-editor";
import Form from "@opentiny/vue-form";
import { default as default158, default as default159 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default160, default as default161 } from "@opentiny/vue-form-item";
import Fullscreen from "@opentiny/vue-fullscreen";
import { default as default162, default as default163 } from "@opentiny/vue-fullscreen";
import Grid from "@opentiny/vue-grid";
import { default as default164, default as default165 } from "@opentiny/vue-grid";
import GridColumn from "@opentiny/vue-grid-column";
import { default as default166, default as default167 } from "@opentiny/vue-grid-column";
import GridManager from "@opentiny/vue-grid-manager";
import { default as default168, default as default169 } from "@opentiny/vue-grid-manager";
import GridSelect from "@opentiny/vue-grid-select";
import { default as default170, default as default171 } from "@opentiny/vue-grid-select";
import GridToolbar from "@opentiny/vue-grid-toolbar";
import { default as default172, default as default173 } from "@opentiny/vue-grid-toolbar";
import Guide from "@opentiny/vue-guide";
import { default as default174, default as default175 } from "@opentiny/vue-guide";
import Hrapprover from "@opentiny/vue-hrapprover";
import { default as default176, default as default177 } from "@opentiny/vue-hrapprover";
import Image from "@opentiny/vue-image";
import { default as default178, default as default179 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default180, default as default181 } from "@opentiny/vue-image-viewer";
import Input from "@opentiny/vue-input";
import { default as default182, default as default183 } from "@opentiny/vue-input";
import IpAddress from "@opentiny/vue-ip-address";
import { default as default184, default as default185 } from "@opentiny/vue-ip-address";
import Layout from "@opentiny/vue-layout";
import { default as default186, default as default187 } from "@opentiny/vue-layout";
import Link from "@opentiny/vue-link";
import { default as default188, default as default189 } from "@opentiny/vue-link";
import LinkMenu from "@opentiny/vue-link-menu";
import { default as default190, default as default191 } from "@opentiny/vue-link-menu";
import LoadList from "@opentiny/vue-load-list";
import { default as default192, default as default193 } from "@opentiny/vue-load-list";
import Loading from "@opentiny/vue-loading";
import { default as default194, default as default195 } from "@opentiny/vue-loading";
import Locales from "@opentiny/vue-locales";
import { default as default196, default as default197 } from "@opentiny/vue-locales";
import LogonUser from "@opentiny/vue-logon-user";
import { default as default198, default as default199 } from "@opentiny/vue-logon-user";
import Logout from "@opentiny/vue-logout";
import { default as default200, default as default201 } from "@opentiny/vue-logout";
import Menu from "@opentiny/vue-menu";
import { default as default202, default as default203 } from "@opentiny/vue-menu";
import Message from "@opentiny/vue-message";
import { default as default204, default as default205 } from "@opentiny/vue-message";
import Milestone from "@opentiny/vue-milestone";
import { default as default206, default as default207 } from "@opentiny/vue-milestone";
import MindMap from "@opentiny/vue-mind-map";
import { default as default208, default as default209 } from "@opentiny/vue-mind-map";
import Modal from "@opentiny/vue-modal";
import { default as default210, default as default211 } from "@opentiny/vue-modal";
import MonthRange from "@opentiny/vue-month-range";
import { default as default212, default as default213 } from "@opentiny/vue-month-range";
import MonthTable from "@opentiny/vue-month-table";
import { default as default214, default as default215 } from "@opentiny/vue-month-table";
import NavMenu from "@opentiny/vue-nav-menu";
import { default as default216, default as default217 } from "@opentiny/vue-nav-menu";
import Notify from "@opentiny/vue-notify";
import { default as default218, default as default219 } from "@opentiny/vue-notify";
import NumberAnimation from "@opentiny/vue-number-animation";
import { default as default220, default as default221 } from "@opentiny/vue-number-animation";
import Numeric from "@opentiny/vue-numeric";
import { default as default222, default as default223 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default224, default as default225 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default226, default as default227 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default228, default as default229 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default230, default as default231 } from "@opentiny/vue-pager-item";
import Panel from "@opentiny/vue-panel";
import { default as default232, default as default233 } from "@opentiny/vue-panel";
import Picker from "@opentiny/vue-picker";
import { default as default234, default as default235 } from "@opentiny/vue-picker";
import PopUpload from "@opentiny/vue-pop-upload";
import { default as default236, default as default237 } from "@opentiny/vue-pop-upload";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default238, default as default239 } from "@opentiny/vue-popconfirm";
import Popeditor from "@opentiny/vue-popeditor";
import { default as default240, default as default241 } from "@opentiny/vue-popeditor";
import Popover from "@opentiny/vue-popover";
import { default as default242, default as default243 } from "@opentiny/vue-popover";
import Popup from "@opentiny/vue-popup";
import { default as default244, default as default245 } from "@opentiny/vue-popup";
import Progress from "@opentiny/vue-progress";
import { default as default246, default as default247 } from "@opentiny/vue-progress";
import PullRefresh from "@opentiny/vue-pull-refresh";
import { default as default248, default as default249 } from "@opentiny/vue-pull-refresh";
import QrCode from "@opentiny/vue-qr-code";
import { default as default250, default as default251 } from "@opentiny/vue-qr-code";
import QuarterPanel from "@opentiny/vue-quarter-panel";
import { default as default252, default as default253 } from "@opentiny/vue-quarter-panel";
import QueryBuilder from "@opentiny/vue-query-builder";
import { default as default254, default as default255 } from "@opentiny/vue-query-builder";
import Radio from "@opentiny/vue-radio";
import { default as default256, default as default257 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default258, default as default259 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default260, default as default261 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default262, default as default263 } from "@opentiny/vue-rate";
import Record from "@opentiny/vue-record";
import { default as default264, default as default265 } from "@opentiny/vue-record";
import RecycleScroller from "@opentiny/vue-recycle-scroller";
import { default as default266, default as default267 } from "@opentiny/vue-recycle-scroller";
import River from "@opentiny/vue-river";
import { default as default268, default as default269 } from "@opentiny/vue-river";
import Roles from "@opentiny/vue-roles";
import { default as default270, default as default271 } from "@opentiny/vue-roles";
import Row from "@opentiny/vue-row";
import { default as default272, default as default273 } from "@opentiny/vue-row";
import ScrollText from "@opentiny/vue-scroll-text";
import { default as default274, default as default275 } from "@opentiny/vue-scroll-text";
import Scrollbar from "@opentiny/vue-scrollbar";
import { default as default276, default as default277 } from "@opentiny/vue-scrollbar";
import Search from "@opentiny/vue-search";
import { default as default278, default as default279 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default280, default as default281 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default282, default as default283 } from "@opentiny/vue-select-dropdown";
import SelectMobile from "@opentiny/vue-select-mobile";
import { default as default284, default as default285 } from "@opentiny/vue-select-mobile";
import SelectView from "@opentiny/vue-select-view";
import { default as default286, default as default287 } from "@opentiny/vue-select-view";
import SelectWrapper from "@opentiny/vue-select-wrapper";
import { default as default288, default as default289 } from "@opentiny/vue-select-wrapper";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default290, default as default291 } from "@opentiny/vue-selected-box";
import Signature from "@opentiny/vue-signature";
import { default as default292, default as default293 } from "@opentiny/vue-signature";
import Skeleton from "@opentiny/vue-skeleton";
import { default as default294, default as default295 } from "@opentiny/vue-skeleton";
import SkeletonItem from "@opentiny/vue-skeleton-item";
import { default as default296, default as default297 } from "@opentiny/vue-skeleton-item";
import Slider from "@opentiny/vue-slider";
import { default as default298, default as default299 } from "@opentiny/vue-slider";
import SliderButton from "@opentiny/vue-slider-button";
import { default as default300, default as default301 } from "@opentiny/vue-slider-button";
import SliderButtonGroup from "@opentiny/vue-slider-button-group";
import { default as default302, default as default303 } from "@opentiny/vue-slider-button-group";
import Space from "@opentiny/vue-space";
import { default as default304, default as default305 } from "@opentiny/vue-space";
import Split from "@opentiny/vue-split";
import { default as default306, default as default307 } from "@opentiny/vue-split";
import StandardListItem from "@opentiny/vue-standard-list-item";
import { default as default308, default as default309 } from "@opentiny/vue-standard-list-item";
import Statistic from "@opentiny/vue-statistic";
import { default as default310, default as default311 } from "@opentiny/vue-statistic";
import Steps from "@opentiny/vue-steps";
import { default as default312, default as default313 } from "@opentiny/vue-steps";
import Sticky from "@opentiny/vue-sticky";
import { default as default314, default as default315 } from "@opentiny/vue-sticky";
import Switch from "@opentiny/vue-switch";
import { default as default316, default as default317 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default318, default as default319 } from "@opentiny/vue-tab-item";
import Tabbar from "@opentiny/vue-tabbar";
import { default as default320, default as default321 } from "@opentiny/vue-tabbar";
import TabbarItem from "@opentiny/vue-tabbar-item";
import { default as default322, default as default323 } from "@opentiny/vue-tabbar-item";
import Table from "@opentiny/vue-table";
import { default as default324, default as default325 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default326, default as default327 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default328, default as default329 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default330, default as default331 } from "@opentiny/vue-tag-group";
import TagInput from "@opentiny/vue-tag-input";
import { default as default332, default as default333 } from "@opentiny/vue-tag-input";
import TextPopup from "@opentiny/vue-text-popup";
import { default as default334, default as default335 } from "@opentiny/vue-text-popup";
import Time from "@opentiny/vue-time";
import { default as default336, default as default337 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default338, default as default339 } from "@opentiny/vue-time-line";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default340, default as default341 } from "@opentiny/vue-time-panel";
import TimePicker from "@opentiny/vue-time-picker";
import { default as default342, default as default343 } from "@opentiny/vue-time-picker";
import TimePickerMobile from "@opentiny/vue-time-picker-mobile";
import { default as default344, default as default345 } from "@opentiny/vue-time-picker-mobile";
import TimeRange from "@opentiny/vue-time-range";
import { default as default346, default as default347 } from "@opentiny/vue-time-range";
import TimeSelect from "@opentiny/vue-time-select";
import { default as default348, default as default349 } from "@opentiny/vue-time-select";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default350, default as default351 } from "@opentiny/vue-time-spinner";
import TimelineItem from "@opentiny/vue-timeline-item";
import { default as default352, default as default353 } from "@opentiny/vue-timeline-item";
import ToggleMenu from "@opentiny/vue-toggle-menu";
import { default as default354, default as default355 } from "@opentiny/vue-toggle-menu";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default356, default as default357 } from "@opentiny/vue-tooltip";
import TopBox from "@opentiny/vue-top-box";
import { default as default358, default as default359 } from "@opentiny/vue-top-box";
import Transfer from "@opentiny/vue-transfer";
import { default as default360, default as default361 } from "@opentiny/vue-transfer";
import TransferPanel from "@opentiny/vue-transfer-panel";
import { default as default362, default as default363 } from "@opentiny/vue-transfer-panel";
import Tree from "@opentiny/vue-tree";
import { default as default364, default as default365 } from "@opentiny/vue-tree";
import TreeMenu from "@opentiny/vue-tree-menu";
import { default as default366, default as default367 } from "@opentiny/vue-tree-menu";
import TreeSelect from "@opentiny/vue-tree-select";
import { default as default368, default as default369 } from "@opentiny/vue-tree-select";
import Upload from "@opentiny/vue-upload";
import { default as default370, default as default371 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default372, default as default373 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default374, default as default375 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default376, default as default377 } from "@opentiny/vue-user";
import UserAccount from "@opentiny/vue-user-account";
import { default as default378, default as default379 } from "@opentiny/vue-user-account";
import UserContact from "@opentiny/vue-user-contact";
import { default as default380, default as default381 } from "@opentiny/vue-user-contact";
import UserHead from "@opentiny/vue-user-head";
import { default as default382, default as default383 } from "@opentiny/vue-user-head";
import UserHeadGroup from "@opentiny/vue-user-head-group";
import { default as default384, default as default385 } from "@opentiny/vue-user-head-group";
import UserLink from "@opentiny/vue-user-link";
import { default as default386, default as default387 } from "@opentiny/vue-user-link";
import VirtualScrollBox from "@opentiny/vue-virtual-scroll-box";
import { default as default388, default as default389 } from "@opentiny/vue-virtual-scroll-box";
import VirtualTree from "@opentiny/vue-virtual-tree";
import { default as default390, default as default391 } from "@opentiny/vue-virtual-tree";
import Watermark from "@opentiny/vue-watermark";
import { default as default392, default as default393 } from "@opentiny/vue-watermark";
import Wizard from "@opentiny/vue-wizard";
import { default as default394, default as default395 } from "@opentiny/vue-wizard";
import YearRange from "@opentiny/vue-year-range";
import { default as default396, default as default397 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default398, default as default399 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionMenu, ActionSheet, Alert, Amount, Anchor, Area, AsyncFlowchart, Autocomplete, Badge, BaseSelect, Breadcrumb, BreadcrumbItem, BulletinBoard, Button, ButtonGroup, Calendar, CalendarBar, CalendarView, Card, CardGroup, CardTemplate, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderMobile, CascaderNode, CascaderPanel, CascaderSelect, CascaderView, Cell, Checkbox, CheckboxButton, CheckboxGroup, Col, Collapse, CollapseItem, CollapseTransition, ColorPicker, ColorSelectPanel, ColumnListGroup, ColumnListItem, Company, ConfigProvider, Container, Country, Crop, Currency, DatePanel, DatePicker, DatePickerMobileFirst, DateRange, DateTable, Dept, DialogBox, DialogSelect, Divider, Drawer, DropRoles, DropTimes, Dropdown, DropdownItem, DropdownMenu, DynamicScroller, DynamicScrollerItem, Espace, Exception, FallMenu, FileUpload, Filter, FilterBar, FilterBox, FilterPanel, FloatButton, Floatbar, FloatingButton, Flowchart, FluentEditor, Form, FormItem, Fullscreen, Grid, GridColumn, GridManager, GridSelect, GridToolbar, Guide, Hrapprover, Image, ImageViewer, Input, IpAddress, Layout, Link, LinkMenu, LoadList, Loading, Locales, LogonUser, Logout, Menu, Message, Milestone, MindMap, Modal, MonthRange, MonthTable, NavMenu, Notify, NumberAnimation, Numeric, Option, OptionGroup, Pager, PagerItem, Panel, Picker, PopUpload, Popconfirm, Popeditor, Popover, Popup, Progress, PullRefresh, QrCode, QuarterPanel, QueryBuilder, Radio, RadioButton, RadioGroup, Rate, Record, RecycleScroller, River, Roles, Row, ScrollText, Scrollbar, Search, Select, SelectDropdown, SelectMobile, SelectView, SelectWrapper, SelectedBox, Signature, Skeleton, SkeletonItem, Slider, SliderButton, SliderButtonGroup, Space, Split, StandardListItem, Statistic, Steps, Sticky, Switch, TabItem, Tabbar, TabbarItem, Table, Tabs, Tag, TagGroup, TagInput, TextPopup, Time, TimeLine, TimePanel, TimePicker, TimePickerMobile, TimeRange, TimeSelect, TimeSpinner, TimelineItem, ToggleMenu, Tooltip, TopBox, Transfer, TransferPanel, Tree, TreeMenu, TreeSelect, Upload, UploadDragger, UploadList, User, UserAccount, UserContact, UserHead, UserHeadGroup, UserLink, VirtualScrollBox, VirtualTree, Watermark, Wizard, YearRange, YearTable];
var install = function install2(app, opts) {
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function") return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.undefined";
var index = {
  ActionMenu,
  TinyActionMenu: ActionMenu,
  ActionSheet,
  TinyActionSheet: ActionSheet,
  Alert,
  TinyAlert: Alert,
  Amount,
  TinyAmount: Amount,
  Anchor,
  TinyAnchor: Anchor,
  Area,
  TinyArea: Area,
  AsyncFlowchart,
  TinyAsyncFlowchart: AsyncFlowchart,
  Autocomplete,
  TinyAutocomplete: Autocomplete,
  Badge,
  TinyBadge: Badge,
  BaseSelect,
  TinyBaseSelect: BaseSelect,
  Breadcrumb,
  TinyBreadcrumb: Breadcrumb,
  BreadcrumbItem,
  TinyBreadcrumbItem: BreadcrumbItem,
  BulletinBoard,
  TinyBulletinBoard: BulletinBoard,
  Button,
  TinyButton: Button,
  ButtonGroup,
  TinyButtonGroup: ButtonGroup,
  Calendar,
  TinyCalendar: Calendar,
  CalendarBar,
  TinyCalendarBar: CalendarBar,
  CalendarView,
  TinyCalendarView: CalendarView,
  Card,
  TinyCard: Card,
  CardGroup,
  TinyCardGroup: CardGroup,
  CardTemplate,
  TinyCardTemplate: CardTemplate,
  Carousel,
  TinyCarousel: Carousel,
  CarouselItem,
  TinyCarouselItem: CarouselItem,
  Cascader,
  TinyCascader: Cascader,
  CascaderMenu,
  TinyCascaderMenu: CascaderMenu,
  CascaderMobile,
  TinyCascaderMobile: CascaderMobile,
  CascaderNode,
  TinyCascaderNode: CascaderNode,
  CascaderPanel,
  TinyCascaderPanel: CascaderPanel,
  CascaderSelect,
  TinyCascaderSelect: CascaderSelect,
  CascaderView,
  TinyCascaderView: CascaderView,
  Cell,
  TinyCell: Cell,
  Checkbox,
  TinyCheckbox: Checkbox,
  CheckboxButton,
  TinyCheckboxButton: CheckboxButton,
  CheckboxGroup,
  TinyCheckboxGroup: CheckboxGroup,
  Col,
  TinyCol: Col,
  Collapse,
  TinyCollapse: Collapse,
  CollapseItem,
  TinyCollapseItem: CollapseItem,
  CollapseTransition,
  TinyCollapseTransition: CollapseTransition,
  ColorPicker,
  TinyColorPicker: ColorPicker,
  ColorSelectPanel,
  TinyColorSelectPanel: ColorSelectPanel,
  ColumnListGroup,
  TinyColumnListGroup: ColumnListGroup,
  ColumnListItem,
  TinyColumnListItem: ColumnListItem,
  Company,
  TinyCompany: Company,
  ConfigProvider,
  TinyConfigProvider: ConfigProvider,
  Container,
  TinyContainer: Container,
  Country,
  TinyCountry: Country,
  Crop,
  TinyCrop: Crop,
  Currency,
  TinyCurrency: Currency,
  DatePanel,
  TinyDatePanel: DatePanel,
  DatePicker,
  TinyDatePicker: DatePicker,
  DatePickerMobileFirst,
  TinyDatePickerMobileFirst: DatePickerMobileFirst,
  DateRange,
  TinyDateRange: DateRange,
  DateTable,
  TinyDateTable: DateTable,
  Dept,
  TinyDept: Dept,
  DialogBox,
  TinyDialogBox: DialogBox,
  DialogSelect,
  TinyDialogSelect: DialogSelect,
  Divider,
  TinyDivider: Divider,
  Drawer,
  TinyDrawer: Drawer,
  DropRoles,
  TinyDropRoles: DropRoles,
  DropTimes,
  TinyDropTimes: DropTimes,
  Dropdown,
  TinyDropdown: Dropdown,
  DropdownItem,
  TinyDropdownItem: DropdownItem,
  DropdownMenu,
  TinyDropdownMenu: DropdownMenu,
  DynamicScroller,
  TinyDynamicScroller: DynamicScroller,
  DynamicScrollerItem,
  TinyDynamicScrollerItem: DynamicScrollerItem,
  Espace,
  TinyEspace: Espace,
  Exception,
  TinyException: Exception,
  FallMenu,
  TinyFallMenu: FallMenu,
  FileUpload,
  TinyFileUpload: FileUpload,
  Filter,
  TinyFilter: Filter,
  FilterBar,
  TinyFilterBar: FilterBar,
  FilterBox,
  TinyFilterBox: FilterBox,
  FilterPanel,
  TinyFilterPanel: FilterPanel,
  FloatButton,
  TinyFloatButton: FloatButton,
  Floatbar,
  TinyFloatbar: Floatbar,
  FloatingButton,
  TinyFloatingButton: FloatingButton,
  Flowchart,
  TinyFlowchart: Flowchart,
  FluentEditor,
  TinyFluentEditor: FluentEditor,
  Form,
  TinyForm: Form,
  FormItem,
  TinyFormItem: FormItem,
  Fullscreen,
  TinyFullscreen: Fullscreen,
  Grid,
  TinyGrid: Grid,
  GridColumn,
  TinyGridColumn: GridColumn,
  GridManager,
  TinyGridManager: GridManager,
  GridSelect,
  TinyGridSelect: GridSelect,
  GridToolbar,
  TinyGridToolbar: GridToolbar,
  Guide,
  TinyGuide: Guide,
  Hrapprover,
  TinyHrapprover: Hrapprover,
  Image,
  TinyImage: Image,
  ImageViewer,
  TinyImageViewer: ImageViewer,
  Input,
  TinyInput: Input,
  IpAddress,
  TinyIpAddress: IpAddress,
  Layout,
  TinyLayout: Layout,
  Link,
  TinyLink: Link,
  LinkMenu,
  TinyLinkMenu: LinkMenu,
  LoadList,
  TinyLoadList: LoadList,
  Loading,
  TinyLoading: Loading,
  Locales,
  TinyLocales: Locales,
  LogonUser,
  TinyLogonUser: LogonUser,
  Logout,
  TinyLogout: Logout,
  Menu,
  TinyMenu: Menu,
  Message,
  TinyMessage: Message,
  Milestone,
  TinyMilestone: Milestone,
  MindMap,
  TinyMindMap: MindMap,
  Modal,
  TinyModal: Modal,
  MonthRange,
  TinyMonthRange: MonthRange,
  MonthTable,
  TinyMonthTable: MonthTable,
  NavMenu,
  TinyNavMenu: NavMenu,
  Notify,
  TinyNotify: Notify,
  NumberAnimation,
  TinyNumberAnimation: NumberAnimation,
  Numeric,
  TinyNumeric: Numeric,
  Option,
  TinyOption: Option,
  OptionGroup,
  TinyOptionGroup: OptionGroup,
  Pager,
  TinyPager: Pager,
  PagerItem,
  TinyPagerItem: PagerItem,
  Panel,
  TinyPanel: Panel,
  Picker,
  TinyPicker: Picker,
  PopUpload,
  TinyPopUpload: PopUpload,
  Popconfirm,
  TinyPopconfirm: Popconfirm,
  Popeditor,
  TinyPopeditor: Popeditor,
  Popover,
  TinyPopover: Popover,
  Popup,
  TinyPopup: Popup,
  Progress,
  TinyProgress: Progress,
  PullRefresh,
  TinyPullRefresh: PullRefresh,
  QrCode,
  TinyQrCode: QrCode,
  QuarterPanel,
  TinyQuarterPanel: QuarterPanel,
  QueryBuilder,
  TinyQueryBuilder: QueryBuilder,
  Radio,
  TinyRadio: Radio,
  RadioButton,
  TinyRadioButton: RadioButton,
  RadioGroup,
  TinyRadioGroup: RadioGroup,
  Rate,
  TinyRate: Rate,
  Record,
  TinyRecord: Record,
  RecycleScroller,
  TinyRecycleScroller: RecycleScroller,
  River,
  TinyRiver: River,
  Roles,
  TinyRoles: Roles,
  Row,
  TinyRow: Row,
  ScrollText,
  TinyScrollText: ScrollText,
  Scrollbar,
  TinyScrollbar: Scrollbar,
  Search,
  TinySearch: Search,
  Select,
  TinySelect: Select,
  SelectDropdown,
  TinySelectDropdown: SelectDropdown,
  SelectMobile,
  TinySelectMobile: SelectMobile,
  SelectView,
  TinySelectView: SelectView,
  SelectWrapper,
  TinySelectWrapper: SelectWrapper,
  SelectedBox,
  TinySelectedBox: SelectedBox,
  Signature,
  TinySignature: Signature,
  Skeleton,
  TinySkeleton: Skeleton,
  SkeletonItem,
  TinySkeletonItem: SkeletonItem,
  Slider,
  TinySlider: Slider,
  SliderButton,
  TinySliderButton: SliderButton,
  SliderButtonGroup,
  TinySliderButtonGroup: SliderButtonGroup,
  Space,
  TinySpace: Space,
  Split,
  TinySplit: Split,
  StandardListItem,
  TinyStandardListItem: StandardListItem,
  Statistic,
  TinyStatistic: Statistic,
  Steps,
  TinySteps: Steps,
  Sticky,
  TinySticky: Sticky,
  Switch,
  TinySwitch: Switch,
  TabItem,
  TinyTabItem: TabItem,
  Tabbar,
  TinyTabbar: Tabbar,
  TabbarItem,
  TinyTabbarItem: TabbarItem,
  Table,
  TinyTable: Table,
  Tabs,
  TinyTabs: Tabs,
  Tag,
  TinyTag: Tag,
  TagGroup,
  TinyTagGroup: TagGroup,
  TagInput,
  TinyTagInput: TagInput,
  TextPopup,
  TinyTextPopup: TextPopup,
  Time,
  TinyTime: Time,
  TimeLine,
  TinyTimeLine: TimeLine,
  TimePanel,
  TinyTimePanel: TimePanel,
  TimePicker,
  TinyTimePicker: TimePicker,
  TimePickerMobile,
  TinyTimePickerMobile: TimePickerMobile,
  TimeRange,
  TinyTimeRange: TimeRange,
  TimeSelect,
  TinyTimeSelect: TimeSelect,
  TimeSpinner,
  TinyTimeSpinner: TimeSpinner,
  TimelineItem,
  TinyTimelineItem: TimelineItem,
  ToggleMenu,
  TinyToggleMenu: ToggleMenu,
  Tooltip,
  TinyTooltip: Tooltip,
  TopBox,
  TinyTopBox: TopBox,
  Transfer,
  TinyTransfer: Transfer,
  TransferPanel,
  TinyTransferPanel: TransferPanel,
  Tree,
  TinyTree: Tree,
  TreeMenu,
  TinyTreeMenu: TreeMenu,
  TreeSelect,
  TinyTreeSelect: TreeSelect,
  Upload,
  TinyUpload: Upload,
  UploadDragger,
  TinyUploadDragger: UploadDragger,
  UploadList,
  TinyUploadList: UploadList,
  User,
  TinyUser: User,
  UserAccount,
  TinyUserAccount: UserAccount,
  UserContact,
  TinyUserContact: UserContact,
  UserHead,
  TinyUserHead: UserHead,
  UserHeadGroup,
  TinyUserHeadGroup: UserHeadGroup,
  UserLink,
  TinyUserLink: UserLink,
  VirtualScrollBox,
  TinyVirtualScrollBox: VirtualScrollBox,
  VirtualTree,
  TinyVirtualTree: VirtualTree,
  Watermark,
  TinyWatermark: Watermark,
  Wizard,
  TinyWizard: Wizard,
  YearRange,
  TinyYearRange: YearRange,
  YearTable,
  TinyYearTable: YearTable,
  install
};
export {
  default2 as ActionMenu,
  default4 as ActionSheet,
  default6 as Alert,
  default8 as Amount,
  default10 as Anchor,
  default12 as Area,
  default14 as AsyncFlowchart,
  default16 as Autocomplete,
  default18 as Badge,
  default20 as BaseSelect,
  default22 as Breadcrumb,
  default24 as BreadcrumbItem,
  default26 as BulletinBoard,
  default28 as Button,
  default30 as ButtonGroup,
  default32 as Calendar,
  default34 as CalendarBar,
  default36 as CalendarView,
  default38 as Card,
  default40 as CardGroup,
  default42 as CardTemplate,
  default44 as Carousel,
  default46 as CarouselItem,
  default48 as Cascader,
  default50 as CascaderMenu,
  default52 as CascaderMobile,
  default54 as CascaderNode,
  default56 as CascaderPanel,
  default58 as CascaderSelect,
  default60 as CascaderView,
  default62 as Cell,
  default64 as Checkbox,
  default66 as CheckboxButton,
  default68 as CheckboxGroup,
  default70 as Col,
  default72 as Collapse,
  default74 as CollapseItem,
  default76 as CollapseTransition,
  default78 as ColorPicker,
  default80 as ColorSelectPanel,
  default82 as ColumnListGroup,
  default84 as ColumnListItem,
  default86 as Company,
  default88 as ConfigProvider,
  default90 as Container,
  default92 as Country,
  default94 as Crop,
  default96 as Currency,
  default98 as DatePanel,
  default100 as DatePicker,
  default102 as DatePickerMobileFirst,
  default104 as DateRange,
  default106 as DateTable,
  default108 as Dept,
  default110 as DialogBox,
  default112 as DialogSelect,
  default114 as Divider,
  default116 as Drawer,
  default118 as DropRoles,
  default120 as DropTimes,
  default122 as Dropdown,
  default124 as DropdownItem,
  default126 as DropdownMenu,
  default128 as DynamicScroller,
  default130 as DynamicScrollerItem,
  default132 as Espace,
  default134 as Exception,
  default136 as FallMenu,
  default138 as FileUpload,
  default140 as Filter,
  default142 as FilterBar,
  default144 as FilterBox,
  default146 as FilterPanel,
  default148 as FloatButton,
  default150 as Floatbar,
  default152 as FloatingButton,
  default154 as Flowchart,
  default156 as FluentEditor,
  default158 as Form,
  default160 as FormItem,
  default162 as Fullscreen,
  default164 as Grid,
  default166 as GridColumn,
  default168 as GridManager,
  default170 as GridSelect,
  default172 as GridToolbar,
  default174 as Guide,
  default176 as Hrapprover,
  default178 as Image,
  default180 as ImageViewer,
  default182 as Input,
  default184 as IpAddress,
  default186 as Layout,
  default188 as Link,
  default190 as LinkMenu,
  default192 as LoadList,
  default194 as Loading,
  default196 as Locales,
  default198 as LogonUser,
  default200 as Logout,
  default202 as Menu,
  default204 as Message,
  default206 as Milestone,
  default208 as MindMap,
  default210 as Modal,
  default212 as MonthRange,
  default214 as MonthTable,
  default216 as NavMenu,
  default218 as Notify,
  default220 as NumberAnimation,
  default222 as Numeric,
  default224 as Option,
  default226 as OptionGroup,
  default228 as Pager,
  default230 as PagerItem,
  default232 as Panel,
  default234 as Picker,
  default236 as PopUpload,
  default238 as Popconfirm,
  default240 as Popeditor,
  default242 as Popover,
  default244 as Popup,
  default246 as Progress,
  default248 as PullRefresh,
  default250 as QrCode,
  default252 as QuarterPanel,
  default254 as QueryBuilder,
  default256 as Radio,
  default258 as RadioButton,
  default260 as RadioGroup,
  default262 as Rate,
  default264 as Record,
  default266 as RecycleScroller,
  default268 as River,
  default270 as Roles,
  default272 as Row,
  default274 as ScrollText,
  default276 as Scrollbar,
  default278 as Search,
  default280 as Select,
  default282 as SelectDropdown,
  default284 as SelectMobile,
  default286 as SelectView,
  default288 as SelectWrapper,
  default290 as SelectedBox,
  default292 as Signature,
  default294 as Skeleton,
  default296 as SkeletonItem,
  default298 as Slider,
  default300 as SliderButton,
  default302 as SliderButtonGroup,
  default304 as Space,
  default306 as Split,
  default308 as StandardListItem,
  default310 as Statistic,
  default312 as Steps,
  default314 as Sticky,
  default316 as Switch,
  default318 as TabItem,
  default320 as Tabbar,
  default322 as TabbarItem,
  default324 as Table,
  default326 as Tabs,
  default328 as Tag,
  default330 as TagGroup,
  default332 as TagInput,
  default334 as TextPopup,
  default336 as Time,
  default338 as TimeLine,
  default340 as TimePanel,
  default342 as TimePicker,
  default344 as TimePickerMobile,
  default346 as TimeRange,
  default348 as TimeSelect,
  default350 as TimeSpinner,
  default352 as TimelineItem,
  default3 as TinyActionMenu,
  default5 as TinyActionSheet,
  default7 as TinyAlert,
  default9 as TinyAmount,
  default11 as TinyAnchor,
  default13 as TinyArea,
  default15 as TinyAsyncFlowchart,
  default17 as TinyAutocomplete,
  default19 as TinyBadge,
  default21 as TinyBaseSelect,
  default23 as TinyBreadcrumb,
  default25 as TinyBreadcrumbItem,
  default27 as TinyBulletinBoard,
  default29 as TinyButton,
  default31 as TinyButtonGroup,
  default33 as TinyCalendar,
  default35 as TinyCalendarBar,
  default37 as TinyCalendarView,
  default39 as TinyCard,
  default41 as TinyCardGroup,
  default43 as TinyCardTemplate,
  default45 as TinyCarousel,
  default47 as TinyCarouselItem,
  default49 as TinyCascader,
  default51 as TinyCascaderMenu,
  default53 as TinyCascaderMobile,
  default55 as TinyCascaderNode,
  default57 as TinyCascaderPanel,
  default59 as TinyCascaderSelect,
  default61 as TinyCascaderView,
  default63 as TinyCell,
  default65 as TinyCheckbox,
  default67 as TinyCheckboxButton,
  default69 as TinyCheckboxGroup,
  default71 as TinyCol,
  default73 as TinyCollapse,
  default75 as TinyCollapseItem,
  default77 as TinyCollapseTransition,
  default79 as TinyColorPicker,
  default81 as TinyColorSelectPanel,
  default83 as TinyColumnListGroup,
  default85 as TinyColumnListItem,
  default87 as TinyCompany,
  default89 as TinyConfigProvider,
  default91 as TinyContainer,
  default93 as TinyCountry,
  default95 as TinyCrop,
  default97 as TinyCurrency,
  default99 as TinyDatePanel,
  default101 as TinyDatePicker,
  default103 as TinyDatePickerMobileFirst,
  default105 as TinyDateRange,
  default107 as TinyDateTable,
  default109 as TinyDept,
  default111 as TinyDialogBox,
  default113 as TinyDialogSelect,
  default115 as TinyDivider,
  default117 as TinyDrawer,
  default119 as TinyDropRoles,
  default121 as TinyDropTimes,
  default123 as TinyDropdown,
  default125 as TinyDropdownItem,
  default127 as TinyDropdownMenu,
  default129 as TinyDynamicScroller,
  default131 as TinyDynamicScrollerItem,
  default133 as TinyEspace,
  default135 as TinyException,
  default137 as TinyFallMenu,
  default139 as TinyFileUpload,
  default141 as TinyFilter,
  default143 as TinyFilterBar,
  default145 as TinyFilterBox,
  default147 as TinyFilterPanel,
  default149 as TinyFloatButton,
  default151 as TinyFloatbar,
  default153 as TinyFloatingButton,
  default155 as TinyFlowchart,
  default157 as TinyFluentEditor,
  default159 as TinyForm,
  default161 as TinyFormItem,
  default163 as TinyFullscreen,
  default165 as TinyGrid,
  default167 as TinyGridColumn,
  default169 as TinyGridManager,
  default171 as TinyGridSelect,
  default173 as TinyGridToolbar,
  default175 as TinyGuide,
  default177 as TinyHrapprover,
  default179 as TinyImage,
  default181 as TinyImageViewer,
  default183 as TinyInput,
  default185 as TinyIpAddress,
  default187 as TinyLayout,
  default189 as TinyLink,
  default191 as TinyLinkMenu,
  default193 as TinyLoadList,
  default195 as TinyLoading,
  default197 as TinyLocales,
  default199 as TinyLogonUser,
  default201 as TinyLogout,
  default203 as TinyMenu,
  default205 as TinyMessage,
  default207 as TinyMilestone,
  default209 as TinyMindMap,
  default211 as TinyModal,
  default213 as TinyMonthRange,
  default215 as TinyMonthTable,
  default217 as TinyNavMenu,
  default219 as TinyNotify,
  default221 as TinyNumberAnimation,
  default223 as TinyNumeric,
  default225 as TinyOption,
  default227 as TinyOptionGroup,
  default229 as TinyPager,
  default231 as TinyPagerItem,
  default233 as TinyPanel,
  default235 as TinyPicker,
  default237 as TinyPopUpload,
  default239 as TinyPopconfirm,
  default241 as TinyPopeditor,
  default243 as TinyPopover,
  default245 as TinyPopup,
  default247 as TinyProgress,
  default249 as TinyPullRefresh,
  default251 as TinyQrCode,
  default253 as TinyQuarterPanel,
  default255 as TinyQueryBuilder,
  default257 as TinyRadio,
  default259 as TinyRadioButton,
  default261 as TinyRadioGroup,
  default263 as TinyRate,
  default265 as TinyRecord,
  default267 as TinyRecycleScroller,
  default269 as TinyRiver,
  default271 as TinyRoles,
  default273 as TinyRow,
  default275 as TinyScrollText,
  default277 as TinyScrollbar,
  default279 as TinySearch,
  default281 as TinySelect,
  default283 as TinySelectDropdown,
  default285 as TinySelectMobile,
  default287 as TinySelectView,
  default289 as TinySelectWrapper,
  default291 as TinySelectedBox,
  default293 as TinySignature,
  default295 as TinySkeleton,
  default297 as TinySkeletonItem,
  default299 as TinySlider,
  default301 as TinySliderButton,
  default303 as TinySliderButtonGroup,
  default305 as TinySpace,
  default307 as TinySplit,
  default309 as TinyStandardListItem,
  default311 as TinyStatistic,
  default313 as TinySteps,
  default315 as TinySticky,
  default317 as TinySwitch,
  default319 as TinyTabItem,
  default321 as TinyTabbar,
  default323 as TinyTabbarItem,
  default325 as TinyTable,
  default327 as TinyTabs,
  default329 as TinyTag,
  default331 as TinyTagGroup,
  default333 as TinyTagInput,
  default335 as TinyTextPopup,
  default337 as TinyTime,
  default339 as TinyTimeLine,
  default341 as TinyTimePanel,
  default343 as TinyTimePicker,
  default345 as TinyTimePickerMobile,
  default347 as TinyTimeRange,
  default349 as TinyTimeSelect,
  default351 as TinyTimeSpinner,
  default353 as TinyTimelineItem,
  default354 as TinyToggleMenu,
  default356 as TinyTooltip,
  default358 as TinyTopBox,
  default360 as TinyTransfer,
  default362 as TinyTransferPanel,
  default364 as TinyTree,
  default366 as TinyTreeMenu,
  default368 as TinyTreeSelect,
  default370 as TinyUpload,
  default372 as TinyUploadDragger,
  default374 as TinyUploadList,
  default376 as TinyUser,
  default378 as TinyUserAccount,
  default380 as TinyUserContact,
  default382 as TinyUserHead,
  default384 as TinyUserHeadGroup,
  default386 as TinyUserLink,
  default388 as TinyVirtualScrollBox,
  default390 as TinyVirtualTree,
  default392 as TinyWatermark,
  default394 as TinyWizard,
  default396 as TinyYearRange,
  default398 as TinyYearTable,
  default355 as ToggleMenu,
  default357 as Tooltip,
  default359 as TopBox,
  default361 as Transfer,
  default363 as TransferPanel,
  default365 as Tree,
  default367 as TreeMenu,
  default369 as TreeSelect,
  default371 as Upload,
  default373 as UploadDragger,
  default375 as UploadList,
  default377 as User,
  default379 as UserAccount,
  default381 as UserContact,
  default383 as UserHead,
  default385 as UserHeadGroup,
  default387 as UserLink,
  default389 as VirtualScrollBox,
  default391 as VirtualTree,
  default393 as Watermark,
  default395 as Wizard,
  default397 as YearRange,
  default399 as YearTable,
  index as default,
  install,
  version
};
