// @opentiny/vue-hooks@3.30.0/dist/src/useTouch.js
const X = (t, e) => t > e ? "horizontal" : e > t ? "vertical" : "", h = (t) => t.touches[0], _ = (t) => () => {
  const e = t(0), v = t(0), c = t(0), n = t(0), a = t(0), l = t(0), u = t(""), s = t(!0), E = () => u.value === "vertical", F = () => u.value === "horizontal", r = () => {
    c.value = 0, n.value = 0, a.value = 0, l.value = 0, u.value = "", s.value = !0;
  };
  return {
    move: (i) => {
      const o = h(i);
      c.value = (o.clientX < 0 ? 0 : o.clientX) - e.value, n.value = o.clientY - v.value, a.value = Math.abs(c.value), l.value = Math.abs(n.value);
      const T = 10;
      (!u.value || a.value < T && l.value < T) && (u.value = X(a.value, l.value)), s.value && (a.value > 5 || l.value > 5) && (s.value = !1);
    },
    start: (i) => {
      r();
      const o = h(i);
      e.value = o.clientX, v.value = o.clientY;
    },
    reset: r,
    isVertical: E,
    isHorizontal: F,
    startX: e,
    startY: v,
    deltaX: c,
    deltaY: n,
    offsetX: a,
    offsetY: l,
    direction: u,
    isTap: s
  };
};
export {
  _ as useTouch
};
