// @opentiny/vue-hooks@3.30.0/dist/src/vue-popup.js
import { PopupManager as n, merge as f, isServer as T, addClass as g } from "@opentiny/utils";
let I = 1;
const x = ({
  onMounted: l,
  onBeforeUnmount: e,
  watch: p,
  vm: o,
  api: r,
  props: t,
  state: s,
  nextTick: d
}) => {
  l(() => {
    o._popupId = `popup-${I++}`, n.register(o._popupId, o);
  }), e(() => {
    n.deregister(o._popupId), n.closeModal(o._popupId);
  }), p(
    () => t.visible,
    (i) => {
      if (i) {
        if (o._opening)
          return;
        s.rendered ? r.open() : (s.rendered = !0, d(() => {
          r.open();
        }));
      } else
        r.close();
    }
  );
}, y = ({ state: l, vm: e }) => (p) => {
  l.rendered || (l.rendered = !0);
  const o = f({}, e.$props || e, p);
  e._closeTimer && (clearTimeout(e._closeTimer), e._closeTimer = null), clearTimeout(e._openTimer);
  const r = () => {
    if (T || l.opened)
      return;
    e._opening = !0;
    const s = e.$el, d = o.modal, i = o.zIndex;
    i && (n.zIndex = i), d && (e._closing && (n.closeModal(e._popupId), e._closing = !1), n.openModal(
      e._popupId,
      n.nextZIndex(),
      o.modalAppendToBody ? void 0 : s,
      o.modalClass,
      !o.noAnimation
    ), o.lockScroll && (n.fixBodyBorder(), g(document.body, n.popLockClass))), getComputedStyle(s).position === "static" && (s.style.position = "absolute"), s.style.zIndex = n.nextZIndex().toString(), l.opened = !0, e._opening = !1;
  }, t = Number(o.openDelay);
  t > 0 ? e._openTimer = setTimeout(() => {
    e._openTimer = null, r();
  }, t) : r();
}, M = ({ state: l, vm: e }) => () => {
  e._openTimer !== null && (clearTimeout(e._openTimer), e._openTimer = null), clearTimeout(e._closeTimer);
  const p = () => {
    e._closing = !0, l.opened = !1, n.closeModal(e._popupId), e._closing = !1;
  }, o = Number(e.closeDelay);
  o > 0 ? e._closeTimer = setTimeout(() => {
    e._closeTimer = null, p();
  }, o) : p();
}, m = (l) => {
  const { api: e, nextTick: p, onBeforeUnmount: o, onMounted: r, props: t, reactive: s, toRefs: d, vm: i, watch: u } = l, c = s({
    opened: !1,
    rendered: !1
  });
  x({ onMounted: r, onBeforeUnmount: o, watch: u, vm: i, api: e, props: t, state: c, nextTick: p });
  const a = y({ state: c, vm: i }), _ = M({ state: c, vm: i });
  return { open: a, close: _, PopupManager: n, ...d(c) };
};
export {
  m as usePopup
};
