// @opentiny/vue/mobile-first.js
import ActionSheet from "@opentiny/vue-action-sheet";
import { default as default2, default as default3 } from "@opentiny/vue-action-sheet";
import Alert from "@opentiny/vue-alert";
import { default as default4, default as default5 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default6, default as default7 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default8, default as default9 } from "@opentiny/vue-anchor";
import AsyncFlowchart from "@opentiny/vue-async-flowchart";
import { default as default10, default as default11 } from "@opentiny/vue-async-flowchart";
import Badge from "@opentiny/vue-badge";
import { default as default12, default as default13 } from "@opentiny/vue-badge";
import BaseSelect from "@opentiny/vue-base-select";
import { default as default14, default as default15 } from "@opentiny/vue-base-select";
import Button from "@opentiny/vue-button";
import { default as default16, default as default17 } from "@opentiny/vue-button";
import CalendarBar from "@opentiny/vue-calendar-bar";
import { default as default18, default as default19 } from "@opentiny/vue-calendar-bar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default20, default as default21 } from "@opentiny/vue-calendar-view";
import Card from "@opentiny/vue-card";
import { default as default22, default as default23 } from "@opentiny/vue-card";
import CardGroup from "@opentiny/vue-card-group";
import { default as default24, default as default25 } from "@opentiny/vue-card-group";
import Carousel from "@opentiny/vue-carousel";
import { default as default26, default as default27 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default28, default as default29 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default30, default as default31 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default32, default as default33 } from "@opentiny/vue-cascader-menu";
import CascaderMobile from "@opentiny/vue-cascader-mobile";
import { default as default34, default as default35 } from "@opentiny/vue-cascader-mobile";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default36, default as default37 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default38, default as default39 } from "@opentiny/vue-cascader-panel";
import CascaderSelect from "@opentiny/vue-cascader-select";
import { default as default40, default as default41 } from "@opentiny/vue-cascader-select";
import CascaderView from "@opentiny/vue-cascader-view";
import { default as default42, default as default43 } from "@opentiny/vue-cascader-view";
import Cell from "@opentiny/vue-cell";
import { default as default44, default as default45 } from "@opentiny/vue-cell";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default46, default as default47 } from "@opentiny/vue-checkbox";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default48, default as default49 } from "@opentiny/vue-checkbox-group";
import Collapse from "@opentiny/vue-collapse";
import { default as default50, default as default51 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default52, default as default53 } from "@opentiny/vue-collapse-item";
import ColumnListGroup from "@opentiny/vue-column-list-group";
import { default as default54, default as default55 } from "@opentiny/vue-column-list-group";
import ColumnListItem from "@opentiny/vue-column-list-item";
import { default as default56, default as default57 } from "@opentiny/vue-column-list-item";
import Currency from "@opentiny/vue-currency";
import { default as default58, default as default59 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default60, default as default61 } from "@opentiny/vue-date-panel";
import DateRange from "@opentiny/vue-date-range";
import { default as default62, default as default63 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default64, default as default65 } from "@opentiny/vue-date-table";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default66, default as default67 } from "@opentiny/vue-dialog-box";
import Divider from "@opentiny/vue-divider";
import { default as default68, default as default69 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default70, default as default71 } from "@opentiny/vue-drawer";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default72, default as default73 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default74, default as default75 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default76, default as default77 } from "@opentiny/vue-dropdown-menu";
import Exception from "@opentiny/vue-exception";
import { default as default78, default as default79 } from "@opentiny/vue-exception";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default80, default as default81 } from "@opentiny/vue-file-upload";
import Filter from "@opentiny/vue-filter";
import { default as default82, default as default83 } from "@opentiny/vue-filter";
import FilterBar from "@opentiny/vue-filter-bar";
import { default as default84, default as default85 } from "@opentiny/vue-filter-bar";
import FloatingButton from "@opentiny/vue-floating-button";
import { default as default86, default as default87 } from "@opentiny/vue-floating-button";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default88, default as default89 } from "@opentiny/vue-flowchart";
import FluentEditor from "@opentiny/vue-fluent-editor";
import { default as default90, default as default91 } from "@opentiny/vue-fluent-editor";
import Form from "@opentiny/vue-form";
import { default as default92, default as default93 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default94, default as default95 } from "@opentiny/vue-form-item";
import GridSelect from "@opentiny/vue-grid-select";
import { default as default96, default as default97 } from "@opentiny/vue-grid-select";
import Image from "@opentiny/vue-image";
import { default as default98, default as default99 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default100, default as default101 } from "@opentiny/vue-image-viewer";
import Input from "@opentiny/vue-input";
import { default as default102, default as default103 } from "@opentiny/vue-input";
import LoadList from "@opentiny/vue-load-list";
import { default as default104, default as default105 } from "@opentiny/vue-load-list";
import Loading from "@opentiny/vue-loading";
import { default as default106, default as default107 } from "@opentiny/vue-loading";
import Menu from "@opentiny/vue-menu";
import { default as default108, default as default109 } from "@opentiny/vue-menu";
import Modal from "@opentiny/vue-modal";
import { default as default110, default as default111 } from "@opentiny/vue-modal";
import Numeric from "@opentiny/vue-numeric";
import { default as default112, default as default113 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default114, default as default115 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default116, default as default117 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default118, default as default119 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default120, default as default121 } from "@opentiny/vue-pager-item";
import Picker from "@opentiny/vue-picker";
import { default as default122, default as default123 } from "@opentiny/vue-picker";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default124, default as default125 } from "@opentiny/vue-popconfirm";
import Popover from "@opentiny/vue-popover";
import { default as default126, default as default127 } from "@opentiny/vue-popover";
import Progress from "@opentiny/vue-progress";
import { default as default128, default as default129 } from "@opentiny/vue-progress";
import PullRefresh from "@opentiny/vue-pull-refresh";
import { default as default130, default as default131 } from "@opentiny/vue-pull-refresh";
import Radio from "@opentiny/vue-radio";
import { default as default132, default as default133 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default134, default as default135 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default136, default as default137 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default138, default as default139 } from "@opentiny/vue-rate";
import Record from "@opentiny/vue-record";
import { default as default140, default as default141 } from "@opentiny/vue-record";
import Search from "@opentiny/vue-search";
import { default as default142, default as default143 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default144, default as default145 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default146, default as default147 } from "@opentiny/vue-select-dropdown";
import SelectMobile from "@opentiny/vue-select-mobile";
import { default as default148, default as default149 } from "@opentiny/vue-select-mobile";
import SelectView from "@opentiny/vue-select-view";
import { default as default150, default as default151 } from "@opentiny/vue-select-view";
import SelectWrapper from "@opentiny/vue-select-wrapper";
import { default as default152, default as default153 } from "@opentiny/vue-select-wrapper";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default154, default as default155 } from "@opentiny/vue-selected-box";
import Signature from "@opentiny/vue-signature";
import { default as default156, default as default157 } from "@opentiny/vue-signature";
import Skeleton from "@opentiny/vue-skeleton";
import { default as default158, default as default159 } from "@opentiny/vue-skeleton";
import SkeletonItem from "@opentiny/vue-skeleton-item";
import { default as default160, default as default161 } from "@opentiny/vue-skeleton-item";
import Slider from "@opentiny/vue-slider";
import { default as default162, default as default163 } from "@opentiny/vue-slider";
import SliderButton from "@opentiny/vue-slider-button";
import { default as default164, default as default165 } from "@opentiny/vue-slider-button";
import SliderButtonGroup from "@opentiny/vue-slider-button-group";
import { default as default166, default as default167 } from "@opentiny/vue-slider-button-group";
import Space from "@opentiny/vue-space";
import { default as default168, default as default169 } from "@opentiny/vue-space";
import StandardListItem from "@opentiny/vue-standard-list-item";
import { default as default170, default as default171 } from "@opentiny/vue-standard-list-item";
import Steps from "@opentiny/vue-steps";
import { default as default172, default as default173 } from "@opentiny/vue-steps";
import Sticky from "@opentiny/vue-sticky";
import { default as default174, default as default175 } from "@opentiny/vue-sticky";
import Switch from "@opentiny/vue-switch";
import { default as default176, default as default177 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default178, default as default179 } from "@opentiny/vue-tab-item";
import Tabbar from "@opentiny/vue-tabbar";
import { default as default180, default as default181 } from "@opentiny/vue-tabbar";
import TabbarItem from "@opentiny/vue-tabbar-item";
import { default as default182, default as default183 } from "@opentiny/vue-tabbar-item";
import Tabs from "@opentiny/vue-tabs";
import { default as default184, default as default185 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default186, default as default187 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default188, default as default189 } from "@opentiny/vue-tag-group";
import Time from "@opentiny/vue-time";
import { default as default190, default as default191 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default192, default as default193 } from "@opentiny/vue-time-line";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default194, default as default195 } from "@opentiny/vue-time-panel";
import TimePickerMobile from "@opentiny/vue-time-picker-mobile";
import { default as default196, default as default197 } from "@opentiny/vue-time-picker-mobile";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default198, default as default199 } from "@opentiny/vue-time-spinner";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default200, default as default201 } from "@opentiny/vue-tooltip";
import TreeSelect from "@opentiny/vue-tree-select";
import { default as default202, default as default203 } from "@opentiny/vue-tree-select";
import Upload from "@opentiny/vue-upload";
import { default as default204, default as default205 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default206, default as default207 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default208, default as default209 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default210, default as default211 } from "@opentiny/vue-user";
import UserHead from "@opentiny/vue-user-head";
import { default as default212, default as default213 } from "@opentiny/vue-user-head";
import UserHeadGroup from "@opentiny/vue-user-head-group";
import { default as default214, default as default215 } from "@opentiny/vue-user-head-group";
import YearRange from "@opentiny/vue-year-range";
import { default as default216, default as default217 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default218, default as default219 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionSheet, Alert, Amount, Anchor, AsyncFlowchart, Badge, BaseSelect, Button, CalendarBar, CalendarView, Card, CardGroup, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderMobile, CascaderNode, CascaderPanel, CascaderSelect, CascaderView, Cell, Checkbox, CheckboxGroup, Collapse, CollapseItem, ColumnListGroup, ColumnListItem, Currency, DatePanel, DateRange, DateTable, DialogBox, Divider, Drawer, Dropdown, DropdownItem, DropdownMenu, Exception, FileUpload, Filter, FilterBar, FloatingButton, Flowchart, FluentEditor, Form, FormItem, GridSelect, Image, ImageViewer, Input, LoadList, Loading, Menu, Modal, Numeric, Option, OptionGroup, Pager, PagerItem, Picker, Popconfirm, Popover, Progress, PullRefresh, Radio, RadioButton, RadioGroup, Rate, Record, Search, Select, SelectDropdown, SelectMobile, SelectView, SelectWrapper, SelectedBox, Signature, Skeleton, SkeletonItem, Slider, SliderButton, SliderButtonGroup, Space, StandardListItem, Steps, Sticky, Switch, TabItem, Tabbar, TabbarItem, Tabs, Tag, TagGroup, Time, TimeLine, TimePanel, TimePickerMobile, TimeSpinner, Tooltip, TreeSelect, Upload, UploadDragger, UploadList, User, UserHead, UserHeadGroup, YearRange, YearTable];
var install = function install2(app, opts) {
  var _app$config;
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  var appProperties = ((_app$config = app.config) == null ? void 0 : _app$config.globalProperties) || app.prototype;
  appProperties.tiny_mode = {
    value: "mobile-first"
  };
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function") return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.30.0";
var mobileFirst = {
  ActionSheet,
  TinyActionSheet: ActionSheet,
  Alert,
  TinyAlert: Alert,
  Amount,
  TinyAmount: Amount,
  Anchor,
  TinyAnchor: Anchor,
  AsyncFlowchart,
  TinyAsyncFlowchart: AsyncFlowchart,
  Badge,
  TinyBadge: Badge,
  BaseSelect,
  TinyBaseSelect: BaseSelect,
  Button,
  TinyButton: Button,
  CalendarBar,
  TinyCalendarBar: CalendarBar,
  CalendarView,
  TinyCalendarView: CalendarView,
  Card,
  TinyCard: Card,
  CardGroup,
  TinyCardGroup: CardGroup,
  Carousel,
  TinyCarousel: Carousel,
  CarouselItem,
  TinyCarouselItem: CarouselItem,
  Cascader,
  TinyCascader: Cascader,
  CascaderMenu,
  TinyCascaderMenu: CascaderMenu,
  CascaderMobile,
  TinyCascaderMobile: CascaderMobile,
  CascaderNode,
  TinyCascaderNode: CascaderNode,
  CascaderPanel,
  TinyCascaderPanel: CascaderPanel,
  CascaderSelect,
  TinyCascaderSelect: CascaderSelect,
  CascaderView,
  TinyCascaderView: CascaderView,
  Cell,
  TinyCell: Cell,
  Checkbox,
  TinyCheckbox: Checkbox,
  CheckboxGroup,
  TinyCheckboxGroup: CheckboxGroup,
  Collapse,
  TinyCollapse: Collapse,
  CollapseItem,
  TinyCollapseItem: CollapseItem,
  ColumnListGroup,
  TinyColumnListGroup: ColumnListGroup,
  ColumnListItem,
  TinyColumnListItem: ColumnListItem,
  Currency,
  TinyCurrency: Currency,
  DatePanel,
  TinyDatePanel: DatePanel,
  DateRange,
  TinyDateRange: DateRange,
  DateTable,
  TinyDateTable: DateTable,
  DialogBox,
  TinyDialogBox: DialogBox,
  Divider,
  TinyDivider: Divider,
  Drawer,
  TinyDrawer: Drawer,
  Dropdown,
  TinyDropdown: Dropdown,
  DropdownItem,
  TinyDropdownItem: DropdownItem,
  DropdownMenu,
  TinyDropdownMenu: DropdownMenu,
  Exception,
  TinyException: Exception,
  FileUpload,
  TinyFileUpload: FileUpload,
  Filter,
  TinyFilter: Filter,
  FilterBar,
  TinyFilterBar: FilterBar,
  FloatingButton,
  TinyFloatingButton: FloatingButton,
  Flowchart,
  TinyFlowchart: Flowchart,
  FluentEditor,
  TinyFluentEditor: FluentEditor,
  Form,
  TinyForm: Form,
  FormItem,
  TinyFormItem: FormItem,
  GridSelect,
  TinyGridSelect: GridSelect,
  Image,
  TinyImage: Image,
  ImageViewer,
  TinyImageViewer: ImageViewer,
  Input,
  TinyInput: Input,
  LoadList,
  TinyLoadList: LoadList,
  Loading,
  TinyLoading: Loading,
  Menu,
  TinyMenu: Menu,
  Modal,
  TinyModal: Modal,
  Numeric,
  TinyNumeric: Numeric,
  Option,
  TinyOption: Option,
  OptionGroup,
  TinyOptionGroup: OptionGroup,
  Pager,
  TinyPager: Pager,
  PagerItem,
  TinyPagerItem: PagerItem,
  Picker,
  TinyPicker: Picker,
  Popconfirm,
  TinyPopconfirm: Popconfirm,
  Popover,
  TinyPopover: Popover,
  Progress,
  TinyProgress: Progress,
  PullRefresh,
  TinyPullRefresh: PullRefresh,
  Radio,
  TinyRadio: Radio,
  RadioButton,
  TinyRadioButton: RadioButton,
  RadioGroup,
  TinyRadioGroup: RadioGroup,
  Rate,
  TinyRate: Rate,
  Record,
  TinyRecord: Record,
  Search,
  TinySearch: Search,
  Select,
  TinySelect: Select,
  SelectDropdown,
  TinySelectDropdown: SelectDropdown,
  SelectMobile,
  TinySelectMobile: SelectMobile,
  SelectView,
  TinySelectView: SelectView,
  SelectWrapper,
  TinySelectWrapper: SelectWrapper,
  SelectedBox,
  TinySelectedBox: SelectedBox,
  Signature,
  TinySignature: Signature,
  Skeleton,
  TinySkeleton: Skeleton,
  SkeletonItem,
  TinySkeletonItem: SkeletonItem,
  Slider,
  TinySlider: Slider,
  SliderButton,
  TinySliderButton: SliderButton,
  SliderButtonGroup,
  TinySliderButtonGroup: SliderButtonGroup,
  Space,
  TinySpace: Space,
  StandardListItem,
  TinyStandardListItem: StandardListItem,
  Steps,
  TinySteps: Steps,
  Sticky,
  TinySticky: Sticky,
  Switch,
  TinySwitch: Switch,
  TabItem,
  TinyTabItem: TabItem,
  Tabbar,
  TinyTabbar: Tabbar,
  TabbarItem,
  TinyTabbarItem: TabbarItem,
  Tabs,
  TinyTabs: Tabs,
  Tag,
  TinyTag: Tag,
  TagGroup,
  TinyTagGroup: TagGroup,
  Time,
  TinyTime: Time,
  TimeLine,
  TinyTimeLine: TimeLine,
  TimePanel,
  TinyTimePanel: TimePanel,
  TimePickerMobile,
  TinyTimePickerMobile: TimePickerMobile,
  TimeSpinner,
  TinyTimeSpinner: TimeSpinner,
  Tooltip,
  TinyTooltip: Tooltip,
  TreeSelect,
  TinyTreeSelect: TreeSelect,
  Upload,
  TinyUpload: Upload,
  UploadDragger,
  TinyUploadDragger: UploadDragger,
  UploadList,
  TinyUploadList: UploadList,
  User,
  TinyUser: User,
  UserHead,
  TinyUserHead: UserHead,
  UserHeadGroup,
  TinyUserHeadGroup: UserHeadGroup,
  YearRange,
  TinyYearRange: YearRange,
  YearTable,
  TinyYearTable: YearTable,
  install
};
export {
  default2 as ActionSheet,
  default4 as Alert,
  default6 as Amount,
  default8 as Anchor,
  default10 as AsyncFlowchart,
  default12 as Badge,
  default14 as BaseSelect,
  default16 as Button,
  default18 as CalendarBar,
  default20 as CalendarView,
  default22 as Card,
  default24 as CardGroup,
  default26 as Carousel,
  default28 as CarouselItem,
  default30 as Cascader,
  default32 as CascaderMenu,
  default34 as CascaderMobile,
  default36 as CascaderNode,
  default38 as CascaderPanel,
  default40 as CascaderSelect,
  default42 as CascaderView,
  default44 as Cell,
  default46 as Checkbox,
  default48 as CheckboxGroup,
  default50 as Collapse,
  default52 as CollapseItem,
  default54 as ColumnListGroup,
  default56 as ColumnListItem,
  default58 as Currency,
  default60 as DatePanel,
  default62 as DateRange,
  default64 as DateTable,
  default66 as DialogBox,
  default68 as Divider,
  default70 as Drawer,
  default72 as Dropdown,
  default74 as DropdownItem,
  default76 as DropdownMenu,
  default78 as Exception,
  default80 as FileUpload,
  default82 as Filter,
  default84 as FilterBar,
  default86 as FloatingButton,
  default88 as Flowchart,
  default90 as FluentEditor,
  default92 as Form,
  default94 as FormItem,
  default96 as GridSelect,
  default98 as Image,
  default100 as ImageViewer,
  default102 as Input,
  default104 as LoadList,
  default106 as Loading,
  default108 as Menu,
  default110 as Modal,
  default112 as Numeric,
  default114 as Option,
  default116 as OptionGroup,
  default118 as Pager,
  default120 as PagerItem,
  default122 as Picker,
  default124 as Popconfirm,
  default126 as Popover,
  default128 as Progress,
  default130 as PullRefresh,
  default132 as Radio,
  default134 as RadioButton,
  default136 as RadioGroup,
  default138 as Rate,
  default140 as Record,
  default142 as Search,
  default144 as Select,
  default146 as SelectDropdown,
  default148 as SelectMobile,
  default150 as SelectView,
  default152 as SelectWrapper,
  default154 as SelectedBox,
  default156 as Signature,
  default158 as Skeleton,
  default160 as SkeletonItem,
  default162 as Slider,
  default164 as SliderButton,
  default166 as SliderButtonGroup,
  default168 as Space,
  default170 as StandardListItem,
  default172 as Steps,
  default174 as Sticky,
  default176 as Switch,
  default178 as TabItem,
  default180 as Tabbar,
  default182 as TabbarItem,
  default184 as Tabs,
  default186 as Tag,
  default188 as TagGroup,
  default190 as Time,
  default192 as TimeLine,
  default194 as TimePanel,
  default196 as TimePickerMobile,
  default198 as TimeSpinner,
  default3 as TinyActionSheet,
  default5 as TinyAlert,
  default7 as TinyAmount,
  default9 as TinyAnchor,
  default11 as TinyAsyncFlowchart,
  default13 as TinyBadge,
  default15 as TinyBaseSelect,
  default17 as TinyButton,
  default19 as TinyCalendarBar,
  default21 as TinyCalendarView,
  default23 as TinyCard,
  default25 as TinyCardGroup,
  default27 as TinyCarousel,
  default29 as TinyCarouselItem,
  default31 as TinyCascader,
  default33 as TinyCascaderMenu,
  default35 as TinyCascaderMobile,
  default37 as TinyCascaderNode,
  default39 as TinyCascaderPanel,
  default41 as TinyCascaderSelect,
  default43 as TinyCascaderView,
  default45 as TinyCell,
  default47 as TinyCheckbox,
  default49 as TinyCheckboxGroup,
  default51 as TinyCollapse,
  default53 as TinyCollapseItem,
  default55 as TinyColumnListGroup,
  default57 as TinyColumnListItem,
  default59 as TinyCurrency,
  default61 as TinyDatePanel,
  default63 as TinyDateRange,
  default65 as TinyDateTable,
  default67 as TinyDialogBox,
  default69 as TinyDivider,
  default71 as TinyDrawer,
  default73 as TinyDropdown,
  default75 as TinyDropdownItem,
  default77 as TinyDropdownMenu,
  default79 as TinyException,
  default81 as TinyFileUpload,
  default83 as TinyFilter,
  default85 as TinyFilterBar,
  default87 as TinyFloatingButton,
  default89 as TinyFlowchart,
  default91 as TinyFluentEditor,
  default93 as TinyForm,
  default95 as TinyFormItem,
  default97 as TinyGridSelect,
  default99 as TinyImage,
  default101 as TinyImageViewer,
  default103 as TinyInput,
  default105 as TinyLoadList,
  default107 as TinyLoading,
  default109 as TinyMenu,
  default111 as TinyModal,
  default113 as TinyNumeric,
  default115 as TinyOption,
  default117 as TinyOptionGroup,
  default119 as TinyPager,
  default121 as TinyPagerItem,
  default123 as TinyPicker,
  default125 as TinyPopconfirm,
  default127 as TinyPopover,
  default129 as TinyProgress,
  default131 as TinyPullRefresh,
  default133 as TinyRadio,
  default135 as TinyRadioButton,
  default137 as TinyRadioGroup,
  default139 as TinyRate,
  default141 as TinyRecord,
  default143 as TinySearch,
  default145 as TinySelect,
  default147 as TinySelectDropdown,
  default149 as TinySelectMobile,
  default151 as TinySelectView,
  default153 as TinySelectWrapper,
  default155 as TinySelectedBox,
  default157 as TinySignature,
  default159 as TinySkeleton,
  default161 as TinySkeletonItem,
  default163 as TinySlider,
  default165 as TinySliderButton,
  default167 as TinySliderButtonGroup,
  default169 as TinySpace,
  default171 as TinyStandardListItem,
  default173 as TinySteps,
  default175 as TinySticky,
  default177 as TinySwitch,
  default179 as TinyTabItem,
  default181 as TinyTabbar,
  default183 as TinyTabbarItem,
  default185 as TinyTabs,
  default187 as TinyTag,
  default189 as TinyTagGroup,
  default191 as TinyTime,
  default193 as TinyTimeLine,
  default195 as TinyTimePanel,
  default197 as TinyTimePickerMobile,
  default199 as TinyTimeSpinner,
  default200 as TinyTooltip,
  default202 as TinyTreeSelect,
  default204 as TinyUpload,
  default206 as TinyUploadDragger,
  default208 as TinyUploadList,
  default210 as TinyUser,
  default212 as TinyUserHead,
  default214 as TinyUserHeadGroup,
  default216 as TinyYearRange,
  default218 as TinyYearTable,
  default201 as Tooltip,
  default203 as TreeSelect,
  default205 as Upload,
  default207 as UploadDragger,
  default209 as UploadList,
  default211 as User,
  default213 as UserHead,
  default215 as UserHeadGroup,
  default217 as YearRange,
  default219 as YearTable,
  mobileFirst as default,
  install,
  version
};
