// @opentiny/vue/pc.js
import ActionMenu from "@opentiny/vue-action-menu";
import { default as default2, default as default3 } from "@opentiny/vue-action-menu";
import Alert from "@opentiny/vue-alert";
import { default as default4, default as default5 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default6, default as default7 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default8, default as default9 } from "@opentiny/vue-anchor";
import Area from "@opentiny/vue-area";
import { default as default10, default as default11 } from "@opentiny/vue-area";
import Autocomplete from "@opentiny/vue-autocomplete";
import { default as default12, default as default13 } from "@opentiny/vue-autocomplete";
import Badge from "@opentiny/vue-badge";
import { default as default14, default as default15 } from "@opentiny/vue-badge";
import BaseSelect from "@opentiny/vue-base-select";
import { default as default16, default as default17 } from "@opentiny/vue-base-select";
import Breadcrumb from "@opentiny/vue-breadcrumb";
import { default as default18, default as default19 } from "@opentiny/vue-breadcrumb";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { default as default20, default as default21 } from "@opentiny/vue-breadcrumb-item";
import BulletinBoard from "@opentiny/vue-bulletin-board";
import { default as default22, default as default23 } from "@opentiny/vue-bulletin-board";
import Button from "@opentiny/vue-button";
import { default as default24, default as default25 } from "@opentiny/vue-button";
import ButtonGroup from "@opentiny/vue-button-group";
import { default as default26, default as default27 } from "@opentiny/vue-button-group";
import Calendar from "@opentiny/vue-calendar";
import { default as default28, default as default29 } from "@opentiny/vue-calendar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default30, default as default31 } from "@opentiny/vue-calendar-view";
import Card from "@opentiny/vue-card";
import { default as default32, default as default33 } from "@opentiny/vue-card";
import CardGroup from "@opentiny/vue-card-group";
import { default as default34, default as default35 } from "@opentiny/vue-card-group";
import CardTemplate from "@opentiny/vue-card-template";
import { default as default36, default as default37 } from "@opentiny/vue-card-template";
import Carousel from "@opentiny/vue-carousel";
import { default as default38, default as default39 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default40, default as default41 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default42, default as default43 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default44, default as default45 } from "@opentiny/vue-cascader-menu";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default46, default as default47 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default48, default as default49 } from "@opentiny/vue-cascader-panel";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default50, default as default51 } from "@opentiny/vue-checkbox";
import CheckboxButton from "@opentiny/vue-checkbox-button";
import { default as default52, default as default53 } from "@opentiny/vue-checkbox-button";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default54, default as default55 } from "@opentiny/vue-checkbox-group";
import Col from "@opentiny/vue-col";
import { default as default56, default as default57 } from "@opentiny/vue-col";
import Collapse from "@opentiny/vue-collapse";
import { default as default58, default as default59 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default60, default as default61 } from "@opentiny/vue-collapse-item";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { default as default62, default as default63 } from "@opentiny/vue-collapse-transition";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default64, default as default65 } from "@opentiny/vue-color-picker";
import ColorSelectPanel from "@opentiny/vue-color-select-panel";
import { default as default66, default as default67 } from "@opentiny/vue-color-select-panel";
import Company from "@opentiny/vue-company";
import { default as default68, default as default69 } from "@opentiny/vue-company";
import ConfigProvider from "@opentiny/vue-config-provider";
import { default as default70, default as default71 } from "@opentiny/vue-config-provider";
import Container from "@opentiny/vue-container";
import { default as default72, default as default73 } from "@opentiny/vue-container";
import Country from "@opentiny/vue-country";
import { default as default74, default as default75 } from "@opentiny/vue-country";
import Crop from "@opentiny/vue-crop";
import { default as default76, default as default77 } from "@opentiny/vue-crop";
import Currency from "@opentiny/vue-currency";
import { default as default78, default as default79 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default80, default as default81 } from "@opentiny/vue-date-panel";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default82, default as default83 } from "@opentiny/vue-date-picker";
import DatePickerMobileFirst from "@opentiny/vue-date-picker-mobile-first";
import { default as default84, default as default85 } from "@opentiny/vue-date-picker-mobile-first";
import DateRange from "@opentiny/vue-date-range";
import { default as default86, default as default87 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default88, default as default89 } from "@opentiny/vue-date-table";
import Dept from "@opentiny/vue-dept";
import { default as default90, default as default91 } from "@opentiny/vue-dept";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default92, default as default93 } from "@opentiny/vue-dialog-box";
import DialogSelect from "@opentiny/vue-dialog-select";
import { default as default94, default as default95 } from "@opentiny/vue-dialog-select";
import Divider from "@opentiny/vue-divider";
import { default as default96, default as default97 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default98, default as default99 } from "@opentiny/vue-drawer";
import DropRoles from "@opentiny/vue-drop-roles";
import { default as default100, default as default101 } from "@opentiny/vue-drop-roles";
import DropTimes from "@opentiny/vue-drop-times";
import { default as default102, default as default103 } from "@opentiny/vue-drop-times";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default104, default as default105 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default106, default as default107 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default108, default as default109 } from "@opentiny/vue-dropdown-menu";
import DynamicScroller from "@opentiny/vue-dynamic-scroller";
import { default as default110, default as default111 } from "@opentiny/vue-dynamic-scroller";
import DynamicScrollerItem from "@opentiny/vue-dynamic-scroller-item";
import { default as default112, default as default113 } from "@opentiny/vue-dynamic-scroller-item";
import Espace from "@opentiny/vue-espace";
import { default as default114, default as default115 } from "@opentiny/vue-espace";
import Exception from "@opentiny/vue-exception";
import { default as default116, default as default117 } from "@opentiny/vue-exception";
import FallMenu from "@opentiny/vue-fall-menu";
import { default as default118, default as default119 } from "@opentiny/vue-fall-menu";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default120, default as default121 } from "@opentiny/vue-file-upload";
import FilterBox from "@opentiny/vue-filter-box";
import { default as default122, default as default123 } from "@opentiny/vue-filter-box";
import FilterPanel from "@opentiny/vue-filter-panel";
import { default as default124, default as default125 } from "@opentiny/vue-filter-panel";
import FloatButton from "@opentiny/vue-float-button";
import { default as default126, default as default127 } from "@opentiny/vue-float-button";
import Floatbar from "@opentiny/vue-floatbar";
import { default as default128, default as default129 } from "@opentiny/vue-floatbar";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default130, default as default131 } from "@opentiny/vue-flowchart";
import FluentEditor from "@opentiny/vue-fluent-editor";
import { default as default132, default as default133 } from "@opentiny/vue-fluent-editor";
import Form from "@opentiny/vue-form";
import { default as default134, default as default135 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default136, default as default137 } from "@opentiny/vue-form-item";
import Fullscreen from "@opentiny/vue-fullscreen";
import { default as default138, default as default139 } from "@opentiny/vue-fullscreen";
import Grid from "@opentiny/vue-grid";
import { default as default140, default as default141 } from "@opentiny/vue-grid";
import GridColumn from "@opentiny/vue-grid-column";
import { default as default142, default as default143 } from "@opentiny/vue-grid-column";
import GridManager from "@opentiny/vue-grid-manager";
import { default as default144, default as default145 } from "@opentiny/vue-grid-manager";
import GridSelect from "@opentiny/vue-grid-select";
import { default as default146, default as default147 } from "@opentiny/vue-grid-select";
import GridToolbar from "@opentiny/vue-grid-toolbar";
import { default as default148, default as default149 } from "@opentiny/vue-grid-toolbar";
import Guide from "@opentiny/vue-guide";
import { default as default150, default as default151 } from "@opentiny/vue-guide";
import Hrapprover from "@opentiny/vue-hrapprover";
import { default as default152, default as default153 } from "@opentiny/vue-hrapprover";
import Image from "@opentiny/vue-image";
import { default as default154, default as default155 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default156, default as default157 } from "@opentiny/vue-image-viewer";
import Input from "@opentiny/vue-input";
import { default as default158, default as default159 } from "@opentiny/vue-input";
import IpAddress from "@opentiny/vue-ip-address";
import { default as default160, default as default161 } from "@opentiny/vue-ip-address";
import Layout from "@opentiny/vue-layout";
import { default as default162, default as default163 } from "@opentiny/vue-layout";
import Link from "@opentiny/vue-link";
import { default as default164, default as default165 } from "@opentiny/vue-link";
import LinkMenu from "@opentiny/vue-link-menu";
import { default as default166, default as default167 } from "@opentiny/vue-link-menu";
import Loading from "@opentiny/vue-loading";
import { default as default168, default as default169 } from "@opentiny/vue-loading";
import Locales from "@opentiny/vue-locales";
import { default as default170, default as default171 } from "@opentiny/vue-locales";
import LogonUser from "@opentiny/vue-logon-user";
import { default as default172, default as default173 } from "@opentiny/vue-logon-user";
import Logout from "@opentiny/vue-logout";
import { default as default174, default as default175 } from "@opentiny/vue-logout";
import Message from "@opentiny/vue-message";
import { default as default176, default as default177 } from "@opentiny/vue-message";
import Milestone from "@opentiny/vue-milestone";
import { default as default178, default as default179 } from "@opentiny/vue-milestone";
import MindMap from "@opentiny/vue-mind-map";
import { default as default180, default as default181 } from "@opentiny/vue-mind-map";
import Modal from "@opentiny/vue-modal";
import { default as default182, default as default183 } from "@opentiny/vue-modal";
import MonthRange from "@opentiny/vue-month-range";
import { default as default184, default as default185 } from "@opentiny/vue-month-range";
import MonthTable from "@opentiny/vue-month-table";
import { default as default186, default as default187 } from "@opentiny/vue-month-table";
import NavMenu from "@opentiny/vue-nav-menu";
import { default as default188, default as default189 } from "@opentiny/vue-nav-menu";
import Notify from "@opentiny/vue-notify";
import { default as default190, default as default191 } from "@opentiny/vue-notify";
import NumberAnimation from "@opentiny/vue-number-animation";
import { default as default192, default as default193 } from "@opentiny/vue-number-animation";
import Numeric from "@opentiny/vue-numeric";
import { default as default194, default as default195 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default196, default as default197 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default198, default as default199 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default200, default as default201 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default202, default as default203 } from "@opentiny/vue-pager-item";
import Panel from "@opentiny/vue-panel";
import { default as default204, default as default205 } from "@opentiny/vue-panel";
import Picker from "@opentiny/vue-picker";
import { default as default206, default as default207 } from "@opentiny/vue-picker";
import PopUpload from "@opentiny/vue-pop-upload";
import { default as default208, default as default209 } from "@opentiny/vue-pop-upload";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default210, default as default211 } from "@opentiny/vue-popconfirm";
import Popeditor from "@opentiny/vue-popeditor";
import { default as default212, default as default213 } from "@opentiny/vue-popeditor";
import Popover from "@opentiny/vue-popover";
import { default as default214, default as default215 } from "@opentiny/vue-popover";
import Popup from "@opentiny/vue-popup";
import { default as default216, default as default217 } from "@opentiny/vue-popup";
import Progress from "@opentiny/vue-progress";
import { default as default218, default as default219 } from "@opentiny/vue-progress";
import QrCode from "@opentiny/vue-qr-code";
import { default as default220, default as default221 } from "@opentiny/vue-qr-code";
import QuarterPanel from "@opentiny/vue-quarter-panel";
import { default as default222, default as default223 } from "@opentiny/vue-quarter-panel";
import QueryBuilder from "@opentiny/vue-query-builder";
import { default as default224, default as default225 } from "@opentiny/vue-query-builder";
import Radio from "@opentiny/vue-radio";
import { default as default226, default as default227 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default228, default as default229 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default230, default as default231 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default232, default as default233 } from "@opentiny/vue-rate";
import RecycleScroller from "@opentiny/vue-recycle-scroller";
import { default as default234, default as default235 } from "@opentiny/vue-recycle-scroller";
import River from "@opentiny/vue-river";
import { default as default236, default as default237 } from "@opentiny/vue-river";
import Roles from "@opentiny/vue-roles";
import { default as default238, default as default239 } from "@opentiny/vue-roles";
import Row from "@opentiny/vue-row";
import { default as default240, default as default241 } from "@opentiny/vue-row";
import ScrollText from "@opentiny/vue-scroll-text";
import { default as default242, default as default243 } from "@opentiny/vue-scroll-text";
import Scrollbar from "@opentiny/vue-scrollbar";
import { default as default244, default as default245 } from "@opentiny/vue-scrollbar";
import Search from "@opentiny/vue-search";
import { default as default246, default as default247 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default248, default as default249 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default250, default as default251 } from "@opentiny/vue-select-dropdown";
import SelectWrapper from "@opentiny/vue-select-wrapper";
import { default as default252, default as default253 } from "@opentiny/vue-select-wrapper";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default254, default as default255 } from "@opentiny/vue-selected-box";
import Skeleton from "@opentiny/vue-skeleton";
import { default as default256, default as default257 } from "@opentiny/vue-skeleton";
import SkeletonItem from "@opentiny/vue-skeleton-item";
import { default as default258, default as default259 } from "@opentiny/vue-skeleton-item";
import Slider from "@opentiny/vue-slider";
import { default as default260, default as default261 } from "@opentiny/vue-slider";
import Space from "@opentiny/vue-space";
import { default as default262, default as default263 } from "@opentiny/vue-space";
import Split from "@opentiny/vue-split";
import { default as default264, default as default265 } from "@opentiny/vue-split";
import Statistic from "@opentiny/vue-statistic";
import { default as default266, default as default267 } from "@opentiny/vue-statistic";
import Steps from "@opentiny/vue-steps";
import { default as default268, default as default269 } from "@opentiny/vue-steps";
import Sticky from "@opentiny/vue-sticky";
import { default as default270, default as default271 } from "@opentiny/vue-sticky";
import Switch from "@opentiny/vue-switch";
import { default as default272, default as default273 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default274, default as default275 } from "@opentiny/vue-tab-item";
import Table from "@opentiny/vue-table";
import { default as default276, default as default277 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default278, default as default279 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default280, default as default281 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default282, default as default283 } from "@opentiny/vue-tag-group";
import TagInput from "@opentiny/vue-tag-input";
import { default as default284, default as default285 } from "@opentiny/vue-tag-input";
import TextPopup from "@opentiny/vue-text-popup";
import { default as default286, default as default287 } from "@opentiny/vue-text-popup";
import Time from "@opentiny/vue-time";
import { default as default288, default as default289 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default290, default as default291 } from "@opentiny/vue-time-line";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default292, default as default293 } from "@opentiny/vue-time-panel";
import TimePicker from "@opentiny/vue-time-picker";
import { default as default294, default as default295 } from "@opentiny/vue-time-picker";
import TimeRange from "@opentiny/vue-time-range";
import { default as default296, default as default297 } from "@opentiny/vue-time-range";
import TimeSelect from "@opentiny/vue-time-select";
import { default as default298, default as default299 } from "@opentiny/vue-time-select";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default300, default as default301 } from "@opentiny/vue-time-spinner";
import TimelineItem from "@opentiny/vue-timeline-item";
import { default as default302, default as default303 } from "@opentiny/vue-timeline-item";
import ToggleMenu from "@opentiny/vue-toggle-menu";
import { default as default304, default as default305 } from "@opentiny/vue-toggle-menu";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default306, default as default307 } from "@opentiny/vue-tooltip";
import TopBox from "@opentiny/vue-top-box";
import { default as default308, default as default309 } from "@opentiny/vue-top-box";
import Transfer from "@opentiny/vue-transfer";
import { default as default310, default as default311 } from "@opentiny/vue-transfer";
import TransferPanel from "@opentiny/vue-transfer-panel";
import { default as default312, default as default313 } from "@opentiny/vue-transfer-panel";
import Tree from "@opentiny/vue-tree";
import { default as default314, default as default315 } from "@opentiny/vue-tree";
import TreeMenu from "@opentiny/vue-tree-menu";
import { default as default316, default as default317 } from "@opentiny/vue-tree-menu";
import TreeSelect from "@opentiny/vue-tree-select";
import { default as default318, default as default319 } from "@opentiny/vue-tree-select";
import Upload from "@opentiny/vue-upload";
import { default as default320, default as default321 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default322, default as default323 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default324, default as default325 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default326, default as default327 } from "@opentiny/vue-user";
import UserAccount from "@opentiny/vue-user-account";
import { default as default328, default as default329 } from "@opentiny/vue-user-account";
import UserContact from "@opentiny/vue-user-contact";
import { default as default330, default as default331 } from "@opentiny/vue-user-contact";
import UserHead from "@opentiny/vue-user-head";
import { default as default332, default as default333 } from "@opentiny/vue-user-head";
import UserLink from "@opentiny/vue-user-link";
import { default as default334, default as default335 } from "@opentiny/vue-user-link";
import VirtualScrollBox from "@opentiny/vue-virtual-scroll-box";
import { default as default336, default as default337 } from "@opentiny/vue-virtual-scroll-box";
import VirtualTree from "@opentiny/vue-virtual-tree";
import { default as default338, default as default339 } from "@opentiny/vue-virtual-tree";
import Watermark from "@opentiny/vue-watermark";
import { default as default340, default as default341 } from "@opentiny/vue-watermark";
import Wizard from "@opentiny/vue-wizard";
import { default as default342, default as default343 } from "@opentiny/vue-wizard";
import YearRange from "@opentiny/vue-year-range";
import { default as default344, default as default345 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default346, default as default347 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionMenu, Alert, Amount, Anchor, Area, Autocomplete, Badge, BaseSelect, Breadcrumb, BreadcrumbItem, BulletinBoard, Button, ButtonGroup, Calendar, CalendarView, Card, CardGroup, CardTemplate, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderNode, CascaderPanel, Checkbox, CheckboxButton, CheckboxGroup, Col, Collapse, CollapseItem, CollapseTransition, ColorPicker, ColorSelectPanel, Company, ConfigProvider, Container, Country, Crop, Currency, DatePanel, DatePicker, DatePickerMobileFirst, DateRange, DateTable, Dept, DialogBox, DialogSelect, Divider, Drawer, DropRoles, DropTimes, Dropdown, DropdownItem, DropdownMenu, DynamicScroller, DynamicScrollerItem, Espace, Exception, FallMenu, FileUpload, FilterBox, FilterPanel, FloatButton, Floatbar, Flowchart, FluentEditor, Form, FormItem, Fullscreen, Grid, GridColumn, GridManager, GridSelect, GridToolbar, Guide, Hrapprover, Image, ImageViewer, Input, IpAddress, Layout, Link, LinkMenu, Loading, Locales, LogonUser, Logout, Message, Milestone, MindMap, Modal, MonthRange, MonthTable, NavMenu, Notify, NumberAnimation, Numeric, Option, OptionGroup, Pager, PagerItem, Panel, Picker, PopUpload, Popconfirm, Popeditor, Popover, Popup, Progress, QrCode, QuarterPanel, QueryBuilder, Radio, RadioButton, RadioGroup, Rate, RecycleScroller, River, Roles, Row, ScrollText, Scrollbar, Search, Select, SelectDropdown, SelectWrapper, SelectedBox, Skeleton, SkeletonItem, Slider, Space, Split, Statistic, Steps, Sticky, Switch, TabItem, Table, Tabs, Tag, TagGroup, TagInput, TextPopup, Time, TimeLine, TimePanel, TimePicker, TimeRange, TimeSelect, TimeSpinner, TimelineItem, ToggleMenu, Tooltip, TopBox, Transfer, TransferPanel, Tree, TreeMenu, TreeSelect, Upload, UploadDragger, UploadList, User, UserAccount, UserContact, UserHead, UserLink, VirtualScrollBox, VirtualTree, Watermark, Wizard, YearRange, YearTable];
var install = function install2(app, opts) {
  var _app$config;
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  var appProperties = ((_app$config = app.config) == null ? void 0 : _app$config.globalProperties) || app.prototype;
  appProperties.tiny_mode = {
    value: "pc"
  };
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function") return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.30.0";
var pc = {
  ActionMenu,
  TinyActionMenu: ActionMenu,
  Alert,
  TinyAlert: Alert,
  Amount,
  TinyAmount: Amount,
  Anchor,
  TinyAnchor: Anchor,
  Area,
  TinyArea: Area,
  Autocomplete,
  TinyAutocomplete: Autocomplete,
  Badge,
  TinyBadge: Badge,
  BaseSelect,
  TinyBaseSelect: BaseSelect,
  Breadcrumb,
  TinyBreadcrumb: Breadcrumb,
  BreadcrumbItem,
  TinyBreadcrumbItem: BreadcrumbItem,
  BulletinBoard,
  TinyBulletinBoard: BulletinBoard,
  Button,
  TinyButton: Button,
  ButtonGroup,
  TinyButtonGroup: ButtonGroup,
  Calendar,
  TinyCalendar: Calendar,
  CalendarView,
  TinyCalendarView: CalendarView,
  Card,
  TinyCard: Card,
  CardGroup,
  TinyCardGroup: CardGroup,
  CardTemplate,
  TinyCardTemplate: CardTemplate,
  Carousel,
  TinyCarousel: Carousel,
  CarouselItem,
  TinyCarouselItem: CarouselItem,
  Cascader,
  TinyCascader: Cascader,
  CascaderMenu,
  TinyCascaderMenu: CascaderMenu,
  CascaderNode,
  TinyCascaderNode: CascaderNode,
  CascaderPanel,
  TinyCascaderPanel: CascaderPanel,
  Checkbox,
  TinyCheckbox: Checkbox,
  CheckboxButton,
  TinyCheckboxButton: CheckboxButton,
  CheckboxGroup,
  TinyCheckboxGroup: CheckboxGroup,
  Col,
  TinyCol: Col,
  Collapse,
  TinyCollapse: Collapse,
  CollapseItem,
  TinyCollapseItem: CollapseItem,
  CollapseTransition,
  TinyCollapseTransition: CollapseTransition,
  ColorPicker,
  TinyColorPicker: ColorPicker,
  ColorSelectPanel,
  TinyColorSelectPanel: ColorSelectPanel,
  Company,
  TinyCompany: Company,
  ConfigProvider,
  TinyConfigProvider: ConfigProvider,
  Container,
  TinyContainer: Container,
  Country,
  TinyCountry: Country,
  Crop,
  TinyCrop: Crop,
  Currency,
  TinyCurrency: Currency,
  DatePanel,
  TinyDatePanel: DatePanel,
  DatePicker,
  TinyDatePicker: DatePicker,
  DatePickerMobileFirst,
  TinyDatePickerMobileFirst: DatePickerMobileFirst,
  DateRange,
  TinyDateRange: DateRange,
  DateTable,
  TinyDateTable: DateTable,
  Dept,
  TinyDept: Dept,
  DialogBox,
  TinyDialogBox: DialogBox,
  DialogSelect,
  TinyDialogSelect: DialogSelect,
  Divider,
  TinyDivider: Divider,
  Drawer,
  TinyDrawer: Drawer,
  DropRoles,
  TinyDropRoles: DropRoles,
  DropTimes,
  TinyDropTimes: DropTimes,
  Dropdown,
  TinyDropdown: Dropdown,
  DropdownItem,
  TinyDropdownItem: DropdownItem,
  DropdownMenu,
  TinyDropdownMenu: DropdownMenu,
  DynamicScroller,
  TinyDynamicScroller: DynamicScroller,
  DynamicScrollerItem,
  TinyDynamicScrollerItem: DynamicScrollerItem,
  Espace,
  TinyEspace: Espace,
  Exception,
  TinyException: Exception,
  FallMenu,
  TinyFallMenu: FallMenu,
  FileUpload,
  TinyFileUpload: FileUpload,
  FilterBox,
  TinyFilterBox: FilterBox,
  FilterPanel,
  TinyFilterPanel: FilterPanel,
  FloatButton,
  TinyFloatButton: FloatButton,
  Floatbar,
  TinyFloatbar: Floatbar,
  Flowchart,
  TinyFlowchart: Flowchart,
  FluentEditor,
  TinyFluentEditor: FluentEditor,
  Form,
  TinyForm: Form,
  FormItem,
  TinyFormItem: FormItem,
  Fullscreen,
  TinyFullscreen: Fullscreen,
  Grid,
  TinyGrid: Grid,
  GridColumn,
  TinyGridColumn: GridColumn,
  GridManager,
  TinyGridManager: GridManager,
  GridSelect,
  TinyGridSelect: GridSelect,
  GridToolbar,
  TinyGridToolbar: GridToolbar,
  Guide,
  TinyGuide: Guide,
  Hrapprover,
  TinyHrapprover: Hrapprover,
  Image,
  TinyImage: Image,
  ImageViewer,
  TinyImageViewer: ImageViewer,
  Input,
  TinyInput: Input,
  IpAddress,
  TinyIpAddress: IpAddress,
  Layout,
  TinyLayout: Layout,
  Link,
  TinyLink: Link,
  LinkMenu,
  TinyLinkMenu: LinkMenu,
  Loading,
  TinyLoading: Loading,
  Locales,
  TinyLocales: Locales,
  LogonUser,
  TinyLogonUser: LogonUser,
  Logout,
  TinyLogout: Logout,
  Message,
  TinyMessage: Message,
  Milestone,
  TinyMilestone: Milestone,
  MindMap,
  TinyMindMap: MindMap,
  Modal,
  TinyModal: Modal,
  MonthRange,
  TinyMonthRange: MonthRange,
  MonthTable,
  TinyMonthTable: MonthTable,
  NavMenu,
  TinyNavMenu: NavMenu,
  Notify,
  TinyNotify: Notify,
  NumberAnimation,
  TinyNumberAnimation: NumberAnimation,
  Numeric,
  TinyNumeric: Numeric,
  Option,
  TinyOption: Option,
  OptionGroup,
  TinyOptionGroup: OptionGroup,
  Pager,
  TinyPager: Pager,
  PagerItem,
  TinyPagerItem: PagerItem,
  Panel,
  TinyPanel: Panel,
  Picker,
  TinyPicker: Picker,
  PopUpload,
  TinyPopUpload: PopUpload,
  Popconfirm,
  TinyPopconfirm: Popconfirm,
  Popeditor,
  TinyPopeditor: Popeditor,
  Popover,
  TinyPopover: Popover,
  Popup,
  TinyPopup: Popup,
  Progress,
  TinyProgress: Progress,
  QrCode,
  TinyQrCode: QrCode,
  QuarterPanel,
  TinyQuarterPanel: QuarterPanel,
  QueryBuilder,
  TinyQueryBuilder: QueryBuilder,
  Radio,
  TinyRadio: Radio,
  RadioButton,
  TinyRadioButton: RadioButton,
  RadioGroup,
  TinyRadioGroup: RadioGroup,
  Rate,
  TinyRate: Rate,
  RecycleScroller,
  TinyRecycleScroller: RecycleScroller,
  River,
  TinyRiver: River,
  Roles,
  TinyRoles: Roles,
  Row,
  TinyRow: Row,
  ScrollText,
  TinyScrollText: ScrollText,
  Scrollbar,
  TinyScrollbar: Scrollbar,
  Search,
  TinySearch: Search,
  Select,
  TinySelect: Select,
  SelectDropdown,
  TinySelectDropdown: SelectDropdown,
  SelectWrapper,
  TinySelectWrapper: SelectWrapper,
  SelectedBox,
  TinySelectedBox: SelectedBox,
  Skeleton,
  TinySkeleton: Skeleton,
  SkeletonItem,
  TinySkeletonItem: SkeletonItem,
  Slider,
  TinySlider: Slider,
  Space,
  TinySpace: Space,
  Split,
  TinySplit: Split,
  Statistic,
  TinyStatistic: Statistic,
  Steps,
  TinySteps: Steps,
  Sticky,
  TinySticky: Sticky,
  Switch,
  TinySwitch: Switch,
  TabItem,
  TinyTabItem: TabItem,
  Table,
  TinyTable: Table,
  Tabs,
  TinyTabs: Tabs,
  Tag,
  TinyTag: Tag,
  TagGroup,
  TinyTagGroup: TagGroup,
  TagInput,
  TinyTagInput: TagInput,
  TextPopup,
  TinyTextPopup: TextPopup,
  Time,
  TinyTime: Time,
  TimeLine,
  TinyTimeLine: TimeLine,
  TimePanel,
  TinyTimePanel: TimePanel,
  TimePicker,
  TinyTimePicker: TimePicker,
  TimeRange,
  TinyTimeRange: TimeRange,
  TimeSelect,
  TinyTimeSelect: TimeSelect,
  TimeSpinner,
  TinyTimeSpinner: TimeSpinner,
  TimelineItem,
  TinyTimelineItem: TimelineItem,
  ToggleMenu,
  TinyToggleMenu: ToggleMenu,
  Tooltip,
  TinyTooltip: Tooltip,
  TopBox,
  TinyTopBox: TopBox,
  Transfer,
  TinyTransfer: Transfer,
  TransferPanel,
  TinyTransferPanel: TransferPanel,
  Tree,
  TinyTree: Tree,
  TreeMenu,
  TinyTreeMenu: TreeMenu,
  TreeSelect,
  TinyTreeSelect: TreeSelect,
  Upload,
  TinyUpload: Upload,
  UploadDragger,
  TinyUploadDragger: UploadDragger,
  UploadList,
  TinyUploadList: UploadList,
  User,
  TinyUser: User,
  UserAccount,
  TinyUserAccount: UserAccount,
  UserContact,
  TinyUserContact: UserContact,
  UserHead,
  TinyUserHead: UserHead,
  UserLink,
  TinyUserLink: UserLink,
  VirtualScrollBox,
  TinyVirtualScrollBox: VirtualScrollBox,
  VirtualTree,
  TinyVirtualTree: VirtualTree,
  Watermark,
  TinyWatermark: Watermark,
  Wizard,
  TinyWizard: Wizard,
  YearRange,
  TinyYearRange: YearRange,
  YearTable,
  TinyYearTable: YearTable,
  install
};
export {
  default2 as ActionMenu,
  default4 as Alert,
  default6 as Amount,
  default8 as Anchor,
  default10 as Area,
  default12 as Autocomplete,
  default14 as Badge,
  default16 as BaseSelect,
  default18 as Breadcrumb,
  default20 as BreadcrumbItem,
  default22 as BulletinBoard,
  default24 as Button,
  default26 as ButtonGroup,
  default28 as Calendar,
  default30 as CalendarView,
  default32 as Card,
  default34 as CardGroup,
  default36 as CardTemplate,
  default38 as Carousel,
  default40 as CarouselItem,
  default42 as Cascader,
  default44 as CascaderMenu,
  default46 as CascaderNode,
  default48 as CascaderPanel,
  default50 as Checkbox,
  default52 as CheckboxButton,
  default54 as CheckboxGroup,
  default56 as Col,
  default58 as Collapse,
  default60 as CollapseItem,
  default62 as CollapseTransition,
  default64 as ColorPicker,
  default66 as ColorSelectPanel,
  default68 as Company,
  default70 as ConfigProvider,
  default72 as Container,
  default74 as Country,
  default76 as Crop,
  default78 as Currency,
  default80 as DatePanel,
  default82 as DatePicker,
  default84 as DatePickerMobileFirst,
  default86 as DateRange,
  default88 as DateTable,
  default90 as Dept,
  default92 as DialogBox,
  default94 as DialogSelect,
  default96 as Divider,
  default98 as Drawer,
  default100 as DropRoles,
  default102 as DropTimes,
  default104 as Dropdown,
  default106 as DropdownItem,
  default108 as DropdownMenu,
  default110 as DynamicScroller,
  default112 as DynamicScrollerItem,
  default114 as Espace,
  default116 as Exception,
  default118 as FallMenu,
  default120 as FileUpload,
  default122 as FilterBox,
  default124 as FilterPanel,
  default126 as FloatButton,
  default128 as Floatbar,
  default130 as Flowchart,
  default132 as FluentEditor,
  default134 as Form,
  default136 as FormItem,
  default138 as Fullscreen,
  default140 as Grid,
  default142 as GridColumn,
  default144 as GridManager,
  default146 as GridSelect,
  default148 as GridToolbar,
  default150 as Guide,
  default152 as Hrapprover,
  default154 as Image,
  default156 as ImageViewer,
  default158 as Input,
  default160 as IpAddress,
  default162 as Layout,
  default164 as Link,
  default166 as LinkMenu,
  default168 as Loading,
  default170 as Locales,
  default172 as LogonUser,
  default174 as Logout,
  default176 as Message,
  default178 as Milestone,
  default180 as MindMap,
  default182 as Modal,
  default184 as MonthRange,
  default186 as MonthTable,
  default188 as NavMenu,
  default190 as Notify,
  default192 as NumberAnimation,
  default194 as Numeric,
  default196 as Option,
  default198 as OptionGroup,
  default200 as Pager,
  default202 as PagerItem,
  default204 as Panel,
  default206 as Picker,
  default208 as PopUpload,
  default210 as Popconfirm,
  default212 as Popeditor,
  default214 as Popover,
  default216 as Popup,
  default218 as Progress,
  default220 as QrCode,
  default222 as QuarterPanel,
  default224 as QueryBuilder,
  default226 as Radio,
  default228 as RadioButton,
  default230 as RadioGroup,
  default232 as Rate,
  default234 as RecycleScroller,
  default236 as River,
  default238 as Roles,
  default240 as Row,
  default242 as ScrollText,
  default244 as Scrollbar,
  default246 as Search,
  default248 as Select,
  default250 as SelectDropdown,
  default252 as SelectWrapper,
  default254 as SelectedBox,
  default256 as Skeleton,
  default258 as SkeletonItem,
  default260 as Slider,
  default262 as Space,
  default264 as Split,
  default266 as Statistic,
  default268 as Steps,
  default270 as Sticky,
  default272 as Switch,
  default274 as TabItem,
  default276 as Table,
  default278 as Tabs,
  default280 as Tag,
  default282 as TagGroup,
  default284 as TagInput,
  default286 as TextPopup,
  default288 as Time,
  default290 as TimeLine,
  default292 as TimePanel,
  default294 as TimePicker,
  default296 as TimeRange,
  default298 as TimeSelect,
  default300 as TimeSpinner,
  default302 as TimelineItem,
  default3 as TinyActionMenu,
  default5 as TinyAlert,
  default7 as TinyAmount,
  default9 as TinyAnchor,
  default11 as TinyArea,
  default13 as TinyAutocomplete,
  default15 as TinyBadge,
  default17 as TinyBaseSelect,
  default19 as TinyBreadcrumb,
  default21 as TinyBreadcrumbItem,
  default23 as TinyBulletinBoard,
  default25 as TinyButton,
  default27 as TinyButtonGroup,
  default29 as TinyCalendar,
  default31 as TinyCalendarView,
  default33 as TinyCard,
  default35 as TinyCardGroup,
  default37 as TinyCardTemplate,
  default39 as TinyCarousel,
  default41 as TinyCarouselItem,
  default43 as TinyCascader,
  default45 as TinyCascaderMenu,
  default47 as TinyCascaderNode,
  default49 as TinyCascaderPanel,
  default51 as TinyCheckbox,
  default53 as TinyCheckboxButton,
  default55 as TinyCheckboxGroup,
  default57 as TinyCol,
  default59 as TinyCollapse,
  default61 as TinyCollapseItem,
  default63 as TinyCollapseTransition,
  default65 as TinyColorPicker,
  default67 as TinyColorSelectPanel,
  default69 as TinyCompany,
  default71 as TinyConfigProvider,
  default73 as TinyContainer,
  default75 as TinyCountry,
  default77 as TinyCrop,
  default79 as TinyCurrency,
  default81 as TinyDatePanel,
  default83 as TinyDatePicker,
  default85 as TinyDatePickerMobileFirst,
  default87 as TinyDateRange,
  default89 as TinyDateTable,
  default91 as TinyDept,
  default93 as TinyDialogBox,
  default95 as TinyDialogSelect,
  default97 as TinyDivider,
  default99 as TinyDrawer,
  default101 as TinyDropRoles,
  default103 as TinyDropTimes,
  default105 as TinyDropdown,
  default107 as TinyDropdownItem,
  default109 as TinyDropdownMenu,
  default111 as TinyDynamicScroller,
  default113 as TinyDynamicScrollerItem,
  default115 as TinyEspace,
  default117 as TinyException,
  default119 as TinyFallMenu,
  default121 as TinyFileUpload,
  default123 as TinyFilterBox,
  default125 as TinyFilterPanel,
  default127 as TinyFloatButton,
  default129 as TinyFloatbar,
  default131 as TinyFlowchart,
  default133 as TinyFluentEditor,
  default135 as TinyForm,
  default137 as TinyFormItem,
  default139 as TinyFullscreen,
  default141 as TinyGrid,
  default143 as TinyGridColumn,
  default145 as TinyGridManager,
  default147 as TinyGridSelect,
  default149 as TinyGridToolbar,
  default151 as TinyGuide,
  default153 as TinyHrapprover,
  default155 as TinyImage,
  default157 as TinyImageViewer,
  default159 as TinyInput,
  default161 as TinyIpAddress,
  default163 as TinyLayout,
  default165 as TinyLink,
  default167 as TinyLinkMenu,
  default169 as TinyLoading,
  default171 as TinyLocales,
  default173 as TinyLogonUser,
  default175 as TinyLogout,
  default177 as TinyMessage,
  default179 as TinyMilestone,
  default181 as TinyMindMap,
  default183 as TinyModal,
  default185 as TinyMonthRange,
  default187 as TinyMonthTable,
  default189 as TinyNavMenu,
  default191 as TinyNotify,
  default193 as TinyNumberAnimation,
  default195 as TinyNumeric,
  default197 as TinyOption,
  default199 as TinyOptionGroup,
  default201 as TinyPager,
  default203 as TinyPagerItem,
  default205 as TinyPanel,
  default207 as TinyPicker,
  default209 as TinyPopUpload,
  default211 as TinyPopconfirm,
  default213 as TinyPopeditor,
  default215 as TinyPopover,
  default217 as TinyPopup,
  default219 as TinyProgress,
  default221 as TinyQrCode,
  default223 as TinyQuarterPanel,
  default225 as TinyQueryBuilder,
  default227 as TinyRadio,
  default229 as TinyRadioButton,
  default231 as TinyRadioGroup,
  default233 as TinyRate,
  default235 as TinyRecycleScroller,
  default237 as TinyRiver,
  default239 as TinyRoles,
  default241 as TinyRow,
  default243 as TinyScrollText,
  default245 as TinyScrollbar,
  default247 as TinySearch,
  default249 as TinySelect,
  default251 as TinySelectDropdown,
  default253 as TinySelectWrapper,
  default255 as TinySelectedBox,
  default257 as TinySkeleton,
  default259 as TinySkeletonItem,
  default261 as TinySlider,
  default263 as TinySpace,
  default265 as TinySplit,
  default267 as TinyStatistic,
  default269 as TinySteps,
  default271 as TinySticky,
  default273 as TinySwitch,
  default275 as TinyTabItem,
  default277 as TinyTable,
  default279 as TinyTabs,
  default281 as TinyTag,
  default283 as TinyTagGroup,
  default285 as TinyTagInput,
  default287 as TinyTextPopup,
  default289 as TinyTime,
  default291 as TinyTimeLine,
  default293 as TinyTimePanel,
  default295 as TinyTimePicker,
  default297 as TinyTimeRange,
  default299 as TinyTimeSelect,
  default301 as TinyTimeSpinner,
  default303 as TinyTimelineItem,
  default304 as TinyToggleMenu,
  default306 as TinyTooltip,
  default308 as TinyTopBox,
  default310 as TinyTransfer,
  default312 as TinyTransferPanel,
  default314 as TinyTree,
  default316 as TinyTreeMenu,
  default318 as TinyTreeSelect,
  default320 as TinyUpload,
  default322 as TinyUploadDragger,
  default324 as TinyUploadList,
  default326 as TinyUser,
  default328 as TinyUserAccount,
  default330 as TinyUserContact,
  default332 as TinyUserHead,
  default334 as TinyUserLink,
  default336 as TinyVirtualScrollBox,
  default338 as TinyVirtualTree,
  default340 as TinyWatermark,
  default342 as TinyWizard,
  default344 as TinyYearRange,
  default346 as TinyYearTable,
  default305 as ToggleMenu,
  default307 as Tooltip,
  default309 as TopBox,
  default311 as Transfer,
  default313 as TransferPanel,
  default315 as Tree,
  default317 as TreeMenu,
  default319 as TreeSelect,
  default321 as Upload,
  default323 as UploadDragger,
  default325 as UploadList,
  default327 as User,
  default329 as UserAccount,
  default331 as UserContact,
  default333 as UserHead,
  default335 as UserLink,
  default337 as VirtualScrollBox,
  default339 as VirtualTree,
  default341 as Watermark,
  default343 as Wizard,
  default345 as YearRange,
  default347 as YearTable,
  pc as default,
  install,
  version
};
