// @opentiny/vue-skeleton-item/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/skeleton-item/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { iconPicture } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, renderSlot, createVNode, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  // 骨架屏形态
  "shape-image": "flex justify-center items-center",
  "shape-circle": "rounded-full",
  "shape-square": "w-full h-10 mb-3",
  // 图片尺寸
  "image-small": "w-6 h-6",
  "image-medium": "w-7 h-7",
  "image-large": "w-8 h-8",
  // 圆角尺寸
  "circle-small": "w-6 h-6",
  "circle-medium": "w-7 h-7",
  "circle-large": "w-8 h-8"
};
var _sfc_main = defineComponent({
  props: [].concat(props, ["variant", "size"]),
  components: {
    IconPicture: iconPicture()
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_picture = resolveComponent("icon-picture");
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-skeleton-item",
      class: normalizeClass(_ctx.m("bg-color-fill-3", _ctx.variant === "image" && "fill-color-border", _ctx.variant !== "circle" && "rounded", _ctx.gcls("shape-" + _ctx.variant), _ctx.variant !== "square" ? _ctx.gcls(_ctx.variant + "-" + _ctx.size) : "", _ctx.state.isActive && "bg-w200h100 bg-gradient-to-r from-color-border-separator via-color-bg-2 to-color-border-separator animate-skeleton-loading-1500ms"))
    },
    [_ctx.variant === "image" ? renderSlot(_ctx.$slots, "default", {
      key: 0
    }, function() {
      return [createVNode(_component_icon_picture, {
        class: "w-3/5 h-3/5"
      })];
    }) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
