// @opentiny/vue-color-select-panel@3.30.0/lib/pc.js
import Button from '@opentiny/vue-button';
import Input from '@opentiny/vue-input';
import Numeric from '@opentiny/vue-numeric';
import { renderless as renderless$4, api as api$4 } from '@opentiny/vue-renderless/color-select-panel/vue';
import { defineComponent, setup, props, directive } from '@opentiny/vue-common';
import { renderless as renderless$2, api as api$2 } from '@opentiny/vue-renderless/color-select-panel/hue-select/vue';
import { api, renderless } from '@opentiny/vue-renderless/color-select-panel/sv-select/vue';
import { openBlock, createElementBlock, normalizeStyle, createElementVNode, Fragment, renderList, withModifiers, resolveComponent, createVNode, createBlock, createCommentVNode, resolveDirective, withDirectives, withCtx, toDisplayString, createTextVNode } from 'vue';
import { renderless as renderless$1, api as api$1 } from '@opentiny/vue-renderless/color-select-panel/linear-gradient/vue';
import { renderless as renderless$3, api as api$3 } from '@opentiny/vue-renderless/color-select-panel/alpha-select/vue';
import '@opentiny/vue-theme/color-select-panel/index.css';
import { Clickoutside } from '@opentiny/vue-directive';
import Select from '@opentiny/vue-select';
import Option from '@opentiny/vue-option';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main$4 = defineComponent({
  emit: ["ready", "svUpdate"],
  props: {
    color: {
      type: Object
    }
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      api,
      renderless,
      context,
      mono: true
    });
  }
});
var _hoisted_1$3 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "white"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_2$2 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "black"
  },
  null,
  -1
  /* HOISTED */
);
function _sfc_render$4(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-color-select-panel__inner__color-select",
      ref: "wrapper",
      style: normalizeStyle({
        background: _ctx.state.bg
      })
    },
    [_hoisted_1$3, _hoisted_2$2, createElementVNode(
      "div",
      {
        class: "cursor",
        ref: "cursor",
        style: normalizeStyle({
          top: _ctx.state.cursorTop + "px",
          left: _ctx.state.cursorLeft + "px",
          transform: "translate(-50%, -50%)"
        })
      },
      null,
      4
      /* STYLE */
    )],
    4
    /* STYLE */
  );
}
var SvSelect = /* @__PURE__ */ _export_sfc(_sfc_main$4, [["render", _sfc_render$4]]);
var _sfc_main$3 = defineComponent({
  props: {},
  setup: function setup2(props2, context) {
    return setup({
      props: props2,
      context,
      renderless: renderless$1,
      api: api$1,
      mono: true
    });
  }
});
var _hoisted_1$2 = ["onMousedown"];
function _sfc_render$3(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-color-select-panel__linear-gradient",
      ref: "linearGradientBar",
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.onClickBar && _ctx.onClickBar.apply(_ctx, arguments);
      }),
      style: normalizeStyle({
        background: _ctx.state.linearGradientBarBackground
      })
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.context.colorPoints, function(point, idx) {
        return openBlock(), createElementBlock("div", {
          class: "tiny-color-select-panel__linear-gradient__thumb",
          ref_for: true,
          ref: "thumb",
          key: idx,
          onClick: _cache[0] || (_cache[0] = withModifiers(function() {
          }, ["stop"])),
          onMousedown: function onMousedown(event) {
            return _ctx.onThumbMouseDown(event, point);
          },
          style: normalizeStyle({
            left: point.cursorLeft + "px",
            transform: "translateX(-50%)"
          })
        }, [createElementVNode(
          "div",
          {
            class: "tiny-color-select-panel__linear-gradient__thumb-heart",
            style: normalizeStyle({
              background: point.color.value
            })
          },
          null,
          4
          /* STYLE */
        )], 44, _hoisted_1$2);
      }),
      128
      /* KEYED_FRAGMENT */
    ))],
    4
    /* STYLE */
  );
}
var linearGradient = /* @__PURE__ */ _export_sfc(_sfc_main$3, [["render", _sfc_render$3]]);
var _sfc_main$2 = defineComponent({
  emits: ["hueUpdate", "svReady", "hueReady"],
  props: {
    color: {
      type: Object
    },
    alpha: {
      type: Boolean
    }
  },
  components: {
    SvSelect,
    LinearGradient: linearGradient
  },
  setup: function setup3(props2, context) {
    return setup({
      props: props2,
      context,
      renderless: renderless$2,
      api: api$2,
      mono: true
    });
  }
});
var _hoisted_1$1 = {
  class: "tiny-color-select-panel__inner"
};
var _hoisted_2$1 = {
  class: "tiny-color-select-panel__inner__hue-select",
  ref: "bar"
};
function _sfc_render$2(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_sv_select = resolveComponent("sv-select");
  var _component_linear_gradient = resolveComponent("linear-gradient");
  return openBlock(), createElementBlock("div", _hoisted_1$1, [createVNode(_component_sv_select, {
    color: _ctx.color,
    onReady: _ctx.onSvReady
  }, null, 8, ["color", "onReady"]), createElementVNode(
    "div",
    _hoisted_2$1,
    [createElementVNode(
      "div",
      {
        ref: "thumb",
        class: "tiny-color-select-panel__inner__hue-select-thumb",
        style: normalizeStyle({
          left: _ctx.state.thumbLeft + "px",
          transform: "translateX(-50%)"
        })
      },
      [createElementVNode(
        "div",
        {
          class: "tiny-color-select-panel__inner__hue-select-thumb-heart",
          style: normalizeStyle({
            background: _ctx.state.ctx.activeColor.color.value
          })
        },
        null,
        4
        /* STYLE */
      )],
      4
      /* STYLE */
    )],
    512
    /* NEED_PATCH */
  ), _ctx.state.ctx.colorMode === "linear-gradient" ? (openBlock(), createBlock(_component_linear_gradient, {
    key: 0
  })) : createCommentVNode("v-if", true)]);
}
var HueSelect = /* @__PURE__ */ _export_sfc(_sfc_main$2, [["render", _sfc_render$2]]);
var _sfc_main$1 = defineComponent({
  emits: ["ready"],
  props: {
    color: {
      type: Object
    }
  },
  setup: function setup4(props2, context) {
    return setup({
      props: props2,
      context,
      renderless: renderless$3,
      api: api$3,
      mono: true
    });
  }
});
function _sfc_render$1(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-color-select-panel__alpha",
      ref: "alphaWrapper",
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.onClick && _ctx.onClick.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      {
        class: "tiny-color-select-panel__alpha__slider",
        style: normalizeStyle({
          background: _ctx.state.background
        }),
        ref: "slider"
      },
      null,
      4
      /* STYLE */
    ), createElementVNode(
      "div",
      {
        class: "tiny-color-select-panel__alpha__thumb",
        style: normalizeStyle({
          top: "-2px",
          left: _ctx.state.left + "px",
          transform: "translateX(-50%)"
        }),
        ref: "alphaThumb"
      },
      [createElementVNode(
        "div",
        {
          class: "tiny-color-select-panel__inner__hue-select-thumb-heart",
          style: normalizeStyle({
            background: _ctx.state.activeColor.color.value
          })
        },
        null,
        4
        /* STYLE */
      )],
      4
      /* STYLE */
    )],
    512
    /* NEED_PATCH */
  );
}
var AlphaSelect = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render$1]]);
var GradientParser = GradientParser || {};
GradientParser.stringify = /* @__PURE__ */ (function() {
  var visitor = {
    "visit_linear-gradient": function visit_linearGradient(node) {
      return visitor.visit_gradient(node);
    },
    "visit_repeating-linear-gradient": function visit_repeatingLinearGradient(node) {
      return visitor.visit_gradient(node);
    },
    "visit_radial-gradient": function visit_radialGradient(node) {
      return visitor.visit_gradient(node);
    },
    "visit_repeating-radial-gradient": function visit_repeatingRadialGradient(node) {
      return visitor.visit_gradient(node);
    },
    "visit_conic-gradient": function visit_conicGradient(node) {
      return visitor.visit_gradient(node);
    },
    "visit_repeating-conic-gradient": function visit_repeatingConicGradient(node) {
      return visitor.visit_gradient(node);
    },
    "visit_gradient": function visit_gradient(node) {
      var orientation = visitor.visit(node.orientation);
      if (orientation) {
        orientation += ", ";
      }
      return node.type + "(" + orientation + visitor.visit(node.colorStops) + ")";
    },
    "visit_shape": function visit_shape(node) {
      var result = node.value, at = visitor.visit(node.at), style = visitor.visit(node.style);
      if (style) {
        result += " " + style;
      }
      if (at) {
        result += " at " + at;
      }
      return result;
    },
    "visit_default-radial": function visit_defaultRadial(node) {
      var result = "", at = visitor.visit(node.at);
      if (at) {
        if (node.hasAtKeyword) {
          result += "at " + at;
        } else {
          result += at;
        }
      }
      return result;
    },
    "visit_extent-keyword": function visit_extentKeyword(node) {
      var result = node.value, at = visitor.visit(node.at);
      if (at) {
        result += " at " + at;
      }
      return result;
    },
    "visit_position-keyword": function visit_positionKeyword(node) {
      return node.value;
    },
    "visit_position": function visit_position(node) {
      return visitor.visit(node.value.x) + " " + visitor.visit(node.value.y);
    },
    "visit_%": function visit_(node) {
      return node.value + "%";
    },
    "visit_em": function visit_em(node) {
      return node.value + "em";
    },
    "visit_px": function visit_px(node) {
      return node.value + "px";
    },
    "visit_rem": function visit_rem(node) {
      return node.value + "rem";
    },
    "visit_vw": function visit_vw(node) {
      return node.value + "vw";
    },
    "visit_vh": function visit_vh(node) {
      return node.value + "vh";
    },
    "visit_vmin": function visit_vmin(node) {
      return node.value + "vmin";
    },
    "visit_vmax": function visit_vmax(node) {
      return node.value + "vmax";
    },
    "visit_ch": function visit_ch(node) {
      return node.value + "ch";
    },
    "visit_ex": function visit_ex(node) {
      return node.value + "ex";
    },
    "visit_calc": function visit_calc(node) {
      return "calc(" + node.value + ")";
    },
    "visit_literal": function visit_literal(node) {
      return visitor.visit_color(node.value, node);
    },
    "visit_hex": function visit_hex(node) {
      return visitor.visit_color("#" + node.value, node);
    },
    "visit_rgb": function visit_rgb(node) {
      return visitor.visit_color("rgb(" + node.value.join(", ") + ")", node);
    },
    "visit_rgba": function visit_rgba(node) {
      return visitor.visit_color("rgba(" + node.value.join(", ") + ")", node);
    },
    "visit_hsl": function visit_hsl(node) {
      return visitor.visit_color("hsl(" + node.value[0] + ", " + node.value[1] + "%, " + node.value[2] + "%)", node);
    },
    "visit_hsla": function visit_hsla(node) {
      return visitor.visit_color("hsla(" + node.value[0] + ", " + node.value[1] + "%, " + node.value[2] + "%, " + node.value[3] + ")", node);
    },
    "visit_var": function visit_var(node) {
      return visitor.visit_color("var(" + node.value + ")", node);
    },
    "visit_color": function visit_color(resultColor, node) {
      var result = resultColor, length = visitor.visit(node.length);
      if (length) {
        result += " " + length;
      }
      var length2 = visitor.visit(node.length2);
      if (length2) {
        result += " " + length2;
      }
      return result;
    },
    "visit_angular": function visit_angular(node) {
      return node.value + (node.unit || "deg");
    },
    "visit_directional": function visit_directional(node) {
      return "to " + node.value;
    },
    "visit_conic": function visit_conic(node) {
      var result = "";
      if (node.angle) {
        result += "from " + visitor.visit(node.angle);
      }
      if (node.at) {
        if (result) {
          result += " ";
        }
        result += "at " + visitor.visit(node.at);
      }
      return result;
    },
    "visit_array": function visit_array(elements) {
      var result = "", size = elements.length;
      elements.forEach(function(element, i) {
        result += visitor.visit(element);
        if (i < size - 1) {
          result += ", ";
        }
      });
      return result;
    },
    "visit_object": function visit_object(obj) {
      if (obj.width && obj.height) {
        return visitor.visit(obj.width) + " " + visitor.visit(obj.height);
      }
      return "";
    },
    "visit": function visit(element) {
      if (!element) {
        return "";
      }
      if (element instanceof Array) {
        return visitor.visit_array(element);
      } else if (typeof element === "object" && !element.type) {
        return visitor.visit_object(element);
      } else if (element.type) {
        var nodeVisitor = visitor["visit_" + element.type];
        if (nodeVisitor) {
          return nodeVisitor(element);
        } else {
          throw Error("Missing visitor visit_" + element.type);
        }
      } else {
        throw Error("Invalid node.");
      }
    }
  };
  return function(root) {
    return visitor.visit(root);
  };
})();
var GradientParser = GradientParser || {};
GradientParser.parse = /* @__PURE__ */ (function() {
  var tokens = {
    linearGradient: /^(\-(webkit|o|ms|moz)\-)?(linear\-gradient)/i,
    repeatingLinearGradient: /^(\-(webkit|o|ms|moz)\-)?(repeating\-linear\-gradient)/i,
    radialGradient: /^(\-(webkit|o|ms|moz)\-)?(radial\-gradient)/i,
    repeatingRadialGradient: /^(\-(webkit|o|ms|moz)\-)?(repeating\-radial\-gradient)/i,
    conicGradient: /^(\-(webkit|o|ms|moz)\-)?(conic\-gradient)/i,
    repeatingConicGradient: /^(\-(webkit|o|ms|moz)\-)?(repeating\-conic\-gradient)/i,
    sideOrCorner: /^to (left (top|bottom)|right (top|bottom)|top (left|right)|bottom (left|right)|left|right|top|bottom)/i,
    extentKeywords: /^(closest\-side|closest\-corner|farthest\-side|farthest\-corner|contain|cover)/,
    positionKeywords: /^(left|center|right|top|bottom)/i,
    pixelValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))px/,
    percentageValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))\%/,
    emValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))em/,
    remValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))rem/,
    vwValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))vw/,
    vhValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))vh/,
    vminValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))vmin/,
    vmaxValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))vmax/,
    chValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))ch/,
    exValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))ex/,
    angleValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))deg/,
    radianValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))rad/,
    gradianValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))grad/,
    turnValue: /^(-?(([0-9]*\.[0-9]+)|([0-9]+\.?)))turn/,
    startCall: /^\(/,
    endCall: /^\)/,
    comma: /^,/,
    slash: /^\//,
    hexColor: /^\#([0-9a-fA-F]{8}|[0-9a-fA-F]{6}|[0-9a-fA-F]{4}|[0-9a-fA-F]{3})(?![0-9a-fA-F])/,
    literalColor: /^([a-zA-Z]+)/,
    rgbColor: /^rgb/i,
    rgbaColor: /^rgba/i,
    varColor: /^var/i,
    calcValue: /^calc/i,
    variableName: /^(--[a-zA-Z0-9-,\s\#]+)/,
    number: /^(([0-9]*\.[0-9]+)|([0-9]+\.?))/,
    hslColor: /^hsl/i,
    hslaColor: /^hsla/i
  };
  var input = "";
  function error(msg) {
    var err = new Error(input + ": " + msg);
    err.source = input;
    throw err;
  }
  function getAST() {
    var ast = matchListDefinitions();
    if (input.length > 0) {
      error("Invalid input not EOF");
    }
    return ast;
  }
  function matchListDefinitions() {
    return matchListing(matchDefinition);
  }
  function matchDefinition() {
    return matchGradient("linear-gradient", tokens.linearGradient, matchLinearOrientation) || matchGradient("repeating-linear-gradient", tokens.repeatingLinearGradient, matchLinearOrientation) || matchGradient("radial-gradient", tokens.radialGradient, matchListRadialOrientations) || matchGradient("repeating-radial-gradient", tokens.repeatingRadialGradient, matchListRadialOrientations) || matchGradient("conic-gradient", tokens.conicGradient, matchConicOrientation) || matchGradient("repeating-conic-gradient", tokens.repeatingConicGradient, matchConicOrientation);
  }
  function matchGradient(gradientType, pattern, orientationMatcher) {
    return matchCall(pattern, function(captures) {
      var orientation = orientationMatcher();
      if (orientation) {
        if (!scan(tokens.comma)) {
          error("Missing comma before color stops");
        }
      }
      return {
        type: gradientType,
        orientation,
        colorStops: matchListing(matchColorStop)
      };
    });
  }
  function matchCall(pattern, callback) {
    var captures = scan(pattern);
    if (captures) {
      if (!scan(tokens.startCall)) {
        error("Missing (");
      }
      var result = callback(captures);
      if (!scan(tokens.endCall)) {
        error("Missing )");
      }
      return result;
    }
  }
  function matchLinearOrientation() {
    var sideOrCorner = matchSideOrCorner();
    if (sideOrCorner) {
      return sideOrCorner;
    }
    var legacyDirection = match("position-keyword", tokens.positionKeywords, 1);
    if (legacyDirection) {
      return {
        type: "directional",
        value: legacyDirection.value
      };
    }
    return matchAngle();
  }
  function matchConicOrientation() {
    var angle = matchFrom();
    var atPosition = matchAtPosition();
    if (angle || atPosition) {
      return {
        type: "conic",
        angle: angle || void 0,
        at: atPosition || void 0
      };
    }
  }
  function matchFrom() {
    if (match("from", /^from/, 0)) {
      var angle = matchAngle();
      if (!angle) {
        error('Missing angle after "from" in conic-gradient');
      }
      return angle;
    }
  }
  function matchSideOrCorner() {
    return match("directional", tokens.sideOrCorner, 1);
  }
  function matchAngle() {
    return matchAngularWithUnit("deg", tokens.angleValue) || matchAngularWithUnit("rad", tokens.radianValue) || matchAngularWithUnit("grad", tokens.gradianValue) || matchAngularWithUnit("turn", tokens.turnValue);
  }
  function matchAngularWithUnit(unit, pattern) {
    var captures = scan(pattern);
    if (captures) {
      return {
        type: "angular",
        value: captures[1],
        unit
      };
    }
  }
  function matchListRadialOrientations() {
    var radialOrientations, radialOrientation = matchRadialOrientation(), lookaheadCache;
    if (radialOrientation) {
      radialOrientations = [];
      radialOrientations.push(radialOrientation);
      lookaheadCache = input;
      if (scan(tokens.comma)) {
        radialOrientation = matchRadialOrientation();
        if (radialOrientation) {
          radialOrientations.push(radialOrientation);
        } else {
          input = lookaheadCache;
        }
      }
    }
    return radialOrientations;
  }
  function matchRadialOrientation() {
    var radialType = matchCircle() || matchEllipse();
    if (radialType) {
      radialType.at = matchAtPosition();
    } else {
      var extent = matchExtentKeyword();
      if (extent) {
        radialType = extent;
        var positionAt = matchAtPosition();
        if (positionAt) {
          radialType.at = positionAt;
        }
      } else {
        var atPosition = matchAtPosition();
        if (atPosition) {
          radialType = {
            type: "default-radial",
            at: atPosition,
            hasAtKeyword: true
          };
        } else {
          var defaultPosition = matchPositioning();
          if (defaultPosition) {
            radialType = {
              type: "default-radial",
              at: defaultPosition
            };
          }
        }
      }
    }
    return radialType;
  }
  function matchCircle() {
    var circle = match("shape", /^(circle)/i, 0);
    if (circle) {
      circle.style = matchLength() || matchExtentKeyword();
    }
    return circle;
  }
  function matchEllipse() {
    var ellipse = match("shape", /^(ellipse)/i, 0);
    if (ellipse) {
      ellipse.style = matchPositioning() || matchDistance() || matchExtentKeyword();
    }
    return ellipse;
  }
  function matchExtentKeyword() {
    return match("extent-keyword", tokens.extentKeywords, 1);
  }
  function matchAtPosition() {
    if (match("position", /^at/, 0)) {
      var positioning = matchPositioning();
      if (!positioning) {
        error("Missing positioning value");
      }
      return positioning;
    }
  }
  function matchPositioning() {
    var location = matchCoordinates();
    if (location.x || location.y) {
      return {
        type: "position",
        value: location
      };
    }
  }
  function matchCoordinates() {
    return {
      x: matchDistance(),
      y: matchDistance()
    };
  }
  function matchListing(matcher) {
    var captures = matcher(), result = [];
    if (captures) {
      result.push(captures);
      while (scan(tokens.comma)) {
        captures = matcher();
        if (captures) {
          result.push(captures);
        } else {
          error("One extra comma");
        }
      }
    }
    return result;
  }
  function matchColorStop() {
    var color = matchColor();
    if (!color) {
      error("Expected color definition");
    }
    color.length = matchDistance();
    if (color.length) {
      color.length2 = matchDistance();
    }
    return color;
  }
  function matchColor() {
    return matchHexColor() || matchHSLAColor() || matchHSLColor() || matchRGBAColor() || matchRGBColor() || matchVarColor() || matchLiteralColor();
  }
  function matchLiteralColor() {
    return match("literal", tokens.literalColor, 0);
  }
  function matchHexColor() {
    return match("hex", tokens.hexColor, 1);
  }
  function matchRGBColor() {
    return matchCall(tokens.rgbColor, function() {
      return matchRGBValues("rgb");
    });
  }
  function matchRGBAColor() {
    return matchCall(tokens.rgbaColor, function() {
      return matchRGBValues("rgba");
    });
  }
  function matchRGBValues(baseType) {
    var r = matchNumber();
    if (scan(tokens.comma)) {
      var g = matchNumber();
      scan(tokens.comma);
      var b = matchNumber();
      var values = [r, g, b];
      if (scan(tokens.comma)) {
        values.push(matchNumber());
        return {
          type: "rgba",
          value: values
        };
      }
      return {
        type: baseType,
        value: values
      };
    } else {
      var g = matchNumber();
      var b = matchNumber();
      var values = [r, g, b];
      if (scan(tokens.slash)) {
        values.push(matchNumber());
        return {
          type: "rgba",
          value: values
        };
      }
      return {
        type: baseType,
        value: values
      };
    }
  }
  function matchVarColor() {
    return matchCall(tokens.varColor, function() {
      return {
        type: "var",
        value: matchVariableName()
      };
    });
  }
  function matchHSLColor() {
    return matchCall(tokens.hslColor, function() {
      return matchHSLValues("hsl");
    });
  }
  function matchHSLAColor() {
    return matchCall(tokens.hslaColor, function() {
      return matchHSLValues("hsla");
    });
  }
  function matchHSLValues(baseType) {
    var lookahead = scan(tokens.percentageValue);
    if (lookahead) {
      error("HSL hue value must be a number in degrees (0-360) or normalized (-360 to 360), not a percentage");
    }
    var hue = matchNumber();
    if (scan(tokens.comma)) {
      var captures = scan(tokens.percentageValue);
      var sat = captures ? captures[1] : null;
      scan(tokens.comma);
      captures = scan(tokens.percentageValue);
      var light = captures ? captures[1] : null;
      if (!sat || !light) {
        error("Expected percentage value for saturation and lightness in HSL");
      }
      if (scan(tokens.comma)) {
        var alpha = matchNumber();
        return {
          type: "hsla",
          value: [hue, sat, light, alpha]
        };
      }
      return {
        type: baseType,
        value: [hue, sat, light]
      };
    } else {
      var captures = scan(tokens.percentageValue);
      var sat = captures ? captures[1] : null;
      captures = scan(tokens.percentageValue);
      var light = captures ? captures[1] : null;
      if (!sat || !light) {
        error("Expected percentage value for saturation and lightness in HSL");
      }
      if (scan(tokens.slash)) {
        var alpha = matchNumber();
        return {
          type: "hsla",
          value: [hue, sat, light, alpha]
        };
      }
      return {
        type: baseType,
        value: [hue, sat, light]
      };
    }
  }
  function matchVariableName() {
    return scan(tokens.variableName)[1];
  }
  function matchNumber() {
    return scan(tokens.number)[1];
  }
  function matchDistance() {
    return match("%", tokens.percentageValue, 1) || matchPositionKeyword() || matchCalc() || matchLength();
  }
  function matchPositionKeyword() {
    return match("position-keyword", tokens.positionKeywords, 1);
  }
  function matchCalc() {
    return matchCall(tokens.calcValue, function() {
      var openParenCount = 1;
      var i = 0;
      while (openParenCount > 0 && i < input.length) {
        var char = input.charAt(i);
        if (char === "(") {
          openParenCount++;
        } else if (char === ")") {
          openParenCount--;
        }
        i++;
      }
      if (openParenCount > 0) {
        error("Missing closing parenthesis in calc() expression");
      }
      var calcContent = input.substring(0, i - 1);
      consume(i - 1);
      return {
        type: "calc",
        value: calcContent
      };
    });
  }
  function matchLength() {
    return match("px", tokens.pixelValue, 1) || match("em", tokens.emValue, 1) || match("rem", tokens.remValue, 1) || match("vw", tokens.vwValue, 1) || match("vh", tokens.vhValue, 1) || match("vmin", tokens.vminValue, 1) || match("vmax", tokens.vmaxValue, 1) || match("ch", tokens.chValue, 1) || match("ex", tokens.exValue, 1);
  }
  function match(type, pattern, captureIndex) {
    var captures = scan(pattern);
    if (captures) {
      return {
        type,
        value: captures[captureIndex]
      };
    }
  }
  function scan(regexp) {
    var captures, blankCaptures;
    blankCaptures = /^[\n\r\t\s]+/.exec(input);
    if (blankCaptures) {
      consume(blankCaptures[0].length);
    }
    captures = regexp.exec(input);
    if (captures) {
      consume(captures[0].length);
    }
    return captures;
  }
  function consume(size) {
    input = input.substring(size);
  }
  return function(code) {
    input = code.toString().trim();
    if (input.endsWith(";")) {
      input = input.slice(0, -1);
    }
    return getAST();
  };
})();
var parse = GradientParser.parse;
GradientParser.stringify;
({
  parse: GradientParser.parse,
  stringify: GradientParser.stringify
});
var _sfc_main = defineComponent({
  emits: ["update:modelValue", "cancel", "confirm", "color-update"],
  props: [].concat(props, ["modelValue", "visible", "alpha", "history", "predefine", "format", "enableHistory", "enablePredefineColor", "colorMode"]),
  components: {
    HueSelect,
    AlphaSelect,
    TinyButton: Button,
    TinyInput: Input,
    TinySelect: Select,
    TinyOption: Option,
    TinyNumeric: Numeric
  },
  directives: directive({
    Clickoutside
  }),
  setup: function setup5(props2, context) {
    return setup({
      props: props2,
      context,
      renderless: renderless$4,
      api: api$4,
      extendOptions: {
        parse
      }
    });
  }
});
var _hoisted_1 = {
  key: 1,
  class: "tiny-color-select-panel__no-alpha"
};
var _hoisted_2 = {
  class: "tiny-color-select-panel__tools"
};
var _hoisted_3 = {
  key: 0,
  class: "tiny-color-select-panel__tools__format-select"
};
var _hoisted_4 = {
  class: "tiny-color-select-panel__tools-style"
};
var _hoisted_5 = {
  key: 0,
  class: "tiny-color-select-panel__tools-hex"
};
var _hoisted_6 = {
  key: 0,
  class: "tiny-color-select-panel__tools-deg"
};
var _hoisted_7 = {
  key: 1,
  class: "tiny-color-select-panel__tools-hex"
};
var _hoisted_8 = {
  key: 2
};
var _hoisted_9 = {
  key: 0,
  class: "tiny-color-select-panel__history"
};
var _hoisted_10 = ["onClick"];
var _hoisted_11 = {
  key: 3,
  class: "tiny-color-select-panel__predefine-title"
};
var _hoisted_12 = {
  key: 4,
  class: "tiny-color-select-panel__predefine"
};
var _hoisted_13 = ["onClick"];
var _hoisted_14 = {
  class: "tiny-color-select-panel__tools-btns"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_hue_select = resolveComponent("hue-select");
  var _component_alpha_select = resolveComponent("alpha-select");
  var _component_tiny_option = resolveComponent("tiny-option");
  var _component_tiny_select = resolveComponent("tiny-select");
  var _component_tiny_input = resolveComponent("tiny-input");
  var _component_tiny_numeric = resolveComponent("tiny-numeric");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _directive_clickoutside = resolveDirective("clickoutside");
  return _ctx.state.showPicker ? withDirectives((openBlock(), createElementBlock("div", {
    key: 0,
    class: "tiny-color-select-panel",
    onClick: _cache[7] || (_cache[7] = withModifiers(function() {
    }, ["stop"]))
  }, [createVNode(_component_hue_select, {
    color: _ctx.state.color,
    onHueReady: _ctx.onHueReady,
    onSvReady: _ctx.onSvReady
  }, null, 8, ["color", "onHueReady", "onSvReady"]), _ctx.alpha ? (openBlock(), createBlock(_component_alpha_select, {
    key: 0,
    color: _ctx.state.color,
    onReady: _ctx.onAlphaReady
  }, null, 8, ["color", "onReady"])) : createCommentVNode("v-if", true), !_ctx.alpha ? (openBlock(), createElementBlock("div", _hoisted_1)) : createCommentVNode("v-if", true), createElementVNode(
    "div",
    {
      class: "tiny-color-select-panel__display",
      style: normalizeStyle({
        background: _ctx.state.isLinearGradient ? _ctx.state.linearGradient : _ctx.state.color.value
      })
    },
    null,
    4
    /* STYLE */
  ), createElementVNode("div", _hoisted_2, [_ctx.state.formats.length ? (openBlock(), createElementBlock("div", _hoisted_3, [createVNode(_component_tiny_select, {
    modelValue: _ctx.state.currentFormat,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.currentFormat = $event;
    }),
    size: "small"
  }, {
    default: withCtx(function() {
      return [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.formats, function(formatValue) {
          return openBlock(), createBlock(_component_tiny_option, {
            key: formatValue,
            value: formatValue,
            label: formatValue
          }, null, 8, ["value", "label"]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))];
    }),
    _: 1
    /* STABLE */
  }, 8, ["modelValue"])])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_4, [_ctx.state.currentFormat === "hex" || _ctx.state.currentFormat === "css" || !_ctx.state.currentFormat ? (openBlock(), createElementBlock("div", _hoisted_5, [createVNode(_component_tiny_input, {
    class: "tiny-color-select-panel__tools-hex1",
    modelValue: _ctx.state.input,
    "onUpdate:modelValue": _cache[1] || (_cache[1] = function($event) {
      return _ctx.state.input = $event;
    })
  }, null, 8, ["modelValue"]), _ctx.state.isLinearGradient ? (openBlock(), createElementBlock("div", _hoisted_6, [createVNode(_component_tiny_numeric, {
    modelValue: _ctx.state.ctx.deg,
    "onUpdate:modelValue": _cache[2] || (_cache[2] = function($event) {
      return _ctx.state.ctx.deg = $event;
    }),
    unit: "deg",
    "mouse-wheel": ""
  }, null, 8, ["modelValue"])])) : createCommentVNode("v-if", true)])) : (openBlock(), createElementBlock("div", _hoisted_7, [createVNode(_component_tiny_input, {
    class: "tiny-color-select-panel__tools-hex4",
    modelValue: _ctx.state.hexInput4,
    "onUpdate:modelValue": _cache[3] || (_cache[3] = function($event) {
      return _ctx.state.hexInput4 = $event;
    })
  }, null, 8, ["modelValue"]), createVNode(_component_tiny_input, {
    class: "tiny-color-select-panel__tools-hex5",
    modelValue: _ctx.state.hexInput5,
    "onUpdate:modelValue": _cache[4] || (_cache[4] = function($event) {
      return _ctx.state.hexInput5 = $event;
    })
  }, null, 8, ["modelValue"]), createVNode(_component_tiny_input, {
    class: "tiny-color-select-panel__tools-hex6",
    modelValue: _ctx.state.hexInput6,
    "onUpdate:modelValue": _cache[5] || (_cache[5] = function($event) {
      return _ctx.state.hexInput6 = $event;
    })
  }, null, 8, ["modelValue"]), createVNode(_component_tiny_input, {
    class: "tiny-color-select-panel__tools-hex7",
    modelValue: _ctx.state.hexInput7,
    "onUpdate:modelValue": _cache[6] || (_cache[6] = function($event) {
      return _ctx.state.hexInput7 = $event;
    })
  }, null, 8, ["modelValue"])]))])]), _ctx.state.enableHistory ? (openBlock(), createElementBlock("div", _hoisted_8, [_ctx.state.enableHistory && _ctx.state.stack.length ? (openBlock(), createElementBlock("div", _hoisted_9, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.stack, function(color, key) {
      return openBlock(), createElementBlock("div", {
        class: "tiny-color-select-panel__history__color-block",
        key,
        style: normalizeStyle({
          background: color
        }),
        onClick: function onClick() {
          return _ctx.onHistoryClick(color);
        }
      }, null, 12, _hoisted_10);
    }),
    128
    /* KEYED_FRAGMENT */
  ))])) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), _ctx.state.enablePredefineColor ? (openBlock(), createElementBlock(
    "div",
    _hoisted_11,
    toDisplayString(_ctx.t("ui.colorSelectPanel.predefine")),
    1
    /* TEXT */
  )) : createCommentVNode("v-if", true), _ctx.state.enablePredefineColor && _ctx.state.predefineStack.length ? (openBlock(), createElementBlock("div", _hoisted_12, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.predefineStack, function(color, key) {
      return openBlock(), createElementBlock("div", {
        class: "tiny-color-select-panel__predefine__color-block",
        key,
        style: normalizeStyle({
          background: color
        }),
        onClick: function onClick() {
          return _ctx.onPredefineColorClick(color);
        }
      }, null, 12, _hoisted_13);
    }),
    128
    /* KEYED_FRAGMENT */
  ))])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_14, [createVNode(_component_tiny_button, {
    size: "small",
    type: "primary",
    onClick: _ctx.onConfirm
  }, {
    default: withCtx(function() {
      return [createTextVNode(
        toDisplayString(_ctx.t("ui.colorSelectPanel.confirm")),
        1
        /* TEXT */
      )];
    }),
    _: 1
    /* STABLE */
  }, 8, ["onClick"]), createVNode(_component_tiny_button, {
    type: "text",
    text: _ctx.t("ui.colorSelectPanel.cancel"),
    size: "small",
    onClick: _ctx.onCancel
  }, null, 8, ["text", "onClick"])])])), [[_directive_clickoutside, _ctx.onClickOutside]]) : createCommentVNode("v-if", true);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
