// @opentiny/vue-grid-select@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var GridSelect = defineComponent({
  name: $prefix + "GridSelect",
  componentName: "GridSelect",
  props: _extends({}, $props, {
    // 基础 props
    clearable: Boolean,
    disabled: Boolean,
    size: String,
    placeholder: String,
    readonly: Boolean,
    modelValue: {},
    multiple: Boolean,
    // 弹框相关
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    placement: {
      type: String,
      default: "bottom-start"
    },
    dropStyle: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    // 标签相关
    collapseTags: Boolean,
    copyable: Boolean,
    textSplit: {
      type: String,
      default: ","
    },
    hoverExpand: Boolean,
    clickExpand: Boolean,
    showTips: {
      type: Boolean,
      default: true
    },
    // 搜索和过滤
    filterable: Boolean,
    filterMethod: Function,
    searchable: Boolean,
    // 远程搜索
    remote: Boolean,
    remoteMethod: Function,
    remoteConfig: {
      type: Object,
      default: function _default2() {
        return {
          showIcon: false,
          clearData: false,
          autoSearch: false
        };
      }
    },
    reserveKeyword: Boolean,
    extraQueryParams: {
      type: [Object, String, Boolean, Array, Number],
      default: ""
    },
    initQuery: Function,
    // 表格配置
    gridOp: {
      type: Object,
      default: function _default3() {
        return {};
      }
    },
    selectConfig: {
      type: Object,
      default: function _default4() {
        return {
          checkMethod: function checkMethod() {
            return true;
          }
        };
      }
    },
    radioConfig: {
      type: Object,
      default: function _default5() {
        return {
          checkMethod: function checkMethod() {
            return true;
          }
        };
      }
    },
    // 字段映射
    textField: {
      type: String,
      default: "label"
    },
    valueField: {
      type: String,
      default: "value"
    },
    // 多选相关
    multipleLimit: {
      type: Number,
      default: 0
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
GridSelect.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
GridSelect.install = function(Vue) {
  Vue.component(GridSelect.name, GridSelect);
};
GridSelect.version = version;
export {
  GridSelect as default
};
