// @opentiny/vue-grid-select@3.30.0/lib/pc.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/grid-select/vue';
import Grid from '@opentiny/vue-grid';
import BaseSelect from '@opentiny/vue-base-select';
import { resolveComponent, openBlock, createBlock, withCtx, createVNode, mergeProps, withModifiers } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "GridSelect",
  components: {
    TinyGrid: Grid,
    TinyBaseSelect: BaseSelect
  },
  props: _extends({}, props, {
    // 基础 props
    clearable: Boolean,
    disabled: Boolean,
    size: String,
    placeholder: String,
    readonly: Boolean,
    modelValue: {},
    multiple: Boolean,
    // 弹框相关
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    placement: {
      type: String,
      default: "bottom-start"
    },
    dropStyle: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    // 标签相关
    collapseTags: Boolean,
    copyable: Boolean,
    textSplit: {
      type: String,
      default: ","
    },
    hoverExpand: Boolean,
    clickExpand: Boolean,
    showTips: {
      type: Boolean,
      default: true
    },
    // 搜索和过滤
    filterable: Boolean,
    filterMethod: Function,
    searchable: Boolean,
    // 远程搜索
    remote: Boolean,
    remoteMethod: Function,
    remoteConfig: {
      type: Object,
      default: function _default2() {
        return {
          showIcon: false,
          clearData: false,
          autoSearch: false
        };
      }
    },
    reserveKeyword: Boolean,
    extraQueryParams: {
      type: [Object, String, Boolean, Array, Number],
      default: ""
    },
    initQuery: Function,
    // 表格配置
    gridOp: {
      type: Object,
      default: function _default3() {
        return {};
      }
    },
    selectConfig: {
      type: Object,
      default: function _default4() {
        return {
          checkMethod: function checkMethod() {
            return true;
          }
        };
      }
    },
    radioConfig: {
      type: Object,
      default: function _default5() {
        return {
          checkMethod: function checkMethod() {
            return true;
          }
        };
      }
    },
    // 字段映射
    textField: {
      type: String,
      default: "label"
    },
    valueField: {
      type: String,
      default: "value"
    },
    // 多选相关
    multipleLimit: {
      type: Number,
      default: 0
    }
  }),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_grid = resolveComponent("tiny-grid");
  var _component_tiny_base_select = resolveComponent("tiny-base-select");
  return openBlock(), createBlock(_component_tiny_base_select, {
    ref: "baseSelectRef",
    class: "tiny-grid-select",
    modelValue: _ctx.state.modelValue,
    "onUpdate:modelValue": _cache[1] || (_cache[1] = function($event) {
      return _ctx.state.modelValue = $event;
    }),
    clearable: _ctx.clearable,
    multiple: _ctx.multiple,
    filterable: _ctx.filterable,
    "filter-method": _ctx.filter,
    "text-field": _ctx.textField,
    "value-field": _ctx.valueField,
    size: _ctx.size,
    disabled: _ctx.disabled,
    placeholder: _ctx.placeholder,
    readonly: _ctx.readonly,
    "popper-class": _ctx.popperClass,
    "popper-append-to-body": _ctx.popperAppendToBody,
    placement: _ctx.placement,
    "drop-style": _ctx.dropStyle,
    "hover-expand": _ctx.hoverExpand,
    "click-expand": _ctx.clickExpand,
    "collapse-tags": _ctx.collapseTags,
    copyable: _ctx.copyable,
    "text-split": _ctx.textSplit,
    "show-tips": _ctx.showTips,
    searchable: _ctx.searchable,
    "multiple-limit": _ctx.multipleLimit,
    remote: _ctx.remote,
    "remote-method": _ctx.remoteMethod,
    "init-query": _ctx.initQuery,
    "extra-query-params": _ctx.extraQueryParams,
    "remote-config": _ctx.remoteConfig,
    onVisibleChange: _ctx.handleVisibleChange
  }, {
    panel: withCtx(function() {
      var _ctx$gridOp, _ctx$state$gridData;
      return [createVNode(_component_tiny_grid, mergeProps({
        ref: "gridRef",
        "auto-resize": "",
        "row-id": _ctx.valueField,
        "select-config": _ctx.buildSelectConfig(),
        "radio-config": _ctx.buildRadioConfig(),
        "highlight-current-row": true,
        columns: ((_ctx$gridOp = _ctx.gridOp) == null ? void 0 : _ctx$gridOp.columns) || [],
        data: Array.isArray(_ctx.state.gridData) ? _ctx.state.gridData : ((_ctx$state$gridData = _ctx.state.gridData) == null ? void 0 : _ctx$state$gridData.data) || _ctx.state.gridData || [],
        onSelectAll: _ctx.selectChange,
        onSelectChange: _ctx.selectChange,
        onRadioChange: _ctx.radioChange,
        onMousedown: _cache[0] || (_cache[0] = withModifiers(function() {
        }, ["stop"]))
      }, _ctx.gridOp), null, 16, ["row-id", "select-config", "radio-config", "columns", "data", "onSelectAll", "onSelectChange", "onRadioChange"])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["modelValue", "clearable", "multiple", "filterable", "filter-method", "text-field", "value-field", "size", "disabled", "placeholder", "readonly", "popper-class", "popper-append-to-body", "placement", "drop-style", "hover-expand", "click-expand", "collapse-tags", "copyable", "text-split", "show-tips", "searchable", "multiple-limit", "remote", "remote-method", "init-query", "extra-query-params", "remote-config", "onVisibleChange"]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
