// @opentiny/vue-skeleton-item@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/skeleton-item/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var $constants = {};
var skeletonItemProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  modelValue: String,
  variant: {
    type: String,
    default: "square"
  },
  size: {
    type: String,
    default: "medium"
  }
});
var SkeletonItem = defineComponent({
  name: $prefix + "SkeletonItem",
  props: skeletonItemProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
SkeletonItem.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
SkeletonItem.install = function(Vue) {
  Vue.component(SkeletonItem.name, SkeletonItem);
};
SkeletonItem.version = version;
export {
  SkeletonItem as default
};
