// @opentiny/vue-card-group/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/card-group/vue';
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { iconGridNoData } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, renderSlot, createVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "size-large": "grid-cols-[repeat(auto-fill,minmax(theme(spacing.112),1fr))]",
  "size-medium": "grid-cols-[repeat(auto-fill,minmax(theme(spacing.72),1fr))]",
  "size-small": "grid-cols-[repeat(auto-fill,minmax(theme(spacing.54),1fr))]",
  "size-mini": "grid-cols-[repeat(auto-fill,minmax(theme(spacing.32),1fr))]"
};
var _sfc_main = defineComponent({
  name: $prefix + "CardGroup",
  componentName: "CardGroup",
  components: {
    IconGridNoData: iconGridNoData()
  },
  emits: ["update:modelValue", "change"],
  props: [].concat(props, ["modelValue", "customClass", "height", "autoWidth", "size", "status", "checkType", "type", "disabled"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_grid_no_data = resolveComponent("icon-grid-no-data");
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-card-group",
      class: normalizeClass(["w-full grid gap-4", _ctx.gcls("size-" + _ctx.state.size)])
    },
    [renderSlot(_ctx.$slots, "default"), createVNode(_component_icon_grid_no_data, {
      "data-tag": "tiny-card-group-empty",
      class: "hidden first:block !w-full !h-28"
    })],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
