// @opentiny/vue-divider@3.30.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $props, $setup, $prefix } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/divider/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {};
var dividerProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  direction: {
    type: String,
    default: "horizontal"
  },
  color: {
    type: String,
    default: ""
  },
  borderStyle: {
    type: String,
    default: "solid"
  },
  contentPosition: {
    type: String,
    default: "center"
  },
  contentColor: {
    type: String,
    default: ""
  },
  contentBackgroundColor: {
    type: String,
    default: ""
  },
  fontSize: {
    type: String,
    default: function _default2() {
      return "14px";
    }
  },
  margin: String,
  offset: String,
  height: String,
  status: {
    type: String,
    values: ["default", "success", "warning", "error", "info"],
    default: function _default3() {
      return "default";
    }
  }
});
var Divider = defineComponent({
  name: $prefix + "Divider",
  props: dividerProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.undefined";
Divider.install = function(Vue) {
  Vue.component(Divider.name, Divider);
};
Divider.version = version;
export {
  Divider as default
};
