// @opentiny/vue-divider@3.30.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/divider/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, normalizeClass, normalizeStyle, createCommentVNode, renderSlot } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t) return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r) return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  props: [].concat(props, ["direction", "color", "borderStyle", "contentPosition", "contentColor", "contentBackgroundColor", "status", "margin", "offset", "fontSize", "height"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-divider", "tiny-divider--" + _ctx.direction, _ctx.direction === "vertical" ? "tiny-divider--" + _ctx.status : ""]),
      style: normalizeStyle(_ctx.state.rootStyle)
    },
    [_ctx.direction !== "vertical" ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        style: normalizeStyle(_extends({
          "border-color": _ctx.color
        }, _ctx.state.lineStyle)),
        class: normalizeClass(["tiny-divider-line", "tiny-divider--" + _ctx.status])
      },
      null,
      6
      /* CLASS, STYLE */
    )) : createCommentVNode("v-if", true), _ctx.slots.default ? (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        ref: "text",
        class: normalizeClass(["tiny-divider__text"]),
        style: normalizeStyle(_extends({
          color: _ctx.contentColor,
          "background-color": _ctx.contentBackgroundColor
        }, _ctx.state.textStyle))
      },
      [renderSlot(_ctx.$slots, "default")],
      4
      /* STYLE */
    )) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
