// @opentiny/vue-notify/lib/index.js
function _regenerator() {
  /*! regenerator-runtime -- Copyright (c) 2014-present, Facebook, Inc. -- license (MIT): https://github.com/babel/babel/blob/main/packages/babel-helpers/LICENSE */
  var e, t, r = "function" == typeof Symbol ? Symbol : {}, n = r.iterator || "@@iterator", o = r.toStringTag || "@@toStringTag";
  function i(r2, n2, o2, i2) {
    var c2 = n2 && n2.prototype instanceof Generator ? n2 : Generator, u2 = Object.create(c2.prototype);
    return _regeneratorDefine2(u2, "_invoke", (function(r3, n3, o3) {
      var i3, c3, u3, f2 = 0, p = o3 || [], y = false, G = { p: 0, n: 0, v: e, a: d, f: d.bind(e, 4), d: function d2(t2, r4) {
        return i3 = t2, c3 = 0, u3 = e, G.n = r4, a;
      } };
      function d(r4, n4) {
        for (c3 = r4, u3 = n4, t = 0; !y && f2 && !o4 && t < p.length; t++) {
          var o4, i4 = p[t], d2 = G.p, l = i4[2];
          r4 > 3 ? (o4 = l === n4) && (u3 = i4[(c3 = i4[4]) ? 5 : (c3 = 3, 3)], i4[4] = i4[5] = e) : i4[0] <= d2 && ((o4 = r4 < 2 && d2 < i4[1]) ? (c3 = 0, G.v = n4, G.n = i4[1]) : d2 < l && (o4 = r4 < 3 || i4[0] > n4 || n4 > l) && (i4[4] = r4, i4[5] = n4, G.n = l, c3 = 0));
        }
        if (o4 || r4 > 1) return a;
        throw y = true, n4;
      }
      return function(o4, p2, l) {
        if (f2 > 1) throw TypeError("Generator is already running");
        for (y && 1 === p2 && d(p2, l), c3 = p2, u3 = l; (t = c3 < 2 ? e : u3) || !y; ) {
          i3 || (c3 ? c3 < 3 ? (c3 > 1 && (G.n = -1), d(c3, u3)) : G.n = u3 : G.v = u3);
          try {
            if (f2 = 2, i3) {
              if (c3 || (o4 = "next"), t = i3[o4]) {
                if (!(t = t.call(i3, u3))) throw TypeError("iterator result is not an object");
                if (!t.done) return t;
                u3 = t.value, c3 < 2 && (c3 = 0);
              } else 1 === c3 && (t = i3.return) && t.call(i3), c3 < 2 && (u3 = TypeError("The iterator does not provide a '" + o4 + "' method"), c3 = 1);
              i3 = e;
            } else if ((t = (y = G.n < 0) ? u3 : r3.call(n3, G)) !== a) break;
          } catch (t2) {
            i3 = e, c3 = 1, u3 = t2;
          } finally {
            f2 = 1;
          }
        }
        return { value: t, done: y };
      };
    })(r2, o2, i2), true), u2;
  }
  var a = {};
  function Generator() {
  }
  function GeneratorFunction() {
  }
  function GeneratorFunctionPrototype() {
  }
  t = Object.getPrototypeOf;
  var c = [][n] ? t(t([][n]())) : (_regeneratorDefine2(t = {}, n, function() {
    return this;
  }), t), u = GeneratorFunctionPrototype.prototype = Generator.prototype = Object.create(c);
  function f(e2) {
    return Object.setPrototypeOf ? Object.setPrototypeOf(e2, GeneratorFunctionPrototype) : (e2.__proto__ = GeneratorFunctionPrototype, _regeneratorDefine2(e2, o, "GeneratorFunction")), e2.prototype = Object.create(u), e2;
  }
  return GeneratorFunction.prototype = GeneratorFunctionPrototype, _regeneratorDefine2(u, "constructor", GeneratorFunctionPrototype), _regeneratorDefine2(GeneratorFunctionPrototype, "constructor", GeneratorFunction), GeneratorFunction.displayName = "GeneratorFunction", _regeneratorDefine2(GeneratorFunctionPrototype, o, "GeneratorFunction"), _regeneratorDefine2(u), _regeneratorDefine2(u, o, "Generator"), _regeneratorDefine2(u, n, function() {
    return this;
  }), _regeneratorDefine2(u, "toString", function() {
    return "[object Generator]";
  }), (_regenerator = function _regenerator2() {
    return { w: i, m: f };
  })();
}
function _regeneratorDefine2(e, r, n, t) {
  var i = Object.defineProperty;
  try {
    i({}, "", {});
  } catch (e2) {
    i = 0;
  }
  _regeneratorDefine2 = function _regeneratorDefine(e2, r2, n2, t2) {
    function o(r3, n3) {
      _regeneratorDefine2(e2, r3, function(e3) {
        return this._invoke(r3, n3, e3);
      });
    }
    r2 ? i ? i(e2, r2, { value: n2, enumerable: !t2, configurable: !t2, writable: !t2 }) : e2[r2] = n2 : (o("next", 0), o("throw", 1), o("return", 2));
  }, _regeneratorDefine2(e, r, n, t);
}
function asyncGeneratorStep(n, t, e, r, o, a, c) {
  try {
    var i = n[a](c), u = i.value;
  } catch (n2) {
    return void e(n2);
  }
  i.done ? t(u) : Promise.resolve(u).then(r, o);
}
function _asyncToGenerator(n) {
  return function() {
    var t = this, e = arguments;
    return new Promise(function(r, o) {
      var a = n.apply(t, e);
      function _next(n2) {
        asyncGeneratorStep(a, r, o, _next, _throw, "next", n2);
      }
      function _throw(n2) {
        asyncGeneratorStep(a, r, o, _next, _throw, "throw", n2);
      }
      _next(void 0);
    });
  };
}
import { defineComponent, h, setup as _setup, $prefix, createComponent } from "@opentiny/vue-common";
import { createVNode, withDirectives, vShow } from "vue";
import { api, renderless } from "@opentiny/vue-renderless/notify/vue";
import { iconClose, IconSuccess, IconPrompt, iconError, iconWarningTriangle } from "@opentiny/vue-icon";
import "@opentiny/vue-theme/notify/index.css";
var _sfc_main = /* @__PURE__ */ defineComponent({
  name: $prefix + "Notify",
  props: {
    beforeClose: Function,
    closeIcon: {
      type: Object,
      default: function _default() {
        return iconClose();
      }
    },
    customClass: [String, Object],
    duration: {
      type: Number,
      default: 0
    },
    message: [String, Function],
    onClose: Function,
    position: {
      type: String,
      default: "bottom-right"
    },
    showClose: {
      type: Boolean,
      default: true
    },
    showIcon: {
      type: Boolean,
      default: true
    },
    statusIcon: Object,
    title: [String, Function],
    type: String
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  },
  render: function render() {
    var clearTimer = this.clearTimer, click = this.click, close = this.close, closeIcon = this.closeIcon, message = this.message, showClose = this.showClose;
    var showIcon = this.showIcon, startTimer = this.startTimer, state = this.state, statusIcon = this.statusIcon, title = this.title, type = this.type;
    var _ref = {}, _ref$closeVNode = _ref.closeVNode, closeVNode = _ref$closeVNode === void 0 ? null : _ref$closeVNode, _ref$iconVNode = _ref.iconVNode, iconVNode = _ref$iconVNode === void 0 ? null : _ref$iconVNode, _ref$notifyContent = _ref.notifyContent, notifyContent = _ref$notifyContent === void 0 ? null : _ref$notifyContent, _ref$notifyTitle = _ref.notifyTitle, notifyTitle = _ref$notifyTitle === void 0 ? null : _ref$notifyTitle;
    if (showIcon && statusIcon) {
      iconVNode = createVNode("div", {
        "class": "tiny-notify__icon-zone",
        "aria-hidden": "true"
      }, [createVNode("span", {
        "class": "tiny-notify__icon-status"
      }, [createVNode(statusIcon, {
        "class": "tiny-svg-size"
      }, null)])]);
    }
    if (showClose) {
      closeVNode = createVNode("div", {
        "class": "tiny-notify__close-zone"
      }, [createVNode("span", {
        "class": "tiny-notify__icon-close",
        "aria-label": "Close notification"
      }, [createVNode(closeIcon, {
        "class": "tiny-svg-size",
        "aria-hidden": "true",
        "onClick": close
      }, null)])]);
    }
    if (title && typeof title === "string") {
      notifyTitle = h("div", {
        class: "tiny-notify__title",
        id: state.titleId
      }, title);
    } else if (typeof title === "function") {
      notifyTitle = title(h, {
        vm: this,
        titleClass: "tiny-notify__title"
      });
    }
    if (typeof message === "string") {
      notifyContent = h("span", {
        class: "tiny-notify__content",
        id: state.contentId
      }, message);
    } else if (typeof message === "function") {
      notifyContent = message(h, {
        vm: this,
        messageClass: "tiny-notify__content"
      });
    }
    var msgVNode = createVNode("div", {
      "class": "tiny-notify__message-zone",
      "role": "region",
      "aria-label": "Notification content"
    }, [notifyTitle ? createVNode("div", {
      "class": "tiny-notify__title-wrapper"
    }, [notifyTitle]) : null, createVNode("div", {
      "class": "tiny-notify__content-wrapper"
    }, [notifyContent])]);
    return withDirectives(createVNode("div", {
      "class": ["tiny-notify", "tiny-notify--" + type, showIcon ? "" : "tiny-notify--no-icon", showClose ? "" : "tiny-notify--no-close", state.position, state.customClass],
      "role": "alert",
      "aria-hidden": !state.visible,
      "aria-labelledby": state.titleId,
      "aria-describedby": state.contentId,
      "style": state.positionStyle,
      "onMouseenter": clearTimer,
      "onMouseleave": startTimer,
      "onClick": click
    }, [[iconVNode, msgVNode, closeVNode]]), [[vShow, "state.visible"]]);
  }
});
var seed = 1;
var instances = [];
var IconMap = {
  warning: iconWarningTriangle(),
  error: iconError(),
  info: IconPrompt(),
  success: IconSuccess()
};
var durationMap = {
  info: 5e3,
  success: 5e3,
  warning: 1e4,
  error: 1e4
};
var positionList = ["top-right", "bottom-right", "top-left", "bottom-left"];
var debounce = function debounce2(fn, debounceDelay) {
  var timer = null;
  return /* @__PURE__ */ _asyncToGenerator(/* @__PURE__ */ _regenerator().m(function _callee() {
    var _this = this;
    var _len, args, _key, instance, _args = arguments;
    return _regenerator().w(function(_context) {
      while (1) switch (_context.n) {
        case 0:
          for (_len = _args.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = _args[_key];
          }
          if (timer) {
            clearTimeout(timer);
          }
          instance = null;
          _context.n = 1;
          return new Promise(function(resolve) {
            timer = setTimeout(function() {
              instance = fn.apply(_this, args);
              timer = null;
              resolve();
            }, debounceDelay);
          });
        case 1:
          return _context.a(2, instance);
      }
    }, _callee);
  }));
};
var notify = function notify2(options) {
  if (!Object.keys(IconMap).includes(options.type)) {
    options.type = "info";
  }
  options.duration = options.duration ? options.duration : durationMap[options.type];
  options.position = !positionList.includes(options.position) ? "bottom-right" : options.position;
  !options.statusIcon && options.type && (options.statusIcon = IconMap[options.type]);
  var id = "notify_" + seed++;
  var userOnClose = options.onClose;
  var position = options.position;
  options.onClose = function() {
    Notify.close(id, userOnClose);
  };
  var instance = createComponent({
    el: document.createElement("div"),
    propsData: options,
    component: _sfc_main
  });
  instance.id = id;
  document.body.appendChild(instance.$el);
  var verticalOffset = options.offset || 0;
  instances.filter(function(item) {
    return item.state.position === position;
  }).forEach(function(item) {
    verticalOffset += item.$el.offsetHeight + 16;
  });
  verticalOffset += options.verticalOffset ? Number(options.verticalOffset) : 16;
  instances.push(instance);
  instance.dom = instance.$el;
  instance.dom.style.zIndex = instance.getZindex().toString();
  instance.state.verticalOffset = verticalOffset;
  instance.state.visible = true;
  if (verticalOffset + instance.$el.offsetHeight > window.innerHeight) {
    instances[0] && instances[0].close(instances[0].id);
  }
  return instance;
};
var Notify = function Notify2(options) {
  var debounceDelay = options.debounceDelay;
  if (debounceDelay) {
    return debounce(function() {
      return notify(options);
    }, debounceDelay);
  } else {
    return notify(options);
  }
};
Notify.close = function(id, userOnClose) {
  var _instance$$el$parentN;
  var index = -1;
  var len = instances.length;
  var instance;
  for (var i = 0; i < len; i++) {
    var tmp = instances[i];
    if (tmp.id === id) {
      index = i;
      instance = tmp;
      break;
    }
  }
  if (!instance) {
    return;
  }
  typeof userOnClose === "function" && userOnClose(instance);
  var lastHeight = instance.$el.offsetHeight;
  (_instance$$el$parentN = instance.$el.parentNode) == null ? void 0 : _instance$$el$parentN.removeChild(instance.$el);
  instances.splice(index, 1);
  if (len <= 1) {
    return;
  }
  var removedPosition = instance.position;
  var copys = instances.slice(index);
  var verticalOffset = instance.state.verticalOffset ? Number(instance.state.verticalOffset) : 16;
  instances.filter(function(item) {
    return item.state.position === removedPosition;
  }).forEach(function(item) {
    item.state.verticalOffset = verticalOffset;
    verticalOffset += item.$el.offsetHeight + 16;
  });
  copys.forEach(function(copy) {
    if (copy.position === removedPosition) {
      var height = parseInt(copy.dom.style[instance.state.verticalProperty], 10) - lastHeight - 16;
      copy.dom.style[instance.state.verticalProperty] = height + "px";
    }
  });
};
Notify.closeAll = function() {
  var copys = instances.slice(0);
  copys = copys.reverse();
  copys.forEach(function(instance) {
    instance.close();
  });
};
export {
  Notify as default
};
